--- orig/BiDirPolicyC.h	2005-02-18 12:27:27.078513600 +0100
+++ BiDirPolicyC.h	2005-02-18 12:27:27.278801600 +0100
@@ -33,6 +33,9 @@
 
 #include /**/ "ace/pre.h"
 
+#ifndef TAO_BIDIRGIOP_SAFE_INCLUDE
+#error "You should not include BiDirPolicyC.h directly, use BiDirGIOP.h"
+#endif /* !TAO_BIDIRGIOP_SAFE_INCLUDE */
 
 #include "ace/config-all.h"
 
@@ -47,6 +50,8 @@
 #include "tao/Object.h"
 #include "tao/Objref_VarOut_T.h"
 
+#include "tao/PolicyC.h"
+
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
--- orig/BiDirPolicyC.cpp	2005-02-18 12:27:27.078513600 +0100
+++ BiDirPolicyC.cpp	2005-02-18 12:27:27.288816000 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "BiDirPolicyC.h"
+#include "BiDirGIOP.h"
 #include "tao/CDR.h"
 #include "ace/OS_NS_string.h"
 
