--- orig/POAManagerC.h	2005-04-11 16:44:00.771974400 +0200
+++ POAManagerC.h	2005-04-11 16:44:04.216928000 +0200
@@ -207,6 +207,8 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
+
     virtual void hold_requests (
         ::CORBA::Boolean wait_for_completion
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -241,6 +243,8 @@
         ::PortableServer::POAManager::AdapterInactive
       )) = 0;
     
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
