--- orig/ForwardRequestC.h	2005-03-31 13:47:21.174958400 +0200
+++ ForwardRequestC.h	2005-03-31 13:47:23.698587200 +0200
@@ -59,6 +59,8 @@
 #pragma option push -w-rvl -w-rch -w-ccc -w-inl
 #endif /* __BORLANDC__ */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:48
 
@@ -153,6 +155,8 @@
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:947
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 #if defined (__ACE_INLINE__)
 #include "ForwardRequestC.inl"
 #endif /* defined INLINE */
--- orig/ForwardRequestC.cpp	2005-03-31 13:47:21.174958400 +0200
+++ ForwardRequestC.cpp	2005-03-31 13:47:23.698587200 +0200
@@ -46,6 +46,8 @@
 #include "ForwardRequestC.inl"
 #endif /* !defined INLINE */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
@@ -219,3 +221,5 @@
 #elif defined (ACE_HAS_TEMPLATE_INSTANTIATION_PRAGMA)
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ForwardRequestA.cpp	2005-03-31 13:47:21.174958400 +0200
+++ ForwardRequestA.cpp	2005-03-31 13:47:23.698587200 +0200
@@ -60,6 +60,8 @@
 
 
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/any_op_cs.cpp:50
 
@@ -160,3 +162,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
