--- orig/ImplicitActivationPolicyC.h	2005-04-11 16:44:00.311312000 +0200
+++ ImplicitActivationPolicyC.h	2005-04-11 16:46:31.779112000 +0200
@@ -78,6 +78,7 @@
     NO_IMPLICIT_ACTIVATION
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef ImplicitActivationPolicyValue &ImplicitActivationPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -207,6 +208,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 } // module PortableServer
 
 // TAO_IDL - Generated from
@@ -215,6 +218,7 @@
 // Traits specializations.
 namespace TAO
 {
+#if (TAO_HAS_MINIMUM_POA == 0)
 
 #if !defined (_PORTABLESERVER_IMPLICITACTIVATIONPOLICY__TRAITS_CH_)
 #define _PORTABLESERVER_IMPLICITACTIVATIONPOLICY__TRAITS_CH_
@@ -236,11 +240,15 @@
   };
 
 #endif /* end #if !defined */
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
 }
 
 // TAO_IDL - Generated from
 // be\be_visitor_enum/any_op_ch.cpp:51
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ImplicitActivationPolicyValue);
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ImplicitActivationPolicyValue &);
 
@@ -257,6 +265,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, PortableServer::ImplicitActivationPolicyValue);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::ImplicitActivationPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:955
 
--- orig/ImplicitActivationPolicyC.cpp	2005-04-11 16:44:00.311312000 +0200
+++ ImplicitActivationPolicyC.cpp	2005-04-11 16:45:07.347705600 +0200
@@ -40,6 +40,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -211,3 +213,5 @@
   
   return _tao_success;
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ImplicitActivationPolicyA.cpp	2005-04-11 16:44:00.311312000 +0200
+++ ImplicitActivationPolicyA.cpp	2005-04-11 16:45:02.230347200 +0200
@@ -36,6 +36,8 @@
 #include "tao/Any_Impl_T.h"
 #include "tao/Any_Basic_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/enum_typecode.cpp:34
 
@@ -177,3 +179,5 @@
         _tao_elem
       );
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
