--- orig/POAManagerC.h	2005-03-25 15:45:59.657308800 +0100
+++ POAManagerC.h	2005-03-25 15:46:16.331284800 +0100
@@ -225,6 +225,8 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
+
     virtual void hold_requests (
         ::CORBA::Boolean wait_for_completion
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -259,6 +261,8 @@
         ::PortableServer::POAManager::AdapterInactive
       )) = 0;
     
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
