--- orig/PS_ForwardC.h	2004-09-08 11:42:25.290763800 +0200
+++ PS_ForwardC.h	2004-09-08 11:49:23.702410200 +0200
@@ -70,11 +70,23 @@
 #pragma option push -w-rvl -w-rch -w-ccc -w-inl
 #endif /* __BORLANDC__ */
 
+class TAO_ServantBase;
+class TAO_ServantBase_var;
+class TAO_RefCountServantBase;
+class TAO_Local_ServantBase;
+class TAO_POA;
+
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:48
 
 namespace PortableServer
 {
+   typedef TAO_ServantBase ServantBase;
+   typedef TAO_ServantBase_var ServantBase_var;
+   typedef TAO_RefCountServantBase RefCountServantBase;
+   typedef ServantBase *Servant;
+
+   typedef TAO_Local_ServantBase LocalServantBase;
   
   // TAO_IDL - Generated from
   // be\be_visitor_typedef/typedef_ch.cpp:435
