--- orig/RequestProcessingPolicyC.h	2005-03-25 15:46:07.668828800 +0100
+++ RequestProcessingPolicyC.h	2005-03-25 15:46:17.252609600 +0100
@@ -97,6 +97,7 @@
     USE_SERVANT_MANAGER
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef RequestProcessingPolicyValue &RequestProcessingPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -226,11 +227,13 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
 } // module PortableServer
 
 // TAO_IDL - Generated from
 // be\be_visitor_traits.cpp:59
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 // Traits specializations.
 namespace TAO
 {
@@ -276,6 +279,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, PortableServer::RequestProcessingPolicyValue);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::RequestProcessingPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:969
 
--- orig/RequestProcessingPolicyC.cpp	2005-03-25 15:46:07.668828800 +0100
+++ RequestProcessingPolicyC.cpp	2005-03-25 15:46:17.282652800 +0100
@@ -44,6 +44,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -254,3 +256,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/RequestProcessingPolicyA.cpp	2005-03-25 15:46:07.668828800 +0100
+++ RequestProcessingPolicyA.cpp	2005-03-25 17:36:54.516526400 +0100
@@ -37,6 +37,8 @@
 #include "tao/Any_Impl_T.h"
 #include "tao/Any_Basic_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/enum_typecode.cpp:34
 
@@ -205,3 +207,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
