--- orig/RequestProcessingPolicyC.h	2005-02-18 11:19:46.359486400 +0100
+++ RequestProcessingPolicyC.h	2005-02-18 11:19:48.843057600 +0100
@@ -97,6 +97,7 @@
     USE_SERVANT_MANAGER
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef RequestProcessingPolicyValue &RequestProcessingPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -224,11 +225,13 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
 } // module PortableServer
 
 // TAO_IDL - Generated from
 // be\be_visitor_traits.cpp:59
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 // Traits specializations.
 namespace TAO
 {
@@ -274,6 +277,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, const PortableServer::RequestProcessingPolicyValue &);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::RequestProcessingPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:961
 
--- orig/RequestProcessingPolicyC.cpp	2005-02-18 11:19:46.359486400 +0100
+++ RequestProcessingPolicyC.cpp	2005-02-18 11:19:48.853072000 +0100
@@ -44,6 +44,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:66
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -253,3 +255,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/RequestProcessingPolicyA.cpp	2005-02-18 11:19:46.359486400 +0100
+++ RequestProcessingPolicyA.cpp	2005-02-18 11:19:48.863086400 +0100
@@ -103,6 +103,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:295
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_RequestProcessingPolicy[] =
 {
     TAO_ENCAP_BYTE_ORDER, // byte order
@@ -266,3 +268,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
