--- orig/ServantManagerC.h	2005-03-25 15:46:09.541521600 +0100
+++ ServantManagerC.h	2005-03-25 15:46:17.983660800 +0100
@@ -70,6 +70,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_root/root_ch.cpp:62
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace TAO
 {
   class Collocation_Proxy_Broker;
@@ -214,6 +216,8 @@
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ServantManager_ptr *); // non-copying
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ServantManager_ptr &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:969
 
--- orig/ServantManagerC.cpp	2005-03-25 15:46:09.541521600 +0100
+++ ServantManagerC.cpp	2005-03-25 15:46:18.023718400 +0100
@@ -49,6 +49,7 @@
 {
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 
 // TAO_IDL - Generated from
 // be\be_visitor_interface/interface_cs.cpp:60
@@ -229,3 +230,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantManagerA.cpp	2005-03-25 15:46:09.541521600 +0100
+++ ServantManagerA.cpp	2005-03-25 15:46:18.063776000 +0100
@@ -34,6 +34,8 @@
 #include "tao/Any.h"
 #include "tao/Any_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/objref_typecode.cpp:73
 
@@ -136,3 +138,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
