--- orig/ServantRetentionPolicyC.h	2005-03-31 13:47:21.555505600 +0200
+++ ServantRetentionPolicyC.h	2005-03-31 13:47:23.738644800 +0200
@@ -78,6 +78,7 @@
     NON_RETAIN
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef ServantRetentionPolicyValue &ServantRetentionPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -207,11 +208,13 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
 } // module PortableServer
 
 // TAO_IDL - Generated from
 // be\be_visitor_traits.cpp:59
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 // Traits specializations.
 namespace TAO
 {
@@ -257,6 +260,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, PortableServer::ServantRetentionPolicyValue);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::ServantRetentionPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:947
 
--- orig/ServantRetentionPolicyC.cpp	2005-03-31 13:47:21.555505600 +0200
+++ ServantRetentionPolicyC.cpp	2005-03-31 13:47:23.738644800 +0200
@@ -44,6 +44,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -254,3 +256,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantRetentionPolicyA.cpp	2005-03-31 13:47:21.555505600 +0200
+++ ServantRetentionPolicyA.cpp	2005-03-31 13:47:23.738644800 +0200
@@ -37,6 +37,8 @@
 #include "tao/Any_Impl_T.h"
 #include "tao/Any_Basic_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/enum_typecode.cpp:34
 
@@ -204,3 +206,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
