--- orig/ServantLocatorC.h	2005-02-18 11:19:46.770076800 +0100
+++ ServantLocatorC.h	2005-02-18 11:19:49.003288000 +0100
@@ -76,6 +76,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_root/root_ch.cpp:62
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace TAO
 {
   class Collocation_Proxy_Broker;
@@ -123,6 +125,8 @@
     : public virtual ::PortableServer::ServantManager
   {
   public:
+    typedef void *Cookie;
+
     typedef ServantLocator_ptr _ptr_type;
     typedef ServantLocator_var _var_type;
     
@@ -248,6 +252,8 @@
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ServantLocator_ptr *); // non-copying
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ServantLocator_ptr &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:961
 
--- orig/ServantLocatorC.cpp	2005-02-18 11:19:46.770076800 +0100
+++ ServantLocatorC.cpp	2005-02-18 11:19:49.013302400 +0100
@@ -45,6 +45,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:66
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -232,3 +234,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantLocatorA.cpp	2005-02-18 11:19:46.770076800 +0100
+++ ServantLocatorA.cpp	2005-02-18 11:19:49.023316800 +0100
@@ -34,6 +34,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:295
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ServantLocator[] =
 {
     TAO_ENCAP_BYTE_ORDER, // byte order
@@ -155,3 +157,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
