--- orig/ServantManagerC.h	2005-03-31 13:47:20.824454400 +0200
+++ ServantManagerC.h	2005-03-31 13:49:26.244800000 +0200
@@ -63,6 +63,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:48
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace PortableServer
 {
   
@@ -196,6 +198,8 @@
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ServantManager_ptr *); // non-copying
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ServantManager_ptr &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:947
 
--- orig/ServantManagerC.cpp	2005-03-31 13:47:20.834468800 +0200
+++ ServantManagerC.cpp	2005-03-31 13:47:23.668544000 +0200
@@ -49,6 +49,7 @@
 {
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 
 // TAO_IDL - Generated from
 // be\be_visitor_interface/interface_cs.cpp:60
@@ -229,3 +230,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantManagerA.cpp	2005-03-31 13:47:20.834468800 +0200
+++ ServantManagerA.cpp	2005-03-31 13:47:23.668544000 +0200
@@ -34,6 +34,8 @@
 #include "tao/Any.h"
 #include "tao/Any_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/objref_typecode.cpp:73
 
@@ -136,3 +138,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
