--- orig/ServantRetentionPolicyC.h	2005-02-18 11:19:47.481099200 +0100
+++ ServantRetentionPolicyC.h	2005-02-18 11:19:49.193561600 +0100
@@ -96,6 +96,7 @@
     NON_RETAIN
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef ServantRetentionPolicyValue &ServantRetentionPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -223,11 +224,13 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
 } // module PortableServer
 
 // TAO_IDL - Generated from
 // be\be_visitor_traits.cpp:59
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 // Traits specializations.
 namespace TAO
 {
@@ -273,6 +276,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, const PortableServer::ServantRetentionPolicyValue &);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::ServantRetentionPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:961
 
--- orig/ServantRetentionPolicyC.cpp	2005-02-18 11:19:47.481099200 +0100
+++ ServantRetentionPolicyC.cpp	2005-02-18 11:19:49.203576000 +0100
@@ -44,6 +44,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:66
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -253,3 +255,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantRetentionPolicyA.cpp	2005-02-18 11:19:47.481099200 +0100
+++ ServantRetentionPolicyA.cpp	2005-02-18 11:19:49.213590400 +0100
@@ -35,6 +35,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:295
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ServantRetentionPolicyValue[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -252,3 +254,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
