--- orig/ThreadPolicyC.h	2005-03-31 13:47:21.855937600 +0200
+++ ThreadPolicyC.h	2005-03-31 13:47:23.778702400 +0200
@@ -78,6 +78,7 @@
     SINGLE_THREAD_MODEL
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef ThreadPolicyValue &ThreadPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -207,12 +208,14 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
 } // module PortableServer
 
 // TAO_IDL - Generated from
 // be\be_visitor_traits.cpp:59
 
 // Traits specializations.
+#if (TAO_HAS_MINIMUM_POA == 0)
 namespace TAO
 {
 
@@ -257,6 +260,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, PortableServer::ThreadPolicyValue);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::ThreadPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:947
 
--- orig/ThreadPolicyC.cpp	2005-03-31 13:47:21.855937600 +0200
+++ ThreadPolicyC.cpp	2005-03-31 13:47:23.778702400 +0200
@@ -44,6 +44,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -254,3 +256,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ThreadPolicyA.cpp	2005-03-31 13:47:21.855937600 +0200
+++ ThreadPolicyA.cpp	2005-03-31 13:47:23.788716800 +0200
@@ -37,6 +37,8 @@
 #include "tao/Any_Impl_T.h"
 #include "tao/Any_Basic_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/enum_typecode.cpp:34
 
@@ -204,3 +206,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
