--- orig/ThreadPolicyC.h	2005-02-18 11:19:47.691401600 +0100
+++ ThreadPolicyC.h	2005-02-18 11:19:49.263662400 +0100
@@ -96,6 +96,7 @@
     SINGLE_THREAD_MODEL
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef ThreadPolicyValue &ThreadPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -223,12 +224,14 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
 } // module PortableServer
 
 // TAO_IDL - Generated from
 // be\be_visitor_traits.cpp:59
 
 // Traits specializations.
+#if (TAO_HAS_MINIMUM_POA == 0)
 namespace TAO
 {
 
@@ -273,6 +276,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, const PortableServer::ThreadPolicyValue &);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::ThreadPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:961
 
--- orig/ThreadPolicyC.cpp	2005-02-18 11:19:47.691401600 +0100
+++ ThreadPolicyC.cpp	2005-02-18 11:19:49.273676800 +0100
@@ -44,6 +44,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:66
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -253,3 +255,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ThreadPolicyA.cpp	2005-02-18 11:19:47.691401600 +0100
+++ ThreadPolicyA.cpp	2005-02-18 11:19:49.283691200 +0100
@@ -35,6 +35,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:295
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ThreadPolicyValue[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -247,3 +249,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
