--- orig/PS_ForwardC.h	2005-04-11 16:44:01.493011200 +0200
+++ PS_ForwardC.h	2005-04-11 16:44:04.427230400 +0200
@@ -61,11 +61,25 @@
 #pragma option push -w-rvl -w-rch -w-ccc -w-inl
 #endif /* __BORLANDC__ */
 
+class TAO_ServantBase;
+class TAO_ServantBase_var;
+class TAO_RefCountServantBase;
+class TAO_Local_ServantBase;
+class TAO_Root_POA;
+class TAO_DynamicImplementation;
+
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:48
 
 namespace PortableServer
 {
+    typedef TAO_ServantBase ServantBase;
+    typedef TAO_ServantBase_var ServantBase_var;
+    typedef TAO_RefCountServantBase RefCountServantBase;
+    typedef ServantBase *Servant;
+
+    typedef TAO_Local_ServantBase LocalServantBase;
+    typedef TAO_DynamicImplementation DynamicImplementation;
   
   // TAO_IDL - Generated from
   // be\be_visitor_typedef/typedef_ch.cpp:472
