--- orig/ServantManagerC.h	2005-04-11 16:44:02.234076800 +0200
+++ ServantManagerC.h	2005-04-11 16:44:04.827806400 +0200
@@ -63,6 +63,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:48
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace PortableServer
 {
   
@@ -196,6 +198,8 @@
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ServantManager_ptr *); // non-copying
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ServantManager_ptr &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:955
 
--- orig/ServantManagerC.cpp	2005-04-11 16:44:02.234076800 +0200
+++ ServantManagerC.cpp	2005-04-11 16:50:18.164638400 +0200
@@ -45,6 +45,7 @@
 {
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 
 // TAO_IDL - Generated from
 // be\be_visitor_interface/interface_cs.cpp:60
@@ -186,3 +187,5 @@
 {
   return false;
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantManagerA.cpp	2005-04-11 16:44:02.234076800 +0200
+++ ServantManagerA.cpp	2005-04-11 16:50:12.206070400 +0200
@@ -34,6 +34,8 @@
 #include "tao/Any.h"
 #include "tao/Any_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/objref_typecode.cpp:76
 
@@ -120,3 +122,5 @@
         _tao_elem
       );
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
