--- orig/Messaging_SyncScope_PolicyC.h	Tue Sep  3 16:30:44 2002
+++ Messaging_SyncScope_PolicyC.h	Tue Sep  3 16:11:39 2002
@@ -27,13 +27,15 @@
 #define _TAO_IDL_ORIG_MESSAGING_SYNCSCOPE_POLICYC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+#include "tao/corbafwd.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 #include "messaging_export.h"
+#include "tao/PolicyC.h"
+#include "tao/Messaging_SyncScopeC.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
