--- ACE_wrappers/ace/config-macros.h~	2008-03-04 06:51:23.000000000 -0800
+++ ACE_wrappers/ace/config-macros.h	2008-06-02 13:25:50.000000000 -0700
@@ -375,7 +375,7 @@
 // should keep them quiet.
 // ============================================================================
 
-#if defined (ghs) || defined (__GNUC__) || defined (__hpux) || defined (__sgi) || defined (__DECCXX) || defined (__rational__) || defined (__USLC__) || defined (ACE_RM544) || defined (__DCC__) || defined (__PGI) || defined (__TANDEM)
+#if defined (ghs) || defined (__hpux) || defined (__sgi) || defined (__DECCXX) || defined (__rational__) || defined (__USLC__) || defined (ACE_RM544) || defined (__DCC__) || defined (__PGI) || defined (__TANDEM)
 // Some compilers complain about "statement with no effect" with (a).
 // This eliminates the warnings, and no code is generated for the null
 // conditional statement.  @note that may only be true if -O is enabled,
@@ -385,6 +385,8 @@
   #define ACE_UNUSED_ID(identifier)
   template <class T>
   inline void ACE_UNUSED_ARG(const T& ACE_UNUSED_ID(t)) { }
+#elif defined (__GNUC__)
+# define ACE_UNUSED_ARG(a) (void) (a)
 #else /* ghs || __GNUC__ || ..... */
 # define ACE_UNUSED_ARG(a) (a)
 #endif /* ghs || __GNUC__ || ..... */
