--- orig/IORC.h	2005-03-31 12:04:13.717832000 +0200
+++ IORC.h	2005-03-31 12:04:13.878062400 +0200
@@ -40,6 +40,10 @@
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
+ #ifndef TAO_IORMANIP_SAFE_INCLUDE
+ #error "You should not include IORC.h directly, use IORManip_Loader.h"
+ #endif /* !TAO_IORMANIP_SAFE_INCLUDE */
+
 #include "ior_manip_export.h"
 #include "tao/ORB.h"
 #include "tao/SystemException.h"
--- orig/IORC.cpp	2005-03-31 12:04:13.717832000 +0200
+++ IORC.cpp	2005-03-31 12:04:13.878062400 +0200
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:291
 
 
-#include "IORC.h"
+#include "IORManip_Loader.h"
 #include "tao/CDR.h"
 #include "tao/Null_RefCount_Policy.h"
 #include "tao/TypeCode_Constants.h"
--- orig/IORA.cpp	2005-03-31 12:04:13.717832000 +0200
+++ IORA.cpp	2005-03-31 12:05:19.502425600 +0200
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "IORC.h"
+#include "IORManip_Loader.h"
 #include "tao/Null_RefCount_Policy.h"
 #include "tao/TypeCode_Constants.h"
 #include "tao/Alias_TypeCode.h"
