--- PortableServerC.i	Tue Jul 11 01:25:50 2000
+++ PortableServerC.i.mod	Tue Jul 11 01:16:42 2000
@@ -739,6 +744,8 @@
 // *************************************************************
 
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 #if !defined (_PORTABLESERVER_THREADPOLICY___VAR_CI_)
 #define _PORTABLESERVER_THREADPOLICY___VAR_CI_
 
@@ -908,6 +920,7 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
 
 #if !defined (_PORTABLESERVER_LIFESPANPOLICY___VAR_CI_)
 #define _PORTABLESERVER_LIFESPANPOLICY___VAR_CI_
@@ -1418,6 +1446,7 @@
 
 #endif /* end #if !defined */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 
 #if !defined (_PORTABLESERVER_IMPLICITACTIVATIONPOLICY___VAR_CI_)
 #define _PORTABLESERVER_IMPLICITACTIVATIONPOLICY___VAR_CI_
@@ -1928,6 +1972,8 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 // *************************************************************
 // Inline operations for exception PortableServer::POAManager::AdapterInactive
 // *************************************************************
@@ -2102,6 +2153,7 @@
 
 #endif /* end #if !defined */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 
 #if !defined (_PORTABLESERVER_ADAPTERACTIVATOR___VAR_CI_)
 #define _PORTABLESERVER_ADAPTERACTIVATOR___VAR_CI_
@@ -2782,6 +2854,8 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 // *************************************************************
 // Inline operations for exception PortableServer::POA::AdapterAlreadyExists
 // *************************************************************
@@ -3021,6 +3100,8 @@
 
 #endif /* _TAO_CDR_OP_PortableServer_ObjectId_I_ */
 
+#if (TAO_HAS_MINIMUM_CORBA == 0)
+
 ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const PortableServer::ForwardRequest &_tao_aggregate)
 {
   // first marshal the repository ID
@@ -3069,6 +3150,10 @@
 
 #endif /* ACE_LACKS_IOSTREAM_TOTALLY */
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const PortableServer::ThreadPolicyValue &_tao_enumval)
 {
   return strm.write_ulong ((CORBA::ULong) _tao_enumval);
@@ -3086,6 +3171,8 @@
     return 0;
 }
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const PortableServer::LifespanPolicyValue &_tao_enumval)
 {
   return strm.write_ulong ((CORBA::ULong) _tao_enumval);
@@ -3137,6 +3224,8 @@
     return 0;
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const PortableServer::ImplicitActivationPolicyValue &_tao_enumval)
 {
   return strm.write_ulong ((CORBA::ULong) _tao_enumval);
@@ -3188,3 +3277,4 @@
     return 0;
 }
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
