--- orig/PollableC.i	Fri Apr 28 12:20:09 2000
+++ PollableC.i	Fri Apr 28 12:02:10 2000
@@ -69,14 +69,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 CORBA_Pollable_var::operator const ::CORBA_Pollable_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-CORBA_Pollable_var::operator ::CORBA_Pollable_ptr &() // cast 
+ACE_INLINE
+CORBA_Pollable_var::operator ::CORBA_Pollable_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -168,7 +168,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 CORBA_Pollable_out::operator ::CORBA_Pollable_ptr &() // cast
 {
   return this->ptr_;
@@ -251,14 +251,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 CORBA_DIIPollable_var::operator const ::CORBA_DIIPollable_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-CORBA_DIIPollable_var::operator ::CORBA_DIIPollable_ptr &() // cast 
+ACE_INLINE
+CORBA_DIIPollable_var::operator ::CORBA_DIIPollable_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -350,7 +350,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 CORBA_DIIPollable_out::operator ::CORBA_DIIPollable_ptr &() // cast
 {
   return this->ptr_;
@@ -433,14 +433,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 CORBA_PollableSet_var::operator const ::CORBA_PollableSet_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-CORBA_PollableSet_var::operator ::CORBA_PollableSet_ptr &() // cast 
+ACE_INLINE
+CORBA_PollableSet_var::operator ::CORBA_PollableSet_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -532,7 +532,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 CORBA_PollableSet_out::operator ::CORBA_PollableSet_ptr &() // cast
 {
   return this->ptr_;
