--- idl.yy.cpp.orig	2009-06-23 13:18:22.000000000 -0500
+++ idl.yy.cpp	2009-06-23 13:49:49.000000000 -0500
@@ -46,7 +46,7 @@
 
 /* C99 systems have <inttypes.h>. Non-C99 systems may or may not. */
 
-#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L || defined (__HP_aCC)
 
 /* C99 says to define __STDC_LIMIT_MACROS before including stdint.h,
  * if you want the limit (max/min) macros for int types.
@@ -54,6 +54,7 @@
 #ifndef __STDC_LIMIT_MACROS
 #define __STDC_LIMIT_MACROS 1
 #endif
+#include <inttypes.h>
 
 typedef int8_t flex_int8_t;
 typedef uint8_t flex_uint8_t;

 
