--- lex.yy.cpp.orig	Wed Dec 29 07:18:42 1999
+++ lex.yy.cpp	Wed Dec 29 07:17:42 1999
@@ -963,7 +963,7 @@
 
 /* Code executed at the end of each rule. */
 #ifndef TAO_YY_BREAK
-#define TAO_YY_BREAK break;
+#define TAO_YY_BREAK ACE_NOTREACHED (break;)
 #endif
 
 #define TAO_YY_RULE_SETUP \
@@ -975,7 +975,7 @@
 TAO_YY_DECL
 	{
 	register tao_yy_state_type tao_yy_current_state;
-	register char *tao_yy_cp, *tao_yy_bp;
+	register char *tao_yy_cp = 0, *tao_yy_bp = 0;
 	register int tao_yy_act;
 
 #line 126 "fe/idl.ll"
@@ -1479,7 +1479,7 @@
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		  idl_store_pragma(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 68:
 #line 283 "fe/idl.ll"
 case 69:
@@ -1488,7 +1488,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 70:
 #line 287 "fe/idl.ll"
 case 71:
@@ -1497,7 +148897,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 72:
 #line 291 "fe/idl.ll"
 case 73:
@@ -1506,7 +1506,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 74:
 #line 295 "fe/idl.ll"
 case 75:
@@ -1515,7 +1515,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 	        }
-	TAO_YY_BREAK
+	break;
 case 76:
 #line 299 "fe/idl.ll"
 case 77:
@@ -1525,7 +1525,7 @@
 		  /* ignore cpp ident */
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 78:
 TAO_YY_RULE_SETUP
 #line 303 "fe/idl.ll"
@@ -1533,7 +1533,7 @@
 		  /* ignore comments */
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 79:
 TAO_YY_RULE_SETUP
 #line 307 "fe/idl.ll"
@@ -1551,19 +1551,19 @@
 		    }
 	          }
 	        }
-	TAO_YY_BREAK
+	break;
 case 80:
 TAO_YY_RULE_SETUP
 #line 321 "fe/idl.ll"
 ;
-	TAO_YY_BREAK
+  break;
 case 81:
 TAO_YY_RULE_SETUP
 #line 322 "fe/idl.ll"
 {
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 82:
 TAO_YY_RULE_SETUP
 #line 325 "fe/idl.ll"
@@ -1573,7 +1573,7 @@
 TAO_YY_RULE_SETUP
 #line 327 "fe/idl.ll"
 TAO_YY_ECHO;
-	TAO_YY_BREAK
+  break;
 case TAO_YY_STATE_EOF(INITIAL):
 	tao_yyterminate();
 
