--- lex.yy.cpp	Fri Jun 11 12:09:33 1999
+++ lex.yy.cpp.new	Fri Jun 11 12:22:49 1999
@@ -899,7 +899,7 @@
 
 /* Code executed at the end of each rule. */
 #ifndef TAO_YY_BREAK
-#define TAO_YY_BREAK break;
+#define TAO_YY_BREAK ACE_NOTREACHED (break;)
 #endif
 
 #define TAO_YY_RULE_SETUP \
@@ -1353,7 +1353,7 @@
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		  idl_store_pragma(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 61:
 #line 243 "fe/idl.ll"
 case 62:
@@ -1362,7 +1362,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 63:
 #line 247 "fe/idl.ll"
 case 64:
@@ -1371,7 +1371,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 65:
 #line 251 "fe/idl.ll"
 case 66:
@@ -1380,7 +1380,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 67:
 #line 255 "fe/idl.ll"
 case 68:
@@ -1389,7 +1389,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 	        }
-	TAO_YY_BREAK
+	break;
 case 69:
 #line 259 "fe/idl.ll"
 case 70:
@@ -1399,7 +1399,7 @@
 		  /* ignore cpp ident */
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 71:
 TAO_YY_RULE_SETUP
 #line 263 "fe/idl.ll"
@@ -1407,7 +1407,7 @@
 		  /* ignore comments */
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 72:
 TAO_YY_RULE_SETUP
 #line 267 "fe/idl.ll"
@@ -1425,7 +1425,7 @@
 		    }
 	          }
 	        }
-	TAO_YY_BREAK
+	break;
 case 73:
 TAO_YY_RULE_SETUP
 #line 281 "fe/idl.ll"
@@ -1437,7 +1437,7 @@
 {
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 75:
 TAO_YY_RULE_SETUP
 #line 285 "fe/idl.ll"
