--- orig/MessagingC.h	2005-02-07 11:55:22.321433600 +0100
+++ MessagingC.h	2005-02-07 11:52:07.681555200 +0100
@@ -36,6 +36,10 @@
 
 #include "ace/config-all.h"
 
+#ifndef TAO_MESSAGING_SAFE_INCLUDE
+#error "You should not include MessagingC.h directly, use Messaging.h"
+#endif /* !TAO_MESSAGING_SAFE_INCLUDE */
+
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
--- orig/MessagingC.cpp	2005-02-07 11:55:22.321433600 +0100
+++ MessagingC.cpp	2005-02-07 11:52:41.349968000 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "MessagingC.h"
+#include "Messaging.h"
 #include "tao/CDR.h"
 #include "tao/Object_T.h"
 #include "tao/ORB_Core.h"
--- orig/MessagingA.cpp	2005-02-07 11:55:22.321433600 +0100
+++ MessagingA.cpp	2005-02-07 12:00:18.076708800 +0100
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "MessagingC.h"
+#include "Messaging.h"
 #include "tao/Typecode.h"
 #include "tao/CDR.h"
 #include "tao/Any.h"
