--- orig/BiDirPolicyC.h	2004-11-09 16:09:54.752499200 +0100
+++ BiDirPolicyC.h	2004-11-09 16:14:36.808075200 +0100
@@ -33,6 +33,9 @@
 
 #include /**/ "ace/pre.h"
 
+#ifndef TAO_BIDIRGIOP_SAFE_INCLUDE
+#error "You should not include BiDirPolicyC.h directly, use BiDirGIOP.h"
+#endif /* !TAO_BIDIRGIOP_SAFE_INCLUDE */
 
 #include "ace/config-all.h"
 
@@ -47,6 +50,8 @@
 #include "tao/Object.h"
 #include "tao/Objref_VarOut_T.h"
 
+#include "tao/PolicyC.h"
+
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
