--- lex.yy.cpp	Fri Jan 25 11:06:56 2002
+++ lex.yy.cpp.mod	Fri Jan 25 11:09:03 2002
@@ -987,7 +987,7 @@
 /* Code executed at the end of each rule. */
 #ifndef TAO_YY_BREAK
-#define TAO_YY_BREAK break;
+#define TAO_YY_BREAK ACE_NOTREACHED (break;)
 #endif
 
 #define TAO_YY_RULE_SETUP \
@@ -999,7 +999,7 @@
 TAO_YY_DECL
 	{
 	register tao_yy_state_type tao_yy_current_state;
-	register char *tao_yy_cp, *tao_yy_bp;
+	register char *tao_yy_cp = 0, *tao_yy_bp = 0;
 	register int tao_yy_act;
 
@@ -1568,7 +1568,7 @@
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		  idl_store_pragma(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+   break;
 case 86:
 case 87:

@@ -1575,7 +1575,7 @@
{/* ignore file */
  		  idl_global->set_lineno(idl_global->lineno() + 1);
		}
-	TAO_YY_BREAK
+       break;
case 88:
case 89:
 TAO_YY_RULE_SETUP
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 90:
 case 91:

@@ -1588,7 +1588,7 @@
 TAO_YY_RULE_SETUP
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 92:
 case 93:

@@ -1595,7 +1595,7 @@
 TAO_YY_RULE_SETUP
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 94:
 case 95:

@@ -1602,7 +1602,7 @@
 TAO_YY_RULE_SETUP
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 	        }
-	TAO_YY_BREAK
+	break;
 case 96:
 case 97:

@@ -1611,14 +1611,14 @@
 		  /* ignore cpp ident */
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 98:
 TAO_YY_RULE_SETUP

@@ -1617,7 +1617,7 @@
 {
 		  /* ignore comments */
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 99:
 TAO_YY_RULE_SETUP

@@ -1635,25 +1635,25 @@
 		    }
 	          }
 	        }
-	TAO_YY_BREAK
+	break;
 case 100:
 TAO_YY_RULE_SETUP
 ;
-	TAO_YY_BREAK
+	break;
 case 101:
 TAO_YY_RULE_SETUP
 {
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 102:
 TAO_YY_RULE_SETUP

@@ -1651,7 +1651,7 @@
 return ace_tao_yytext [0];
-	TAO_YY_BREAK
+	break;
 case 103:
 TAO_YY_RULE_SETUP
 TAO_YY_ECHO;
-	TAO_YY_BREAK
+	break;
 case TAO_YY_STATE_EOF(INITIAL):
 	tao_yyterminate();
 
