--- Policy.pidl	Sat Jun 24 04:57:24 2000
+++ Policy.pidl.mod	Sat Jun 24 04:40:35 2000
@@ -78,57 +78,53 @@
 module CORBA
 {
   typedef short PolicyErrorCode;
-  const PolicyErrorCode BAD_POLICY               = 0;
-  const PolicyErrorCode UNSUPPORTED_POLICY       = 1;
-  const PolicyErrorCode BAD_POLICY_TYPE          = 2;
-  const PolicyErrorCode BAD_POLICY_VALUE         = 3;
-  const PolicyErrorCode UNSUPPORTED_POLICY_VALUE = 4;
 
-  exception PolicyError
-  {
-    PolicyErrorCode reason;
-  };
-
-  exception InvalidPolicies
-  {
-    sequence<unsigned short> indices;
-  };
-
   typedef unsigned long PolicyType;
-  interface Policy
-  {
-    readonly attribute PolicyType policy_type;
-    Policy copy ();
-    void destroy ();
-  };
-
-  typedef sequence<Policy> PolicyList;
 
-  typedef sequence<PolicyType> PolicyTypeSeq;
-
   enum SetOverrideType
   {
     SET_OVERRIDE,
     ADD_OVERRIDE
   };
 
-  local interface PolicyManager
-  {
-    PolicyList get_policy_overrides (in PolicyTypeSeq ts);
-    void set_policy_overrides (
-             in PolicyList policies,
-             in SetOverrideType set_add)
-        raises (InvalidPolicies);
-  };
-
-  local interface Current
-  {
-  };
+};
 
-  local interface PolicyCurrent : PolicyManager, Current
-  {
-  };
+exception CORBA_PolicyError
+{
+  CORBA::PolicyErrorCode reason;
+};
+
+exception CORBA_InvalidPolicies
+{
+  sequence<unsigned short> indices;
+};
+
+interface CORBA_Policy
+{
+  readonly attribute CORBA::PolicyType policy_type;
+  CORBA_Policy copy ();
+  void destroy ();
+};
+
+typedef sequence<CORBA_Policy> CORBA_PolicyList;
+
+typedef sequence<CORBA::PolicyType> CORBA_PolicyTypeSeq;
+
+local interface CORBA_PolicyManager
+{
+  CORBA_PolicyList get_policy_overrides (in CORBA_PolicyTypeSeq ts);
+  void set_policy_overrides (
+                             in CORBA_PolicyList policies,
+                             in CORBA::SetOverrideType set_add)
+    raises (CORBA_InvalidPolicies);
+};
+
+local interface CORBA_Current
+{
+};
 
+local interface CORBA_PolicyCurrent : CORBA_PolicyManager, CORBA_Current
+{
 };
 
 #pragma prefix ""
