--- orig/ServantRetentionPolicyC.h	2005-04-11 16:44:02.694739200 +0200
+++ ServantRetentionPolicyC.h	2005-04-11 16:50:59.964744000 +0200
@@ -78,6 +78,7 @@
     NON_RETAIN
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef ServantRetentionPolicyValue &ServantRetentionPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -207,6 +208,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 } // module PortableServer
 
 // TAO_IDL - Generated from
@@ -216,6 +219,8 @@
 namespace TAO
 {
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 #if !defined (_PORTABLESERVER_SERVANTRETENTIONPOLICY__TRAITS_CH_)
 #define _PORTABLESERVER_SERVANTRETENTIONPOLICY__TRAITS_CH_
   
@@ -236,11 +241,15 @@
   };
 
 #endif /* end #if !defined */
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
 }
 
 // TAO_IDL - Generated from
 // be\be_visitor_enum/any_op_ch.cpp:51
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ServantRetentionPolicyValue);
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ServantRetentionPolicyValue &);
 
@@ -257,6 +266,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, PortableServer::ServantRetentionPolicyValue);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::ServantRetentionPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:955
 
--- orig/ServantRetentionPolicyC.cpp	2005-04-11 16:44:02.694739200 +0200
+++ ServantRetentionPolicyC.cpp	2005-04-11 16:50:33.026008000 +0200
@@ -40,6 +40,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -211,3 +213,5 @@
   
   return _tao_success;
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantRetentionPolicyA.cpp	2005-04-11 16:44:02.694739200 +0200
+++ ServantRetentionPolicyA.cpp	2005-04-11 16:50:28.539556800 +0200
@@ -36,6 +36,8 @@
 #include "tao/Any_Impl_T.h"
 #include "tao/Any_Basic_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/enum_typecode.cpp:34
 
@@ -177,3 +179,5 @@
         _tao_elem
       );
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
