--- InterceptorC.cpp.orig	Sat Apr 29 23:33:47 2000
+++ InterceptorC.cpp	Sat Apr 29 23:37:56 2000
@@ -9,6 +9,8 @@
 
 #include "InterceptorC.h"
 
+#if (TAO_HAS_INTERCEPTORS == 1)
+
 #if !defined (__ACE_INLINE__)
 #include "InterceptorC.i"
 #endif /* !defined INLINE */
@@ -403,3 +405,4 @@
   return "IDL:TAO/PortableInterceptor/ClientRequestInterceptor:1.0";
 }
 
+#endif /* TAO_HAS_INTERCEPTORS == 1 */
