--- orig/MessagingC.i	Sun Apr 30 18:03:08 2000
+++ MessagingC.i	Sun Apr 30 18:06:01 2000
@@ -7,6 +7,8 @@
 // Information about TAO is available at:
 //                 http://www.cs.wustl.edu/~schmidt/TAO.html
 
+#if (TAO_HAS_REBIND_POLICY == 1)
+
 ACE_INLINE
 Messaging::RebindPolicy::RebindPolicy (void) // default constructor
 {}
@@ -69,14 +71,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RebindPolicy_var::operator const ::Messaging::RebindPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RebindPolicy_var::operator ::Messaging::RebindPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RebindPolicy_var::operator ::Messaging::RebindPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -168,7 +170,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RebindPolicy_out::operator ::Messaging::RebindPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -189,6 +191,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REBIND_POLICY == 1 */
+
+#if (TAO_HAS_SYNC_SCOPE_POLICY == 1)
+
 ACE_INLINE
 Messaging::SyncScopePolicy::SyncScopePolicy (void) // default constructor
 {}
@@ -251,14 +257,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::SyncScopePolicy_var::operator const ::Messaging::SyncScopePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::SyncScopePolicy_var::operator ::Messaging::SyncScopePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::SyncScopePolicy_var::operator ::Messaging::SyncScopePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -350,7 +356,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::SyncScopePolicy_out::operator ::Messaging::SyncScopePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -371,6 +377,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_SYNC_SCOPE_POLICY == 1 */
+
+#if (TAO_HAS_PRIORITY_POLICIES == 1)
+
 // *************************************************************
 // Inline operations for class Messaging::PriorityRange_var
 // *************************************************************
@@ -457,13 +467,13 @@
 }
 
 ACE_INLINE
-Messaging::PriorityRange_var::operator ::Messaging::PriorityRange &() // cast 
+Messaging::PriorityRange_var::operator ::Messaging::PriorityRange &() // cast
 {
   return *this->ptr_;
 }
 
 ACE_INLINE
-Messaging::PriorityRange_var::operator ::Messaging::PriorityRange &() const // cast 
+Messaging::PriorityRange_var::operator ::Messaging::PriorityRange &() const // cast
 {
   return *this->ptr_;
 }
@@ -480,7 +490,7 @@
   return *this->ptr_;
 }
 
-// mapping for fixed size 
+// mapping for fixed size
 ACE_INLINE ::Messaging::PriorityRange &
 Messaging::PriorityRange_var::out (void)
 {
@@ -561,14 +571,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestPriorityPolicy_var::operator const ::Messaging::RequestPriorityPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RequestPriorityPolicy_var::operator ::Messaging::RequestPriorityPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RequestPriorityPolicy_var::operator ::Messaging::RequestPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -660,7 +670,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestPriorityPolicy_out::operator ::Messaging::RequestPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -743,14 +753,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyPriorityPolicy_var::operator const ::Messaging::ReplyPriorityPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ReplyPriorityPolicy_var::operator ::Messaging::ReplyPriorityPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::ReplyPriorityPolicy_var::operator ::Messaging::ReplyPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -842,7 +852,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyPriorityPolicy_out::operator ::Messaging::ReplyPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -863,6 +873,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_PRIORITY_POLICIES == 1 */
+
+#if (TAO_HAS_REQUEST_START_TIME_POLICY == 1)
+
 ACE_INLINE
 Messaging::RequestStartTimePolicy::RequestStartTimePolicy (void) // default constructor
 {}
@@ -925,14 +939,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestStartTimePolicy_var::operator const ::Messaging::RequestStartTimePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RequestStartTimePolicy_var::operator ::Messaging::RequestStartTimePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RequestStartTimePolicy_var::operator ::Messaging::RequestStartTimePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1024,7 +1038,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestStartTimePolicy_out::operator ::Messaging::RequestStartTimePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1045,6 +1059,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REQUEST_START_TIME_POLICY == 1 */
+
+#if (TAO_HAS_REQUEST_END_TIME_POLICY == 1)
+
 ACE_INLINE
 Messaging::RequestEndTimePolicy::RequestEndTimePolicy (void) // default constructor
 {}
@@ -1107,14 +1125,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestEndTimePolicy_var::operator const ::Messaging::RequestEndTimePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RequestEndTimePolicy_var::operator ::Messaging::RequestEndTimePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RequestEndTimePolicy_var::operator ::Messaging::RequestEndTimePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1206,7 +1224,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestEndTimePolicy_out::operator ::Messaging::RequestEndTimePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1227,6 +1245,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REQUEST_END_TIME_POLICY == 1 */
+
+#if (TAO_HAS_REPLY_START_TIME_POLICY == 1)
+
 ACE_INLINE
 Messaging::ReplyStartTimePolicy::ReplyStartTimePolicy (void) // default constructor
 {}
@@ -1289,14 +1311,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyStartTimePolicy_var::operator const ::Messaging::ReplyStartTimePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ReplyStartTimePolicy_var::operator ::Messaging::ReplyStartTimePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::ReplyStartTimePolicy_var::operator ::Messaging::ReplyStartTimePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1388,7 +1410,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyStartTimePolicy_out::operator ::Messaging::ReplyStartTimePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1409,6 +1431,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REPLY_START_TIME_POLICY == 1 */
+
+#if (TAO_HAS_REPLY_END_TIME_POLICY == 1)
+
 ACE_INLINE
 Messaging::ReplyEndTimePolicy::ReplyEndTimePolicy (void) // default constructor
 {}
@@ -1471,14 +1497,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyEndTimePolicy_var::operator const ::Messaging::ReplyEndTimePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ReplyEndTimePolicy_var::operator ::Messaging::ReplyEndTimePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::ReplyEndTimePolicy_var::operator ::Messaging::ReplyEndTimePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1570,7 +1596,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyEndTimePolicy_out::operator ::Messaging::ReplyEndTimePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1591,6 +1617,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REPLY_END_TIME_POLICY == 1 */
+
+#if (TAO_HAS_RELATIVE_REQUEST_TIMEOUT_POLICY == 1)
+
 ACE_INLINE
 Messaging::RelativeRequestTimeoutPolicy::RelativeRequestTimeoutPolicy (void) // default constructor
 {}
@@ -1653,14 +1683,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RelativeRequestTimeoutPolicy_var::operator const ::Messaging::RelativeRequestTimeoutPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RelativeRequestTimeoutPolicy_var::operator ::Messaging::RelativeRequestTimeoutPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RelativeRequestTimeoutPolicy_var::operator ::Messaging::RelativeRequestTimeoutPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1752,7 +1782,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RelativeRequestTimeoutPolicy_out::operator ::Messaging::RelativeRequestTimeoutPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1773,6 +1803,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_RELATIVE_REQUEST_TIMEOUT_POLICY == 1 */
+
+#if (TAO_HAS_RELATIVE_ROUNDTRIP_TIMEOUT_POLICY == 1)
+
 ACE_INLINE
 Messaging::RelativeRoundtripTimeoutPolicy::RelativeRoundtripTimeoutPolicy (void) // default constructor
 {}
@@ -1835,14 +1869,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RelativeRoundtripTimeoutPolicy_var::operator const ::Messaging::RelativeRoundtripTimeoutPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RelativeRoundtripTimeoutPolicy_var::operator ::Messaging::RelativeRoundtripTimeoutPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RelativeRoundtripTimeoutPolicy_var::operator ::Messaging::RelativeRoundtripTimeoutPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1934,7 +1968,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RelativeRoundtripTimeoutPolicy_out::operator ::Messaging::RelativeRoundtripTimeoutPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1959,6 +1993,10 @@
 // Inline operations for class Messaging::RoutingTypeRange_var
 // *************************************************************
 
+#endif /* TAO_HAS_RELATIVE_ROUNDTRIP_TIMEOUT_POLICY == 1 */
+
+#if (TAO_HAS_ROUTING_POLICY == 1)
+
 ACE_INLINE
 Messaging::RoutingTypeRange_var::RoutingTypeRange_var (void) // default constructor
   : ptr_ (0)
@@ -2041,13 +2079,13 @@
 }
 
 ACE_INLINE
-Messaging::RoutingTypeRange_var::operator ::Messaging::RoutingTypeRange &() // cast 
+Messaging::RoutingTypeRange_var::operator ::Messaging::RoutingTypeRange &() // cast
 {
   return *this->ptr_;
 }
 
 ACE_INLINE
-Messaging::RoutingTypeRange_var::operator ::Messaging::RoutingTypeRange &() const // cast 
+Messaging::RoutingTypeRange_var::operator ::Messaging::RoutingTypeRange &() const // cast
 {
   return *this->ptr_;
 }
@@ -2064,7 +2102,7 @@
   return *this->ptr_;
 }
 
-// mapping for fixed size 
+// mapping for fixed size
 ACE_INLINE ::Messaging::RoutingTypeRange &
 Messaging::RoutingTypeRange_var::out (void)
 {
@@ -2145,14 +2183,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RoutingPolicy_var::operator const ::Messaging::RoutingPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RoutingPolicy_var::operator ::Messaging::RoutingPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RoutingPolicy_var::operator ::Messaging::RoutingPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -2244,7 +2282,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RoutingPolicy_out::operator ::Messaging::RoutingPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -2265,6 +2303,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_ROUTING_POLICY == 1 */
+
+#if (TAO_HAS_MAX_HOPS_POLICY == 1)
+
 ACE_INLINE
 Messaging::MaxHopsPolicy::MaxHopsPolicy (void) // default constructor
 {}
@@ -2327,14 +2369,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::MaxHopsPolicy_var::operator const ::Messaging::MaxHopsPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::MaxHopsPolicy_var::operator ::Messaging::MaxHopsPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::MaxHopsPolicy_var::operator ::Messaging::MaxHopsPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -2426,7 +2468,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::MaxHopsPolicy_out::operator ::Messaging::MaxHopsPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -2447,6 +2489,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_MAX_HOPS_POLICY == 1 */
+
+#if (TAO_HAS_QUEUE_ORDER_POLICY == 1)
+
 ACE_INLINE
 Messaging::QueueOrderPolicy::QueueOrderPolicy (void) // default constructor
 {}
@@ -2509,14 +2555,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::QueueOrderPolicy_var::operator const ::Messaging::QueueOrderPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::QueueOrderPolicy_var::operator ::Messaging::QueueOrderPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::QueueOrderPolicy_var::operator ::Messaging::QueueOrderPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -2608,7 +2654,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::QueueOrderPolicy_out::operator ::Messaging::QueueOrderPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -2629,6 +2675,8 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_QUEUE_ORDER_POLICY == 1 */
+
 // *************************************************************
 // Inline operations for class Messaging::PolicyValue_var
 // *************************************************************
@@ -2696,20 +2744,20 @@
 }
 
 ACE_INLINE
-Messaging::PolicyValue_var::operator ::Messaging::PolicyValue &() // cast 
+Messaging::PolicyValue_var::operator ::Messaging::PolicyValue &() // cast
 {
   return *this->ptr_;
 }
 
 ACE_INLINE
-Messaging::PolicyValue_var::operator ::Messaging::PolicyValue &() const // cast 
+Messaging::PolicyValue_var::operator ::Messaging::PolicyValue &() const // cast
 {
   return *this->ptr_;
 }
 
 // variable-size types only
 ACE_INLINE
-Messaging::PolicyValue_var::operator ::Messaging::PolicyValue *&() // cast 
+Messaging::PolicyValue_var::operator ::Messaging::PolicyValue *&() // cast
 {
   return this->ptr_;
 }
@@ -2726,7 +2774,7 @@
   return *this->ptr_;
 }
 
-// mapping for variable size 
+// mapping for variable size
 ACE_INLINE ::Messaging::PolicyValue *&
 Messaging::PolicyValue_var::out (void)
 {
@@ -2787,7 +2835,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValue_out::operator ::Messaging::PolicyValue *&() // cast
 {
   return this->ptr_;
@@ -2888,20 +2936,20 @@
   return this->ptr_;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValue::_tao_seq_Octet_var::operator const ::Messaging::PolicyValue::_tao_seq_Octet &() const // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::PolicyValue::_tao_seq_Octet_var::operator ::Messaging::PolicyValue::_tao_seq_Octet &() // cast 
+ACE_INLINE
+Messaging::PolicyValue::_tao_seq_Octet_var::operator ::Messaging::PolicyValue::_tao_seq_Octet &() // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::PolicyValue::_tao_seq_Octet_var::operator ::Messaging::PolicyValue::_tao_seq_Octet &() const // cast 
+ACE_INLINE
+Messaging::PolicyValue::_tao_seq_Octet_var::operator ::Messaging::PolicyValue::_tao_seq_Octet &() const // cast
 {
   return *this->ptr_;
 }
@@ -2924,7 +2972,7 @@
   return *this->ptr_;
 }
 
-// mapping for variable size 
+// mapping for variable size
 ACE_INLINE ::Messaging::PolicyValue::_tao_seq_Octet *&
 Messaging::PolicyValue::_tao_seq_Octet_var::out (void)
 {
@@ -2985,7 +3033,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValue::_tao_seq_Octet_out::operator ::Messaging::PolicyValue::_tao_seq_Octet *&() // cast
 {
   return this->ptr_;
@@ -3014,7 +3062,7 @@
 
 
 #if !defined (TAO_USE_SEQUENCE_TEMPLATES)
-  
+
 #if !defined (__TAO_UNBOUNDED_SEQUENCE_MESSAGING_POLICYVALUESEQ_CI_)
 #define __TAO_UNBOUNDED_SEQUENCE_MESSAGING_POLICYVALUESEQ_CI_
 
@@ -3027,24 +3075,24 @@
     ACE_NEW_RETURN (retval, Messaging::PolicyValue[size], 0);
     return retval;
   }
-  
+
   ACE_INLINE void Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::freebuf (Messaging::PolicyValue *buffer)
   // Free the sequence.
   {
     delete [] buffer;
   }
-  
+
   ACE_INLINE
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq (void) // Default constructor.
   {
   }
-  
+
   ACE_INLINE
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq (CORBA::ULong maximum) // Constructor using a maximum length value.
     : TAO_Unbounded_Base_Sequence (maximum, _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::allocbuf (maximum))
   {
   }
-  
+
   ACE_INLINE
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq (CORBA::ULong maximum,
     CORBA::ULong length,
@@ -3053,7 +3101,7 @@
   : TAO_Unbounded_Base_Sequence (maximum, length, data, release)
   {
   }
-  
+
   ACE_INLINE
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq (const _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq &rhs)
   // Copy constructor.
@@ -3063,10 +3111,10 @@
     {
       Messaging::PolicyValue *tmp1 = _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::allocbuf (this->maximum_);
       Messaging::PolicyValue * const tmp2 = ACE_reinterpret_cast (Messaging::PolicyValue * ACE_CAST_CONST, rhs.buffer_);
-      
+
       for (CORBA::ULong i = 0; i < this->length_; ++i)
         tmp1[i] = tmp2[i];
-      
+
       this->buffer_ = tmp1;
     }
     else
@@ -3074,14 +3122,14 @@
       this->buffer_ = 0;
     }
   }
-  
+
   ACE_INLINE Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq &
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::operator= (const _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq &rhs)
   // Assignment operator.
   {
     if (this == &rhs)
       return *this;
-    
+
     if (this->release_)
     {
       if (this->maximum_ < rhs.maximum_)
@@ -3094,18 +3142,18 @@
     }
     else
       this->buffer_ = _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::allocbuf (rhs.maximum_);
-    
+
     TAO_Unbounded_Base_Sequence::operator= (rhs);
-    
+
     Messaging::PolicyValue *tmp1 = ACE_reinterpret_cast (Messaging::PolicyValue *, this->buffer_);
     Messaging::PolicyValue * const tmp2 = ACE_reinterpret_cast (Messaging::PolicyValue * ACE_CAST_CONST, rhs.buffer_);
-    
+
     for (CORBA::ULong i = 0; i < this->length_; ++i)
       tmp1[i] = tmp2[i];
-    
+
     return *this;
   }
-  
+
   // = Accessors.
   ACE_INLINE Messaging::PolicyValue &
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::operator[] (CORBA::ULong i)
@@ -3115,7 +3163,7 @@
     Messaging::PolicyValue* tmp = ACE_reinterpret_cast(Messaging::PolicyValue*,this->buffer_);
     return tmp[i];
   }
-  
+
   ACE_INLINE const Messaging::PolicyValue &
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::operator[] (CORBA::ULong i) const
   // operator []
@@ -3124,9 +3172,9 @@
     Messaging::PolicyValue * const tmp = ACE_reinterpret_cast (Messaging::PolicyValue* ACE_CAST_CONST, this->buffer_);
     return tmp[i];
   }
-  
+
   // Implement the TAO_Base_Sequence methods (see Sequence.h)
-  
+
   ACE_INLINE Messaging::PolicyValue *
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::get_buffer (CORBA::Boolean orphan)
   {
@@ -3159,13 +3207,13 @@
     }
     return result;
   }
-  
+
   ACE_INLINE const Messaging::PolicyValue *
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::get_buffer (void) const
   {
     return ACE_reinterpret_cast(const Messaging::PolicyValue * ACE_CAST_CONST, this->buffer_);
   }
-  
+
   ACE_INLINE void
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::replace (CORBA::ULong max,
   CORBA::ULong length,
@@ -3182,11 +3230,11 @@
     this->buffer_ = data;
     this->release_ = release;
   }
-  
+
 #endif /* end #if !defined */
 
 
-#endif /* !TAO_USE_SEQUENCE_TEMPLATES */ 
+#endif /* !TAO_USE_SEQUENCE_TEMPLATES */
 
 #if !defined (_MESSAGING_POLICYVALUESEQ_CI_)
 #define _MESSAGING_POLICYVALUESEQ_CI_
@@ -3251,27 +3299,27 @@
   return this->ptr_;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValueSeq_var::operator const ::Messaging::PolicyValueSeq &() const // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq &() // cast 
+ACE_INLINE
+Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq &() // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq &() const // cast 
+ACE_INLINE
+Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq &() const // cast
 {
   return *this->ptr_;
 }
 
 // variable-size types only
 ACE_INLINE
-Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq *&() // cast 
+Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq *&() // cast
 {
   return this->ptr_;
 }
@@ -3294,7 +3342,7 @@
   return *this->ptr_;
 }
 
-// mapping for variable size 
+// mapping for variable size
 ACE_INLINE ::Messaging::PolicyValueSeq *&
 Messaging::PolicyValueSeq_var::out (void)
 {
@@ -3355,7 +3403,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValueSeq_out::operator ::Messaging::PolicyValueSeq *&() // cast
 {
   return this->ptr_;
@@ -3382,6 +3430,8 @@
 
 #endif /* end #if !defined */
 
+#if (TAO_HAS_AMI_CALLBACK == 1) || (TAO_HAS_AMI_POLLER == 1)
+
 ACE_INLINE
 Messaging::ExceptionHolder::ExceptionHolder () // default constructor
 {}
@@ -3390,7 +3440,7 @@
 Messaging::ExceptionHolder::~ExceptionHolder () // destructor
 {}
 
-ACE_INLINE const char* 
+ACE_INLINE const char*
 Messaging::ExceptionHolder::_tao_obv_static_repository_id ()
 {
   return "IDL:omg.org/Messaging/ExceptionHolder:1.0";
@@ -3419,7 +3469,7 @@
   : ptr_ (ACE_const_cast(Messaging::ExceptionHolder*, p))
 {}
 
-ACE_INLINE Messaging::ExceptionHolder* 
+ACE_INLINE Messaging::ExceptionHolder*
 Messaging::ExceptionHolder_var::ptr (void) const
 {
   return this->ptr_;
@@ -3459,19 +3509,19 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ExceptionHolder_var::operator const Messaging::ExceptionHolder* () const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ExceptionHolder_var::operator Messaging::ExceptionHolder* () // cast 
+ACE_INLINE
+Messaging::ExceptionHolder_var::operator Messaging::ExceptionHolder* () // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE Messaging::ExceptionHolder* 
+ACE_INLINE Messaging::ExceptionHolder*
 Messaging::ExceptionHolder_var::operator-> (void) const
 {
   return this->ptr_;
@@ -3497,7 +3547,7 @@
   return this->ptr_;
 }
 
-ACE_INLINE Messaging::ExceptionHolder* 
+ACE_INLINE Messaging::ExceptionHolder*
 Messaging::ExceptionHolder_var::_retn (void)
 {
   // yield ownership of managed obj reference
@@ -3560,7 +3610,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ExceptionHolder_out::operator Messaging::ExceptionHolder* &() // cast
 {
   return this->ptr_;
@@ -3572,7 +3622,7 @@
   return this->ptr_;
 }
 
-ACE_INLINE Messaging::ExceptionHolder* 
+ACE_INLINE Messaging::ExceptionHolder*
 Messaging::ExceptionHolder_out::operator-> (void)
 {
   return this->ptr_;
@@ -3648,14 +3698,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyHandler_var::operator const ::Messaging::ReplyHandler_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ReplyHandler_var::operator ::Messaging::ReplyHandler_ptr &() // cast 
+ACE_INLINE
+Messaging::ReplyHandler_var::operator ::Messaging::ReplyHandler_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -3747,7 +3797,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyHandler_out::operator ::Messaging::ReplyHandler_ptr &() // cast
 {
   return this->ptr_;
@@ -3768,6 +3818,8 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_AMI_CALLBACK == 1 || TAO_HAS_AMI_POLLER == 1 */
+
 ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const Messaging::PriorityRange &_tao_aggregate)
 {
   if (
@@ -3777,7 +3829,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &strm, Messaging::PriorityRange &_tao_aggregate)
@@ -3789,7 +3841,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const Messaging::RoutingTypeRange &_tao_aggregate)
@@ -3801,7 +3853,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &strm, Messaging::RoutingTypeRange &_tao_aggregate)
@@ -3813,7 +3865,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 
@@ -3840,7 +3892,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &strm, Messaging::PolicyValue &_tao_aggregate)
@@ -3852,7 +3904,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 
@@ -3870,6 +3922,8 @@
 
 #endif /* _TAO_CDR_OP_Messaging_PolicyValueSeq_I_ */
 
+#if (TAO_HAS_AMI_CALLBACK == 1) || (TAO_HAS_AMI_POLLER == 1)
+
 ACE_INLINE CORBA::Boolean
 operator<< (TAO_OutputCDR &strm, const Messaging::ExceptionHolder *_tao_valuetype)
 {
@@ -3895,7 +3949,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean
@@ -3909,7 +3963,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean
@@ -3961,3 +4015,4 @@
   return 0;
 }
 
+#endif /* TAO_HAS_AMI_CALLBACK == 1 || TAO_HAS_AMI_POLLER == 1 */
