--- PortableServerC.cpp	Tue Jul 11 01:25:50 2000
+++ PortableServerC.cpp.mod	Tue Jul 11 01:16:29 2000
@@ -8,6 +8,8 @@
 //                 http://www.cs.wustl.edu/~schmidt/TAO.html
 
 #include "PortableServerC.h"
+#include "POA.h"
+#include "tao/ORB.h"
 
 #if !defined (__ACE_INLINE__)
 #include "PortableServerC.i"
@@ -209,6 +211,9 @@
 TAO_NAMESPACE_BEGIN (PortableServer)
 TAO_NAMESPACE_DEFINE (CORBA::TypeCode_ptr, _tc_ObjectId, &_tc_TAO_tc_PortableServer_ObjectId)
 TAO_NAMESPACE_END
+
+#if (TAO_HAS_MINIMUM_CORBA == 0)
+
 // default constructor
 PortableServer::ForwardRequest::ForwardRequest (void)
   : CORBA_UserException (::PortableServer::_tc_ForwardRequest)
@@ -340,6 +340,8 @@
   return ::PortableServer::_tc_ForwardRequest;
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 TAO_NAMESPACE_TYPE (const CORBA::ULong)
 TAO_NAMESPACE_BEGIN (PortableServer)
 TAO_NAMESPACE_DEFINE (const CORBA::ULong, THREAD_POLICY_ID, 16U)
@@ -342,6 +350,9 @@
 TAO_NAMESPACE_BEGIN (PortableServer)
 TAO_NAMESPACE_DEFINE (const CORBA::ULong, REQUEST_PROCESSING_POLICY_ID, 22U)
 TAO_NAMESPACE_END
+
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ThreadPolicyValue[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -436,6 +447,8 @@
   return "IDL:PortableServer/ThreadPolicy:1.0";
 }
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 static const CORBA::Long _oc_PortableServer_LifespanPolicyValue[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -718,6 +731,8 @@
   return "IDL:PortableServer/IdAssignmentPolicy:1.0";
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ImplicitActivationPolicyValue[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -1001,6 +1016,7 @@
   return "IDL:PortableServer/RequestProcessingPolicy:1.0";
 }
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
 
 // default constructor
 PortableServer::POAManager::POAManager (void)
@@ -1048,6 +1064,8 @@
   return obj;
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // default constructor
 PortableServer::POAManager::AdapterInactive::AdapterInactive (void)
 {
@@ -1430,6 +1448,7 @@
   return "IDL:PortableServer/ServantLocator:1.0";
 }
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
 
 // default constructor
 PortableServer::POA::POA (void)
@@ -1681,6 +1700,8 @@
     this->index = _tao_index;
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // default constructor
 PortableServer::POA::NoServant::NoServant (void)
 {
@@ -1746,6 +1767,8 @@
   return retval;
 }
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 // default constructor
 PortableServer::POA::ObjectAlreadyActive::ObjectAlreadyActive (void)
 {
@@ -2302,6 +2325,30 @@
   return "IDL:PortableServer/Current:1.0";
 }
 
+char *
+PortableServer::ObjectId_to_string (const PortableServer::ObjectId &id)
+{
+  return TAO_POA::ObjectId_to_string (id);
+}
+
+CORBA::WChar *
+PortableServer::ObjectId_to_wstring (const PortableServer::ObjectId &id)
+{
+  return TAO_POA::ObjectId_to_wstring (id);
+}
+
+PortableServer::ObjectId *
+PortableServer::string_to_ObjectId (const char *id)
+{
+  return TAO_POA::string_to_ObjectId (id);
+}
+
+PortableServer::ObjectId *
+PortableServer::wstring_to_ObjectId (const CORBA::WChar *id)
+{
+  return TAO_POA::wstring_to_ObjectId (id);
+}
+
 void operator<<= (
     CORBA::Any &_tao_any,
     const PortableServer::ObjectId &_tao_elem
@@ -2391,6 +2438,8 @@
   return 0;
 }
 
+#if (TAO_HAS_MINIMUM_CORBA == 0)
+
 void operator<<= (CORBA::Any &_tao_any, const PortableServer::ForwardRequest &_tao_elem) // copying
 {
   TAO_OutputCDR stream;
@@ -2479,6 +2528,10 @@
   return 0;
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 void operator<<= (CORBA::Any &_tao_any, PortableServer::ThreadPolicyValue _tao_elem)
 {
   TAO_OutputCDR stream;
@@ -2524,6 +2577,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::ThreadPolicy,PortableServer::ThreadPolicy_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 void operator<<= (CORBA::Any &_tao_any, PortableServer::LifespanPolicyValue _tao_elem)
 {
   TAO_OutputCDR stream;
@@ -2659,6 +2714,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::IdAssignmentPolicy,PortableServer::IdAssignmentPolicy_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 void operator<<= (CORBA::Any &_tao_any, PortableServer::ImplicitActivationPolicyValue _tao_elem)
 {
   TAO_OutputCDR stream;
@@ -2794,6 +2851,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::RequestProcessingPolicy,PortableServer::RequestProcessingPolicy_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 #if defined (ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION)  || \
     defined (ACE_HAS_GNU_REPO)
     template class TAO_Object_Manager<PortableServer::POAManager,PortableServer::POAManager_var>;
@@ -2801,6 +2860,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::POAManager,PortableServer::POAManager_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 #if defined (ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION)  || \
     defined (ACE_HAS_GNU_REPO)
     template class TAO_Object_Manager<PortableServer::AdapterActivator,PortableServer::AdapterActivator_var>;
@@ -2829,6 +2890,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::ServantLocator,PortableServer::ServantLocator_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 #if defined (ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION)  || \
     defined (ACE_HAS_GNU_REPO)
     template class TAO_Object_Manager<PortableServer::POA,PortableServer::POA_var>;
