--- orig/TAOC.cpp	Fri Apr 28 18:04:26 2000
+++ tmp/TAOC.cpp	Fri Apr 28 18:15:43 2000
@@ -9,10 +9,15 @@
 
 #include "TAOC.h"
 
+#include "tao/TAOS.h"
+#include "tao/Typecode.h"
+
 #if !defined (__ACE_INLINE__)
 #include "TAOC.i"
 #endif /* !defined INLINE */
 
+#if (TAO_HAS_CLIENT_PRIORITY_POLICY == 1)
+
 static const CORBA::Long _oc_TAO_PrioritySelectionMode[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -135,7 +140,7 @@
   else if (type == ACE_reinterpret_cast (ptr_arith_t, &CORBA::Object::_narrow))
     retv = ACE_reinterpret_cast (void *,
       ACE_static_cast (CORBA::Object_ptr, this));
-    
+
   if (retv)
     this->_add_ref ();
   return retv;
@@ -146,6 +151,10 @@
   return "IDL:TAO/ClientPriorityPolicy:1.0";
 }
 
+#endif /* TAO_HAS_CLIENT_PRIORITY_POLICY == 1 */
+
+#if (TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1)
+
 static const CORBA::Long _oc_TAO_BufferingConstraintMode[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -281,7 +290,7 @@
   else if (type == ACE_reinterpret_cast (ptr_arith_t, &CORBA::Object::_narrow))
     retv = ACE_reinterpret_cast (void *,
       ACE_static_cast (CORBA::Object_ptr, this));
-    
+
   if (retv)
     this->_add_ref ();
   return retv;
@@ -454,3 +463,4 @@
   return 0;
 }
 
+#endif /* TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1 */
