--- lex.yy.cpp.orig	2009-06-23 13:18:22.000000000 -0500
+++ lex.yy.cpp	2009-06-23 13:49:49.000000000 -0500
@@ -27,7 +27,7 @@
 
 /* C99 systems have <inttypes.h>. Non-C99 systems may or may not. */
 
-#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L || defined (__HP_aCC)
 
 /* C99 says to define __STDC_LIMIT_MACROS before including stdint.h,
  * if you want the limit (max/min) macros for int types.
@@ -35,6 +35,7 @@
 #ifndef __STDC_LIMIT_MACROS
 #define __STDC_LIMIT_MACROS 1
 #endif
+#include <inttypes.h>
 
 typedef int8_t flex_int8_t;
 typedef uint8_t flex_uint8_t;
@@ -1773,7 +1774,7 @@ TAO_YY_RULE_SETUP
                   idl_global->set_lineno(idl_global->lineno() + 1);
                   idl_store_pragma(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 94:
 /* rule 94 can match eol */
 case 95:
@@ -1782,7 +1783,7 @@ TAO_YY_RULE_SETUP
 {/* ignore file */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+         break;
 case 96:
 /* rule 96 can match eol */
 case 97:
@@ -1791,7 +1792,7 @@ TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 98:
 /* rule 98 can match eol */
 case 99:
@@ -1800,7 +1801,7 @@ TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 100:
 /* rule 100 can match eol */
 case 101:
@@ -1809,7 +1810,7 @@ TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 102:
 /* rule 102 can match eol */
 case 103:
@@ -1818,7 +1819,7 @@ TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 104:
 /* rule 104 can match eol */
 case 105:
@@ -1828,7 +1829,7 @@ TAO_YY_RULE_SETUP
                   /* ignore cpp ident */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+         break;
 case 106:
 /* rule 106 can match eol */
 TAO_YY_RULE_SETUP
@@ -1836,7 +1837,7 @@ TAO_YY_RULE_SETUP
                   /* ignore comments */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+         break;
 case 107:
 TAO_YY_RULE_SETUP
 {
@@ -1854,18 +1855,18 @@ TAO_YY_RULE_SETUP
                     }
                   }
                 }
-        TAO_YY_BREAK
+        break;
 case 108:
 TAO_YY_RULE_SETUP
 ;
-        TAO_YY_BREAK
+         break;
 case 109:
 /* rule 109 can match eol */
 TAO_YY_RULE_SETUP
 {
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+         break;
 case 110:
 TAO_YY_RULE_SETUP
 return ace_tao_yytext [0];
@@ -1873,7 +1874,7 @@ return ace_tao_yytext [0];
 case 111:
 TAO_YY_RULE_SETUP
 TAO_YY_ECHO;
-        TAO_YY_BREAK
+         break;
 case TAO_YY_STATE_EOF(INITIAL):
         tao_yyterminate();
 
