--- orig/RTSchedulerC.h	2005-02-11 15:05:03.696392000 +0100
+++ RTSchedulerC.h	2005-02-11 15:09:32.272585600 +0100
@@ -54,6 +54,10 @@
 #include "tao/PortableServer/PortableServer_includeC.h"
 #include "tao/PortableInterceptorC.h"
 
+#ifndef TAO_RTSCHEDULER_SAFE_INCLUDE
+#error "You should not include RTSchedulerC.h directly, use RTScheduler.h"
+#endif /* !TAO_RTSCHEDULER_SAFE_INCLUDE */
+
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
@@ -95,6 +99,9 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+  // Hand crafted
+  typedef void* VoidData;
+
 } // module CORBA
 
 // TAO_IDL - Generated from
--- orig/RTSchedulerC.cpp	2005-02-11 15:05:03.696392000 +0100
+++ RTSchedulerC.cpp	2005-02-11 15:07:11.239790400 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "RTSchedulerC.h"
+#include "RTScheduler.h"
 #include "tao/CDR.h"
 #include "tao/ORB_Core.h"
 #include "tao/Typecode.h"
