--- lex.yy.cpp.orig	2008-05-31 10:22:13.000000000 -0500
+++ lex.yy.cpp	2008-05-31 10:27:05.000000000 -0500
@@ -35,6 +35,7 @@
 #ifndef __STDC_LIMIT_MACROS
 #define __STDC_LIMIT_MACROS 1
 #endif
+#include <inttypes.h>
 
 typedef int8_t flex_int8_t;
 typedef uint8_t flex_uint8_t;
@@ -1079,7 +1079,7 @@
         if ( TAO_YY_CURRENT_BUFFER_LVALUE->tao_yy_is_interactive ) \
                 { \
                 int c = '*'; \
-                size_t n; \
+                int n;                    \
                 for ( n = 0; n < max_size && \
                              (c = getc( tao_yyin )) != EOF && c != '\n'; ++n ) \
                         buf[n] = (char) c; \
@@ -1698,7 +1698,7 @@
                   idl_global->set_lineno(idl_global->lineno() + 1);
                   idl_store_pragma(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+        break;
 case 87:
 /* rule 87 can match eol */
 case 88:
@@ -1707,7 +1707,7 @@
 {/* ignore file */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+        break;
 case 89:
 /* rule 89 can match eol */
 case 90:
@@ -1716,7 +1716,7 @@
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+        break;
 case 91:
 /* rule 91 can match eol */
 case 92:
@@ -1725,7 +1725,7 @@
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+        break;
 case 93:
 /* rule 93 can match eol */
 case 94:
@@ -1734,7 +1734,7 @@
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+        break;
 case 95:
 /* rule 95 can match eol */
 case 96:
@@ -1743,7 +1743,7 @@
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 97:
 /* rule 97 can match eol */
 case 98:
@@ -1753,7 +1753,7 @@
                   /* ignore cpp ident */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+ break;
 case 99:
 /* rule 99 can match eol */
 TAO_YY_RULE_SETUP
@@ -1761,7 +1761,7 @@
                   /* ignore comments */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+ break;
 case 100:
 TAO_YY_RULE_SETUP
 {
@@ -1783,14 +1783,14 @@
 case 101:
 TAO_YY_RULE_SETUP
 ;
-        TAO_YY_BREAK
+ break;
 case 102:
 /* rule 102 can match eol */
 TAO_YY_RULE_SETUP
 {
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+ break;
 case 103:
 TAO_YY_RULE_SETUP
 return ace_tao_yytext [0];
@@ -1798,7 +1798,7 @@
 case 104:
 TAO_YY_RULE_SETUP
 TAO_YY_ECHO;
-        TAO_YY_BREAK
+        break;
 case TAO_YY_STATE_EOF(INITIAL):
         tao_yyterminate();
 
