--- orig/AdapterActivatorC.h	2005-02-18 11:19:44.306534400 +0100
+++ AdapterActivatorC.h	2005-02-18 11:19:48.512582400 +0100
@@ -73,6 +73,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_root/root_ch.cpp:62
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace TAO
 {
   class Collocation_Proxy_Broker;
@@ -227,6 +229,8 @@
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::AdapterActivator_ptr *); // non-copying
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::AdapterActivator_ptr &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:961
 
--- orig/AdapterActivatorC.cpp	2005-02-18 11:19:44.306534400 +0100
+++ AdapterActivatorC.cpp	2005-02-18 11:19:48.522596800 +0100
@@ -44,6 +44,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:66
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -227,3 +229,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/AdapterActivatorA.cpp	2005-02-18 11:19:44.306534400 +0100
+++ AdapterActivatorA.cpp	2005-02-18 11:19:48.532611200 +0100
@@ -34,6 +34,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:295
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_AdapterActivator[] =
 {
     TAO_ENCAP_BYTE_ORDER, // byte order
@@ -156,3 +158,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
