--- orig/PortableServerC.h	2005-03-25 15:46:01.159468800 +0100
+++ PortableServerC.h	2005-03-25 15:46:16.791947200 +0100
@@ -36,6 +36,10 @@
 
 #include "ace/config-all.h"
 
+#ifndef TAO_PORTABLESERVER_SAFE_INCLUDE
+#error "You should not include PortableServerC.h directly, use PortableServer.h"
+#endif /* !TAO_PORTABLESERVER_SAFE_INCLUDE */
+
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
@@ -483,6 +487,8 @@
     // TAO_IDL - Generated from
     // be\be_visitor_exception/exception_ch.cpp:51
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 #if !defined (_PORTABLESERVER_POA_NOSERVANT_CH_)
 #define _PORTABLESERVER_POA_NOSERVANT_CH_
     
@@ -530,6 +536,8 @@
 
 #endif /* end #if !defined */
     
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
     // TAO_IDL - Generated from
     // be\be_visitor_exception/exception_ch.cpp:51
 
@@ -873,6 +881,7 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ThreadPolicy_ptr create_thread_policy (
         ::PortableServer::ThreadPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -880,6 +889,7 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
@@ -917,6 +927,7 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ImplicitActivationPolicy_ptr create_implicit_activation_policy (
         ::PortableServer::ImplicitActivationPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -924,10 +935,12 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ServantRetentionPolicy_ptr create_servant_retention_policy (
         ::PortableServer::ServantRetentionPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -935,10 +948,12 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::RequestProcessingPolicy_ptr create_request_processing_policy (
         ::PortableServer::RequestProcessingPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -946,6 +961,7 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
@@ -990,6 +1006,8 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
+
     virtual ::PortableServer::AdapterActivator_ptr the_activator (
         ACE_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
@@ -1007,10 +1025,12 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ServantManager_ptr get_servant_manager (
         ACE_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
@@ -1030,10 +1050,13 @@
         CORBA::SystemException,
         ::PortableServer::POA::WrongPolicy
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
+
     virtual ::PortableServer::Servant get_servant (
         ACE_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
@@ -1055,6 +1078,8 @@
         ::PortableServer::POA::WrongPolicy
       )) = 0;
     
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
@@ -1316,11 +1341,15 @@
 // TAO_IDL - Generated from
 // be\be_visitor_exception/any_op_ch.cpp:52
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, const PortableServer::POA::NoServant &); // copying version
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::POA::NoServant*); // noncopying version
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::POA::NoServant *&); // deprecated
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, const PortableServer::POA::NoServant *&);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_visitor_exception/any_op_ch.cpp:52
 
@@ -1388,4 +1417,3 @@
 
 #endif /* ifndef */
 
-
--- orig/PortableServerC.cpp	2005-03-25 15:46:02.180937600 +0100
+++ PortableServerC.cpp	2005-03-25 15:46:16.862048000 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "PortableServerC.h"
+#include "PortableServer.h"
 #include "tao/CDR.h"
 #include "tao/ORB_Core.h"
 #include "tao/Null_RefCount_Policy.h"
@@ -56,7 +56,6 @@
 // Arg traits specializations.
 namespace TAO
 {
-
 #if !defined (_PORTABLESERVER_ADAPTERACTIVATOR__ARG_TRAITS_CS_)
 #define _PORTABLESERVER_ADAPTERACTIVATOR__ARG_TRAITS_CS_
   
@@ -75,7 +74,6 @@
 #endif /* end #if !defined */
 }
 
-
 // TAO_IDL - Generated from 
 // be\be_visitor_sequence/sequence_cs.cpp:65
 
@@ -474,6 +472,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/exception_cs.cpp:63
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 PortableServer::POA::NoServant::NoServant (void)
   : CORBA::UserException (
         "IDL:omg.org/PortableServer/POA/NoServant:2.3",
@@ -566,6 +566,8 @@
   return ::PortableServer::POA::_tc_NoServant;
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/exception_cs.cpp:63
 
--- orig/PortableServerA.cpp	2005-03-25 15:46:02.180937600 +0100
+++ PortableServerA.cpp	2005-03-25 17:28:43.860998400 +0100
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "PortableServerC.h"
+#include "PortableServer.h"
 #include "tao/Null_RefCount_Policy.h"
 #include "tao/TypeCode_Constants.h"
 #include "tao/Alias_TypeCode.h"
@@ -308,6 +308,8 @@
   return false;
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 ACE_TEMPLATE_SPECIALIZATION
 CORBA::Boolean
 TAO::Any_Dual_Impl_T<PortableServer::POAList>::demarshal_value (TAO_InputCDR &)
@@ -437,6 +439,8 @@
       );
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/any_op_cs.cpp:50
 
@@ -1174,11 +1178,15 @@
         PortableServer::POA::InvalidPolicy
       >;
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
   template class
     TAO::Any_Dual_Impl_T<
         PortableServer::POA::NoServant
       >;
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
   template class
     TAO::Any_Dual_Impl_T<
         PortableServer::POA::ObjectAlreadyActive
@@ -1236,11 +1244,15 @@
         PortableServer::POA::InvalidPolicy \
       >
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 # pragma instantiate \
     TAO::Any_Dual_Impl_T< \
         PortableServer::POA::NoServant \
       >
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 # pragma instantiate \
     TAO::Any_Dual_Impl_T< \
         PortableServer::POA::ObjectAlreadyActive \
