--- orig/PortableServerC.h	2005-02-18 11:19:45.718564800 +0100
+++ PortableServerC.h	2005-02-18 11:19:48.712870400 +0100
@@ -36,6 +36,10 @@
 
 #include "ace/config-all.h"
 
+#ifndef TAO_PORTABLESERVER_SAFE_INCLUDE
+#error "You should not include PortableServerC.h directly, use PortableServer.h"
+#endif /* !TAO_PORTABLESERVER_SAFE_INCLUDE */
+
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
@@ -481,6 +485,8 @@
     // TAO_IDL - Generated from
     // be\be_visitor_exception/exception_ch.cpp:51
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 #if !defined (_PORTABLESERVER_POA_NOSERVANT_CH_)
 #define _PORTABLESERVER_POA_NOSERVANT_CH_
     
@@ -528,6 +534,8 @@
 
 #endif /* end #if !defined */
     
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
     // TAO_IDL - Generated from
     // be\be_visitor_exception/exception_ch.cpp:51
 
@@ -871,6 +879,7 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ThreadPolicy_ptr create_thread_policy (
         ::PortableServer::ThreadPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -878,6 +887,7 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
@@ -915,6 +925,7 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ImplicitActivationPolicy_ptr create_implicit_activation_policy (
         ::PortableServer::ImplicitActivationPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -922,10 +933,12 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ServantRetentionPolicy_ptr create_servant_retention_policy (
         ::PortableServer::ServantRetentionPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -933,10 +946,12 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::RequestProcessingPolicy_ptr create_request_processing_policy (
         ::PortableServer::RequestProcessingPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -944,6 +959,7 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
@@ -988,6 +1004,8 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
+
     virtual ::PortableServer::AdapterActivator_ptr the_activator (
         ACE_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
@@ -1005,10 +1023,12 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ServantManager_ptr get_servant_manager (
         ACE_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
@@ -1028,10 +1048,13 @@
         CORBA::SystemException,
         ::PortableServer::POA::WrongPolicy
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
+
     virtual ::PortableServer::Servant get_servant (
         ACE_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
@@ -1053,6 +1076,8 @@
         ::PortableServer::POA::WrongPolicy
       )) = 0;
     
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
@@ -1314,11 +1339,15 @@
 // TAO_IDL - Generated from
 // be\be_visitor_exception/any_op_ch.cpp:52
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, const PortableServer::POA::NoServant &); // copying version
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::POA::NoServant*); // noncopying version
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::POA::NoServant *&); // deprecated
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, const PortableServer::POA::NoServant *&);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_visitor_exception/any_op_ch.cpp:52
 
@@ -1386,4 +1415,3 @@
 
 #endif /* ifndef */
 
-
--- orig/PortableServerC.cpp	2005-02-18 11:19:45.718564800 +0100
+++ PortableServerC.cpp	2005-02-18 11:19:48.732899200 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "PortableServerC.h"
+#include "PortableServer.h"
 #include "tao/CDR.h"
 #include "tao/ORB_Core.h"
 #include "tao/Typecode.h"
@@ -50,7 +50,6 @@
 // Arg traits specializations.
 namespace TAO
 {
-
 #if !defined (_PORTABLESERVER_ADAPTERACTIVATOR__ARG_TRAITS_CS_)
 #define _PORTABLESERVER_ADAPTERACTIVATOR__ARG_TRAITS_CS_
   
@@ -69,7 +68,6 @@
 #endif /* end #if !defined */
 }
 
-
 // TAO_IDL - Generated from 
 // be\be_visitor_sequence/sequence_cs.cpp:65
 
@@ -599,6 +597,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/exception_cs.cpp:63
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 PortableServer::POA::NoServant::NoServant (void)
   : CORBA::UserException (
         "IDL:omg.org/PortableServer/POA/NoServant:2.3",
@@ -728,6 +728,8 @@
 ::CORBA::TypeCode_ptr PortableServer::POA::_tc_NoServant =
   &_tc_TAO_tc_PortableServer_POA_NoServant;
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/exception_cs.cpp:63
 
--- orig/PortableServerA.cpp	2005-02-18 11:19:45.718564800 +0100
+++ PortableServerA.cpp	2005-02-18 11:19:48.752928000 +0100
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "PortableServerC.h"
+#include "PortableServer.h"
 #include "tao/Typecode.h"
 #include "tao/CDR.h"
 #include "tao/Any.h"
@@ -73,6 +73,8 @@
     &_tc_TAO_tc_PortableServer_POAList;
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:295
 
@@ -179,6 +181,8 @@
       );
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_visitor_interface/any_op_cs.cpp:50
 
@@ -467,6 +471,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/any_op_cs.cpp:50
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 ACE_TEMPLATE_SPECIALIZATION
 CORBA::Boolean
 TAO::Any_Dual_Impl_T<PortableServer::POA::NoServant>::marshal_value (TAO_OutputCDR &)
@@ -536,6 +542,8 @@
       );
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/any_op_cs.cpp:50
 
@@ -985,11 +993,15 @@
         PortableServer::POA::InvalidPolicy
       >;
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
   template class
     TAO::Any_Dual_Impl_T<
         PortableServer::POA::NoServant
       >;
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
   template class
     TAO::Any_Dual_Impl_T<
         PortableServer::POA::ObjectAlreadyActive
@@ -1047,11 +1059,15 @@
         PortableServer::POA::InvalidPolicy \
       >
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 # pragma instantiate \
     TAO::Any_Dual_Impl_T< \
         PortableServer::POA::NoServant \
       >
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 # pragma instantiate \
     TAO::Any_Dual_Impl_T< \
         PortableServer::POA::ObjectAlreadyActive \
