--- ACE_wrappers.orig/ace/Makefile.am
+++ ACE_wrappers/ace/Makefile.am
@@ -10,6 +10,7 @@
 
 includedir = @includedir@/ace
 pkgconfigdir = @libdir@/pkgconfig
+nobase_include_HEADERS =
 
 ACE_BUILDDIR = $(top_builddir)
 ACE_ROOT = $(top_srcdir)
@@ -336,7 +337,7 @@
 libACE_la_LDFLAGS = \
  -release @ACE_VERSION_NAME@
 
-nobase_include_HEADERS = \
+nobase_include_HEADERS += \
   ACE.h \
   ACE.inl \
   ACE_export.h \
