--- orig/BiDirPolicyC.h	2003-11-03 21:04:56.000000000 -0600
+++ BiDirPolicyC.h	2003-11-03 21:20:15.000000000 -0600
@@ -48,7 +48,7 @@
 #include "tao/Object.h"
 #include "tao/Objref_VarOut_T.h"
 
-#include "PolicyC.h"
+#include "tao/PolicyC.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
--- orig/BiDirPolicyC.cpp	2003-11-03 21:04:10.000000000 -0600
+++ BiDirPolicyC.cpp	2003-11-03 21:19:02.000000000 -0600
@@ -29,7 +29,7 @@
 // be/be_codegen.cpp:302
 
 
-#include "BiDirPolicyC.h"
+#include "BiDirGIOP.h"
 #include "ace/OS_NS_string.h"
 
 #if defined (__BORLANDC__)
