--- orig/MessagingS_T.i	Sun Apr 30 18:03:08 2000
+++ MessagingS_T.i	Sun Apr 30 18:06:01 2000
@@ -8,6 +8,7 @@
 //                 http://www.cs.wustl.edu/~schmidt/TAO.html
 
 
+#if (TAO_HAS_AMI_CALLBACK == 1)
 #if defined (ACE_HAS_USING_KEYWORD)
 template <class T> ACE_INLINE
 POA_Messaging::ReplyHandler_tie<T>::ReplyHandler_tie (T &t)
@@ -87,3 +88,4 @@
 }
 
 #endif /* ACE_HAS_USING_KEYWORD */
+#endif /* TAO_HAS_AMI_CALLBACK == 1 */
