--- orig/RTPortableServerC.h	2003-12-29 10:19:56.000000000 -0600
+++ RTPortableServerC.h	2003-12-29 11:31:43.000000000 -0600
@@ -45,8 +45,8 @@
 #include "tao/Object.h"
 #include "tao/Objref_VarOut_T.h"
 
-#include "../PortableServer/PortableServerC.h"
-#include "../RTCORBA/RTCORBAC.h"
+#include "tao/PortableServer/PortableServerC.h"
+#include "tao/RTCORBA/RTCORBA.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
@@ -228,11 +228,6 @@
 #endif /* end #if !defined */
   
   // TAO_IDL - Generated from
-  // be/be_visitor_typecode/typecode_decl.cpp:44
-  
-  TAO_NAMESPACE_STORAGE_CLASS ::CORBA::TypeCode_ptr _tc_POA;
-
-// TAO_IDL - Generated from
 // be/be_visitor_module/module_ch.cpp:66
 
 } // module RTPortableServer
--- orig/RTPortableServerC.cpp	2003-12-29 10:19:56.000000000 -0600
+++ RTPortableServerC.cpp	2003-12-29 11:29:44.000000000 -0600
@@ -199,40 +199,6 @@
   return 0;
 }
 
-// TAO_IDL - Generated from
-// be/be_visitor_typecode/typecode_defn.cpp:284
-
-static const CORBA::Long _oc_RTPortableServer_POA[] =
-{
-    TAO_ENCAP_BYTE_ORDER, // byte order
-  37,
-  ACE_NTOHL (0x49444c3a), 
-  ACE_NTOHL (0x6f6d672e), 
-  ACE_NTOHL (0x6f72672f), 
-  ACE_NTOHL (0x5254506f), 
-  ACE_NTOHL (0x72746162), 
-  ACE_NTOHL (0x6c655365), 
-  ACE_NTOHL (0x72766572), 
-  ACE_NTOHL (0x2f504f41), 
-  ACE_NTOHL (0x3a312e30), 
-  ACE_NTOHL (0x0),  // repository ID = IDL:omg.org/RTPortableServer/POA:1.0
-    4,
-  ACE_NTOHL (0x504f4100),  // name = POA
-  };
-
-static CORBA::TypeCode _tc_TAO_tc_RTPortableServer_POA (
-    CORBA::tk_objref,
-    sizeof (_oc_RTPortableServer_POA),
-    (char *) &_oc_RTPortableServer_POA,
-    0,
-    sizeof (RTPortableServer::POA)
-  );
-
-namespace RTPortableServer
-{
-  ::CORBA::TypeCode_ptr _tc_POA =
-    &_tc_TAO_tc_RTPortableServer_POA;
-}
 
 // TAO_IDL - Generated from
 // be/be_visitor_root/root.cpp:1725
--- orig/RTPortableServerA.h	2003-12-29 10:19:56.000000000 -0600
+++ RTPortableServerA.h	2003-12-29 11:29:01.000000000 -0600
@@ -40,6 +40,14 @@
 #endif
 #define TAO_EXPORT_MACRO TAO_RTPortableServer_Export
 
+namespace RTPortableServer
+{
+  // TAO_IDL - Generated from
+  // be/be_visitor_typecode/typecode_decl.cpp:44
+
+  TAO_NAMESPACE_STORAGE_CLASS ::CORBA::TypeCode_ptr _tc_POA;
+}
+
 // TAO_IDL - Generated from
 // be/be_visitor_interface/any_op_ch.cpp:52
 
--- orig/RTPortableServerA.cpp	2003-12-29 10:19:56.000000000 -0600
+++ RTPortableServerA.cpp	2003-12-29 11:29:52.000000000 -0600
@@ -32,6 +32,41 @@
 
 // TAO_IDL - Generated from
 // be/be_visitor_interface/any_op_cs.cpp:50
+// TAO_IDL - Generated from
+// be/be_visitor_typecode/typecode_defn.cpp:284
+
+static const CORBA::Long _oc_RTPortableServer_POA[] =
+{
+    TAO_ENCAP_BYTE_ORDER, // byte order
+  37,
+  ACE_NTOHL (0x49444c3a),
+  ACE_NTOHL (0x6f6d672e),
+  ACE_NTOHL (0x6f72672f),
+  ACE_NTOHL (0x5254506f),
+  ACE_NTOHL (0x72746162),
+  ACE_NTOHL (0x6c655365),
+  ACE_NTOHL (0x72766572),
+  ACE_NTOHL (0x2f504f41),
+  ACE_NTOHL (0x3a312e30),
+  ACE_NTOHL (0x0),  // repository ID = IDL:omg.org/RTPortableServer/POA:1.0
+    4,
+  ACE_NTOHL (0x504f4100),  // name = POA
+  };
+
+static CORBA::TypeCode _tc_TAO_tc_RTPortableServer_POA (
+    CORBA::tk_objref,
+    sizeof (_oc_RTPortableServer_POA),
+    (char *) &_oc_RTPortableServer_POA,
+    0,
+    sizeof (RTPortableServer::POA)
+  );
+
+namespace RTPortableServer
+{
+  ::CORBA::TypeCode_ptr _tc_POA =
+    &_tc_TAO_tc_RTPortableServer_POA;
+}
+
 
 ACE_TEMPLATE_SPECIALIZATION
 CORBA::Boolean
