--- orig/ServantLocatorC.h	2004-09-08 13:11:22.475258200 +0200
+++ ServantLocatorC.h	2004-09-08 13:26:39.694154200 +0200
@@ -94,6 +94,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_root/root_ch.cpp:63
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace TAO
 {
   class Collocation_Proxy_Broker;
@@ -141,6 +143,8 @@
     : public virtual ::PortableServer::ServantManager
   {
   public:
+    typedef void *Cookie;
+
     typedef ServantLocator_ptr _ptr_type;
     typedef ServantLocator_var _var_type;
     
@@ -266,6 +270,8 @@
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ServantLocator_ptr *); // non-copying
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ServantLocator_ptr &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:964
 
--- orig/ServantLocatorC.cpp	2004-09-08 13:11:22.475258200 +0200
+++ ServantLocatorC.cpp	2004-09-08 13:27:07.313869400 +0200
@@ -45,6 +45,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:64
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -238,3 +240,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantLocatorA.cpp	2004-09-08 13:11:22.475258200 +0200
+++ ServantLocatorA.cpp	2004-09-08 13:27:29.876312600 +0200
@@ -32,6 +32,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:290
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ServantLocator[] =
 {
     TAO_ENCAP_BYTE_ORDER, // byte order
@@ -155,3 +157,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
\ No newline at end of file
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
