--- ACE_wrappers/ace/config.h~	2006-04-19 13:19:34.000000000 -0700
+++ ACE_wrappers/ace/config.h	2006-04-19 13:22:10.000000000 -0700
@@ -21,6 +21,8 @@
 #define ACE_HAS_AUTOMATIC_INIT_FINI
 #define ACE_HAS_DLSYM_SEGFAULT_ON_INVALID_HANDLE
 
+#define ACE_HAS_REACTOR_NOTIFICATION_QUEUE  // Ken Sedgwick 2006-04-19
+
 #if !defined (ACE_MT_SAFE)
 #define ACE_MT_SAFE 1                           // JCEJ 12/22/96        #1
 #endif
