diff -uNrp a/ace-tao.spec b/ace-tao.spec
--- a/ace-tao.spec	2010-06-11 13:45:09.000000000 -0400
+++ b/ace-tao.spec	2010-06-11 13:46:17.000000000 -0400
@@ -1,7 +1,7 @@
 # Set the version number here.
-%define ACEVER  5.7.9
-%define TAOVER  1.7.9
-%define CIAOVER 0.7.9
+%define ACEVER  5.7.0
+%define TAOVER  1.7.0
+%define CIAOVER 0.7.0
 
 # Conditional build
 # Default values are
@@ -1281,8 +1281,8 @@ install ${ACE_ROOT}/bin/tao_nslist %{bui
 install -d %{buildroot}%{_sysconfdir}
 mkdir -p %{buildroot}%{_sysconfdir}/logrotate.d
 mkdir -p %{buildroot}%{_sysconfdir}/tao
-cp -R ${ACE_ROOT}/rpmbuild/etc/logrotate.d/* %{buildroot}%{_sysconfdir}/logrotate.d/
-cp -R ${ACE_ROOT}/rpmbuild/etc/tao/* %{buildroot}%{_sysconfdir}/tao/
+cp -R ${ACE_ROOT}/rpmbuild/logrotate.d/* %{buildroot}%{_sysconfdir}/logrotate.d/
+cp -R ${ACE_ROOT}/rpmbuild/tao/* %{buildroot}%{_sysconfdir}/tao/
 
 %if 0%{?suse_version}
 mkdir -p %{buildroot}%{_sysconfdir}/init.d
@@ -2034,7 +2034,6 @@ fi
 %files -n ace-gperf
 %defattr(-,root,root,-)
 %{_bindir}/ace_gperf
-%{_libdir}/libACE_gperf_lib.so.%{ACEVERSO}
 %attr(0644,root,root) %{_mandir}/man1/ace_gperf.1%{_extension}
 %attr(0644,root,root) %{_infodir}/ace_gperf.info%{_extension}
 
@@ -2347,8 +2346,6 @@ fi
 %doc TAO/PROBLEM-REPORT-FORM
 %doc TAO/VERSION
 %doc TAO/README
-%doc TAO/utils/catior/README.catior
-%doc TAO/utils/nslist/README.nslist
 
 # ---------------- tao-cosnaming ----------------
 
@@ -2362,7 +2359,7 @@ fi
 %if 0%{?suse_version}
 %{_sysconfdir}/init.d/tao-cosnaming
 %{_sbindir}/rctao-cosnaming
-%{_sysconfdir}/tao/tao-cosnaming
+%{_sysconfdir}/tao/tao-cosnaming.opt
 %else
 %{_sysconfdir}/rc.d/init.d/tao-cosnaming
 %config(noreplace) %{_sysconfdir}/tao/tao-cosnaming.opt
@@ -2391,7 +2388,7 @@ fi
 %if 0%{?suse_version}
 %{_sysconfdir}/init.d/tao-cosevent
 %{_sbindir}/rctao-cosevent
-%{_sysconfdir}/tao/tao-cosevent
+%{_sysconfdir}/tao/tao-cosevent.opt
 %else
 %{_sysconfdir}/rc.d/init.d/tao-cosevent
 %config(noreplace) %{_sysconfdir}/tao/tao-cosevent.opt
@@ -2418,7 +2415,7 @@ fi
 %if 0%{?suse_version}
 %{_sysconfdir}/init.d/tao-cosnotification
 %{_sbindir}/rctao-cosnotification
-%{_sysconfdir}/tao/tao-cosnotification
+%{_sysconfdir}/tao/tao-cosnotification.opt
 %else
 %{_sysconfdir}/rc.d/init.d/tao-cosnotification
 %config(noreplace) %{_sysconfdir}/tao/tao-cosnotification.opt
@@ -2446,7 +2443,7 @@ fi
 %if 0%{?suse_version}
 %{_sysconfdir}/init.d/tao-costrading
 %{_sbindir}/rctao-costrading
-%{_sysconfdir}/tao/tao-costrading
+%{_sysconfdir}/tao/tao-costrading.opt
 %else
 %{_sysconfdir}/rc.d/init.d/tao-costrading
 %config(noreplace) %{_sysconfdir}/tao/tao-costrading.opt
@@ -2473,7 +2470,7 @@ fi
 %if 0%{?suse_version}
 %{_sysconfdir}/init.d/tao-rtevent
 %{_sbindir}/rctao-rtevent
-%{_sysconfdir}/tao/tao-rtevent
+%{_sysconfdir}/tao/tao-rtevent.opt
 %else
 %{_sysconfdir}/rc.d/init.d/tao-rtevent
 %config(noreplace) %{_sysconfdir}/tao/tao-rtevent.opt
@@ -2500,7 +2497,7 @@ fi
 %if 0%{?suse_version}
 %{_sysconfdir}/init.d/tao-cosconcurrency
 %{_sbindir}/rctao-cosconcurrency
-%{_sysconfdir}/tao/tao-cosconcurrency
+%{_sysconfdir}/tao/tao-cosconcurrency.opt
 %else
 %{_sysconfdir}/rc.d/init.d/tao-cosconcurrency
 %config(noreplace) %{_sysconfdir}/tao/tao-cosconcurrency.opt
