--- ACE_wrappers/ace/Codeset_Registry_db.cpp~	2008-04-23 07:02:05.000000000 -0700
+++ ACE_wrappers/ace/Codeset_Registry_db.cpp	2008-05-19 13:15:15.000000000 -0700
@@ -22,9 +22,12 @@
   {"IBM-1047 (CCSID 01047); Latin-1 Open System","EBCDIC",0x10020417,1,{0x0011},1},
   {"ISO/IEC 10646-1:1993; UCS-4, Level 1","UCS-4",0x00010104,1,{0x1000},4},
   {"ISO/IEC 10646-1:1993; UTF-16, UCS Transformation Format 16-bit form","UTF-16",0x00010109,1,{0x1000},2},
-  {"X/Open UTF-8; UCS Transformation Format 8 (UTF-8)","UTF-8",0x05010001,1,{0x1000},6}
+  {"X/Open UTF-8; UCS Transformation Format 8 (UTF-8)","UTF-8",0x05010001,1,{0x1000},6},
+  {"ISO/IEC 8859-5:1988; Latin-Cyrillic Alphabet","ISO-8859-5",0x00010005,1,{0x0015},1},
+  {"IBM-1251 (CCSID 01251); MS Windows Cyrillic","CP1251",0x100204e3,1,{0x0015},1},
+  {"IBM-855 (CCSID 04951); Cyrillic Personal Computer","CP855",0x10021357,1,{0x0015},1}
 };
 
-size_t const ACE_Codeset_Registry::num_registry_entries_ = 6;
+size_t const ACE_Codeset_Registry::num_registry_entries_ = 9;
 
 ACE_END_VERSIONED_NAMESPACE_DECL
