--- orig/IORTableC.i	Sun Sep 10 16:20:41 2000
+++ IORTableC.i	Sun Sep 10 16:22:20 2000
@@ -354,42 +354,3 @@
 
 
 #endif /* end #if !defined */
-
-ACE_INLINE CORBA::Boolean
-operator<< (
-    TAO_OutputCDR &,
-    const IORTable::Locator_ptr
-  );
-ACE_INLINE CORBA::Boolean
-operator>> (
-    TAO_InputCDR &,
-    IORTable::Locator_ptr &
-  );
-ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const IORTable::AlreadyBound &_tao_aggregate)
-{
-  // first marshal the repository ID
-  if (strm << _tao_aggregate._id ())
-    return 1;
-  else
-    return 0;
-}
-
-ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &,IORTable::AlreadyBound&)
-{
-  return 1;
-}
-
-ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const IORTable::NotFound &_tao_aggregate)
-{
-  // first marshal the repository ID
-  if (strm << _tao_aggregate._id ())
-    return 1;
-  else
-    return 0;
-}
-
-ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &,IORTable::NotFound&)
-{
-  return 1;
-}
-
