--- ACE_wrappers/apps/gperf/gperf.info~	2004-04-13 00:57:34.000000000 -0700
+++ ACE_wrappers/apps/gperf/gperf.info	2005-02-17 10:32:17.831361209 -0800
@@ -3,7 +3,7 @@
 
 INFO-DIR-SECTION GNU programming tools
 START-INFO-DIR-ENTRY
-* Gperf: (gperf).                Perfect Hash Function Generator.
+* Gperf-ace: (gperf-ace).                Perfect Hash Function Generator.
 END-INFO-DIR-ENTRY
 
    This file documents the features of the GNU Perfect Hash Function
