--- orig/ImR_ActivatorC.h	2003-04-25 17:36:27.000000000 -0500
+++ ./ImR_ActivatorC.h	2003-04-25 17:39:34.000000000 -0500
@@ -30,17 +30,16 @@
 
 #ifndef _TAO_IDL_ORIG_IMR_ACTIVATORC_H_
 #define _TAO_IDL_ORIG_IMR_ACTIVATORC_H_
-
 #include "ace/pre.h"
+#include "portableserver_export.h"
 
-#include "tao/corba.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 
-#include "portableserver_export.h"
+
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
@@ -92,4 +91,3 @@
 
 #include "ace/post.h"
 #endif /* ifndef */
-
