--- y.tab.cpp	Fri Jun 18 18:28:04 1999
+++ y.tab.cpp.new	Fri Jun 18 18:20:42 1999
@@ -1101,7 +1101,7 @@
     extern char *ace_foo();
 #endif
 
-    if (tao_yys = ACE_OS::getenv("TAO_YYDEBUG"))
+    if ((tao_yys = ACE_OS::getenv("TAO_YYDEBUG")))
     {
         tao_yyn = *tao_yys;
         if (tao_yyn >= '0' && tao_yyn <= '9')
@@ -1118,7 +1118,7 @@
     *tao_yyssp = tao_yystate = 0;
 
 tao_yyloop:
-    if (tao_yyn = tao_yydefred[tao_yystate]) goto tao_yyreduce;
+    if ((tao_yyn = tao_yydefred[tao_yystate])) goto tao_yyreduce;
     if (tao_yychar < 0)
     {
         if ((tao_yychar = tao_yylex()) < 0) tao_yychar = 0;
@@ -1172,7 +1172,6 @@
 #ifdef lint
     goto tao_yyerrlab;
 #endif
-tao_yyerrlab:
     ++tao_yynerrs;
 tao_yyinrecovery:
     if (tao_yyerrflag < 3)
