--- orig/IOP_IORC.h	Mon Mar 31 11:56:50 2003
+++ IOP_IORC.h	Mon Mar 31 11:58:18 2003
@@ -27,16 +27,15 @@
 
 #include "ace/pre.h"
 
-#include "tao/corba.h"
+#include "tao/corbafwd.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
-
-#include "TAO_Export.h"
-
-#include "OctetSeqC.h"
+#include "tao/TAO_Export.h"
+#include "tao/OctetSeqC.h"
+#include "tao/CDR.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
