--- lex.yy.cpp	Fri Jun 18 18:36:24 1999
+++ lex.yy.cpp.new	Fri Jun 18 18:34:12 1999
@@ -927,7 +927,7 @@
 
 /* Code executed at the end of each rule. */
 #ifndef TAO_YY_BREAK
-#define TAO_YY_BREAK break;
+#define TAO_YY_BREAK ACE_NOTREACHED (break;)
 #endif
 
 #define TAO_YY_RULE_SETUP \
@@ -1405,7 +1405,7 @@
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		  idl_store_pragma(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 64:
 #line 263 "fe/idl.ll"
 case 65:
@@ -1414,7 +1414,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 66:
 #line 267 "fe/idl.ll"
 case 67:
@@ -1423,7 +1423,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 68:
 #line 271 "fe/idl.ll"
 case 69:
@@ -1432,7 +1432,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 70:
 #line 275 "fe/idl.ll"
 case 71:
@@ -1441,7 +1441,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 	        }
-	TAO_YY_BREAK
+	break;
 case 72:
 #line 279 "fe/idl.ll"
 case 73:
@@ -1451,7 +1451,7 @@
 		  /* ignore cpp ident */
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 74:
 TAO_YY_RULE_SETUP
 #line 283 "fe/idl.ll"
@@ -1459,7 +1459,7 @@
 		  /* ignore comments */
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 75:
 TAO_YY_RULE_SETUP
 #line 287 "fe/idl.ll"
@@ -1477,7 +1477,7 @@
 		    }
 	          }
 	        }
-	TAO_YY_BREAK
+	break;
 case 76:
 TAO_YY_RULE_SETUP
 #line 301 "fe/idl.ll"
@@ -1489,7 +1489,7 @@
 {
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 78:
 TAO_YY_RULE_SETUP
 #line 305 "fe/idl.ll"
