--- lex.yy.cpp.orig	2006-11-16 10:17:22.000000000 -0600
+++ lex.yy.cpp	2006-11-16 10:18:54.000000000 -0600
@@ -1600,42 +1600,42 @@
                   idl_global->set_lineno(idl_global->lineno() + 1);
                   idl_store_pragma(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+        break;
 case 87:
 case 88:
 TAO_YY_RULE_SETUP
 {/* ignore file */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+        break;
 case 89:
 case 90:
 TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+        break;
 case 91:
 case 92:
 TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+        break;
 case 93:
 case 94:
 TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+        break;
 case 95:
 case 96:
 TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+        break;
 case 97:
 case 98:
 TAO_YY_RULE_SETUP
+@@ -1643,14 +1643,14 @@
                   /* ignore cpp ident */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+        break;
 case 99:
 TAO_YY_RULE_SETUP
 {
                   /* ignore comments */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+        break;
 case 100:
 TAO_YY_RULE_SETUP
 {
@@ -1667,17 +1667,17 @@
                     }
                   }
                 }
-        TAO_YY_BREAK
+        break;
 case 101:
 TAO_YY_RULE_SETUP
 ;
-        TAO_YY_BREAK
+        break;
 case 102:
 TAO_YY_RULE_SETUP
 {
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+        break;
 case 103:
 TAO_YY_RULE_SETUP
 return ace_tao_yytext [0];
@@ -1685,7 +1685,7 @@
 case 104:
 TAO_YY_RULE_SETUP
 TAO_YY_ECHO;
-        TAO_YY_BREAK
+        break;
 case TAO_YY_STATE_EOF(INITIAL):
         tao_yyterminate();
  