--- lex.yy.cpp.orig	Mon Apr 17 13:15:44 2000
+++ lex.yy.cpp	Wed Jan 24 16:38:19 2001
@@ -983,7 +983,7 @@
 
 /* Code executed at the end of each rule. */
 #ifndef TAO_YY_BREAK
-#define TAO_YY_BREAK break;
+#define TAO_YY_BREAK ACE_NOTREACHED (break;)
 #endif
 
 #define TAO_YY_RULE_SETUP \
@@ -998,7 +998,7 @@
 TAO_YY_DECL
 	{
 	register tao_yy_state_type tao_yy_current_state;
-	register char *tao_yy_cp, *tao_yy_bp;
+	register char *tao_yy_cp = 0, *tao_yy_bp = 0;
 	register int tao_yy_act;
 
 #line 126 "fe/idl.ll"
@@ -1503,7 +1503,7 @@
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		  idl_store_pragma(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 69:
 #line 284 "fe/idl.ll"
 case 70:
@@ -1512,7 +1512,7 @@
{/* ignore file */
  		  idl_global->set_lineno(idl_global->lineno() + 1);
		}
-	TAO_YY_BREAK
+       break;
case 71:
#line 288 "fe/idl.ll"
case 72:
@@ -1521,7 +1521,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 73:
 #line 292 "fe/idl.ll"
 case 74:
@@ -1530,7 +1530,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 75:
 #line 296 "fe/idl.ll"
 case 76:
@@ -1539,7 +1539,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 		}
-	TAO_YY_BREAK
+	break;
 case 77:
 #line 300 "fe/idl.ll"
 case 78:
@@ -1548,7 +1548,7 @@
 {
 		  idl_parse_line_and_file(ace_tao_yytext);
 	        }
-	TAO_YY_BREAK
+	break;
 case 79:
 #line 304 "fe/idl.ll"
 case 80:
@@ -1557,7 +1557,7 @@
{
 		  /* store cpp ident */
                  idl_global->ident_string (ace_tao_yytext);
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 81:
 TAO_YY_RULE_SETUP
 #line 308 "fe/idl.ll"
@@ -1566,7 +1566,7 @@
 		  /* ignore comments */
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 82:
 TAO_YY_RULE_SETUP
 #line 312 "fe/idl.ll"
@@ -1584,19 +1584,19 @@
 		    }
 	          }
 	        }
-	TAO_YY_BREAK
+	break;
 case 83:
 TAO_YY_RULE_SETUP
 #line 326 "fe/idl.ll"
 ;
-	TAO_YY_BREAK
+	break;
 case 84:
 TAO_YY_RULE_SETUP
 #line 327 "fe/idl.ll"
 {
   		  idl_global->set_lineno(idl_global->lineno() + 1);
 		}
-	TAO_YY_BREAK
+	break;
 case 85:
 TAO_YY_RULE_SETUP
 #line 330 "fe/idl.ll"
@@ -1606,7 +1606,7 @@
 TAO_YY_RULE_SETUP
 #line 332 "fe/idl.ll"
 TAO_YY_ECHO;
-	TAO_YY_BREAK
+	break;
 case TAO_YY_STATE_EOF(INITIAL):
 	tao_yyterminate();
 
