--- Domain.pidl	Fri Apr 13 23:29:22 2001
+++ orig/Domain.pidl	Fri Apr 13 23:27:34 2001
@@ -75,18 +75,19 @@
 
 module CORBA
 {
-  interface InterfaceDef;
+  const PolicyType SecConstruction = 11;
+};
+
+interface CORBA_InterfaceDef;
 
-  interface DomainManager
+interface CORBA_DomainManager
   {
-    Policy get_domain_policy (
-        in PolicyType policy_type
+  CORBA_Policy get_domain_policy (
+      in CORBA_PolicyType policy_type
       );
   };
 
-  const PolicyType SecConstruction = 11;
-
-  interface ConstructionPolicy : Policy
+interface CORBA_ConstructionPolicy : CORBA_Policy
   {
     void make_domain_manager (
         in InterfaceDef object_type,
@@ -94,8 +95,7 @@
       );
   };
 
-  typedef sequence <DomainManager> DomainManagerList;
-};
+typedef sequence <CORBA_DomainManager> CORBA_DomainManagerList;
 
 #pragma prefix ""
 
