--- orig/IORC.h	2005-02-14 14:33:17.541928300 +0100
+++ IORC.h	2005-02-14 14:35:33.857941100 +0100
@@ -40,6 +40,10 @@
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
+ #ifndef TAO_IORMANIP_SAFE_INCLUDE
+ #error "You should not include IORC.h directly, use IORManip_Loader.h"
+ #endif /* !TAO_IORMANIP_SAFE_INCLUDE */
+
 #include "ior_manip_export.h"
 #include "tao/ORB.h"
 #include "tao/SystemException.h"
--- orig/IORC.cpp	2005-02-14 14:33:17.541928300 +0100
+++ IORC.cpp	2005-02-14 14:33:48.095862700 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "IORC.h"
+#include "IORManip_Loader.h"
 #include "tao/CDR.h"
 #include "tao/Typecode.h"
 #include "ace/OS_NS_string.h"
--- orig/IORA.cpp	2005-02-14 14:33:17.541928300 +0100
+++ IORA.cpp	2005-02-14 14:38:14.699219500 +0100
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "IORC.h"
+#include "IORManip_Loader.h"
 #include "tao/Typecode.h"
 #include "tao/CDR.h"
 #include "tao/Any.h"
