--- orig/ImplRepoC.h	Thu Jan  2 10:12:25 2003
+++ ImplRepoC.h	Thu Jan  2 10:10:22 2003
@@ -27,7 +27,7 @@
 
 #include "ace/pre.h"
 
-#include "tao/corba.h"
+#include "tao/corbafwd.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
@@ -36,6 +36,11 @@
 
 #include "portableserver_export.h"
 
+#include "tao/Exception.h"
+#include "tao/Object.h"
+#include "tao/Sequence.h"
+#include "tao/Remote_Object_Proxy_Impl.h"
+
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
