--- orig/TAOC.i	Mon May  1 09:12:20 2000
+++ TAOC.i	Mon May  1 09:10:57 2000
@@ -11,6 +11,8 @@
 // Inline operations for class TAO::PrioritySpecification_var
 // *************************************************************
 
+#if (TAO_HAS_CLIENT_PRIORITY_POLICY == 1)
+
 ACE_INLINE
 TAO::PrioritySpecification_var::PrioritySpecification_var (void) // default constructor
   : ptr_ (0)
@@ -93,13 +95,13 @@
 }
 
 ACE_INLINE
-TAO::PrioritySpecification_var::operator ::TAO::PrioritySpecification &() // cast 
+TAO::PrioritySpecification_var::operator ::TAO::PrioritySpecification &() // cast
 {
   return *this->ptr_;
 }
 
 ACE_INLINE
-TAO::PrioritySpecification_var::operator ::TAO::PrioritySpecification &() const // cast 
+TAO::PrioritySpecification_var::operator ::TAO::PrioritySpecification &() const // cast
 {
   return *this->ptr_;
 }
@@ -116,7 +118,7 @@
   return *this->ptr_;
 }
 
-// mapping for fixed size 
+// mapping for fixed size
 ACE_INLINE ::TAO::PrioritySpecification &
 TAO::PrioritySpecification_var::out (void)
 {
@@ -197,14 +199,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 TAO::ClientPriorityPolicy_var::operator const ::TAO::ClientPriorityPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-TAO::ClientPriorityPolicy_var::operator ::TAO::ClientPriorityPolicy_ptr &() // cast 
+ACE_INLINE
+TAO::ClientPriorityPolicy_var::operator ::TAO::ClientPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -296,7 +298,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 TAO::ClientPriorityPolicy_out::operator ::TAO::ClientPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -317,6 +319,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_CLIENT_PRIORITY_POLICY == 1 */
+
+#if (TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1)
+
 // *************************************************************
 // Inline operations for class TAO::BufferingConstraint_var
 // *************************************************************
@@ -403,13 +409,13 @@
 }
 
 ACE_INLINE
-TAO::BufferingConstraint_var::operator ::TAO::BufferingConstraint &() // cast 
+TAO::BufferingConstraint_var::operator ::TAO::BufferingConstraint &() // cast
 {
   return *this->ptr_;
 }
 
 ACE_INLINE
-TAO::BufferingConstraint_var::operator ::TAO::BufferingConstraint &() const // cast 
+TAO::BufferingConstraint_var::operator ::TAO::BufferingConstraint &() const // cast
 {
   return *this->ptr_;
 }
@@ -426,7 +432,7 @@
   return *this->ptr_;
 }
 
-// mapping for fixed size 
+// mapping for fixed size
 ACE_INLINE ::TAO::BufferingConstraint &
 TAO::BufferingConstraint_var::out (void)
 {
@@ -507,14 +513,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 TAO::BufferingConstraintPolicy_var::operator const ::TAO::BufferingConstraintPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-TAO::BufferingConstraintPolicy_var::operator ::TAO::BufferingConstraintPolicy_ptr &() // cast 
+ACE_INLINE
+TAO::BufferingConstraintPolicy_var::operator ::TAO::BufferingConstraintPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -606,7 +612,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 TAO::BufferingConstraintPolicy_out::operator ::TAO::BufferingConstraintPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -637,7 +643,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &strm, TAO::PrioritySpecification &_tao_aggregate)
@@ -650,7 +656,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const TAO::BufferingConstraint &_tao_aggregate)
@@ -664,7 +670,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &strm, TAO::BufferingConstraint &_tao_aggregate)
@@ -678,6 +684,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
+#endif /* TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1 */
