--- orig/IORInterceptorC.h	2005-02-18 12:20:12.583740800 +0100
+++ IORInterceptorC.h	2005-02-18 12:20:13.304777600 +0100
@@ -34,6 +34,9 @@
 #include /**/ "ace/pre.h"
 
 
+#ifndef TAO_IORINTERCEPTOR_SAFE_INCLUDE
+#error "You should not include IORInterceptorC.h directly, use IORInterceptor.h"
+#endif /* !TAO_IORINTERCEPTOR_SAFE_INCLUDE */
 #include "ace/config-all.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
--- orig/IORInterceptorC.cpp	2005-02-18 12:20:12.583740800 +0100
+++ IORInterceptorC.cpp	2005-02-18 12:20:13.304777600 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "IORInterceptorC.h"
+#include "IORInterceptor.h"
 #include "tao/CDR.h"
 #include "ace/OS_NS_string.h"
 
