--- orig/POAManagerC.h	2004-11-17 09:22:01.508046400 +0100
+++ POAManagerC.h	2004-11-17 09:22:11.702705600 +0100
@@ -221,6 +221,8 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
+
     virtual void hold_requests (
         ::CORBA::Boolean wait_for_completion
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -255,6 +257,8 @@
         ::PortableServer::POAManager::AdapterInactive
       )) = 0;
     
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
