--- orig/RequestProcessingPolicyC.h	2004-10-29 14:44:23.775264000 +0200
+++ RequestProcessingPolicyC.h	2004-10-29 14:44:27.370433600 +0200
@@ -73,6 +73,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_root/root_ch.cpp:63
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace TAO
 {
   class Collocation_Proxy_Broker;
@@ -255,6 +257,8 @@
 #endif /* end #if !defined */
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_visitor_enum/any_op_ch.cpp:51
 
--- orig/RequestProcessingPolicyC.cpp	2004-10-29 14:44:23.775264000 +0200
+++ RequestProcessingPolicyC.cpp	2004-10-29 14:44:27.470577600 +0200
@@ -45,6 +45,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:64
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -260,3 +262,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/RequestProcessingPolicyA.cpp	2004-10-29 14:44:23.795292800 +0200
+++ RequestProcessingPolicyA.cpp	2004-10-29 14:44:27.550692800 +0200
@@ -101,6 +101,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:290
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_RequestProcessingPolicy[] =
 {
     TAO_ENCAP_BYTE_ORDER, // byte order
@@ -264,3 +266,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
