--- orig/ServantRetentionPolicyC.h	2004-11-11 16:18:45.654998200 +0100
+++ ServantRetentionPolicyC.h	2004-11-11 16:18:50.632155000 +0100
@@ -73,6 +73,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_root/root_ch.cpp:63
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace TAO
 {
   class Collocation_Proxy_Broker;
@@ -254,6 +256,8 @@
 #endif /* end #if !defined */
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_visitor_enum/any_op_ch.cpp:51
 
--- orig/ServantRetentionPolicyC.cpp	2004-11-11 16:18:45.654998200 +0100
+++ ServantRetentionPolicyC.cpp	2004-11-11 16:18:50.722284600 +0100
@@ -45,6 +45,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:64
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -260,3 +262,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantRetentionPolicyA.cpp	2004-11-11 16:18:45.675027000 +0100
+++ ServantRetentionPolicyA.cpp	2004-11-11 16:18:50.802399800 +0100
@@ -87,6 +87,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:290
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ServantRetentionPolicy[] =
 {
     TAO_ENCAP_BYTE_ORDER, // byte order
@@ -250,3 +252,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
