--- orig/ServantActivatorC.h	2004-11-17 09:22:03.941545600 +0100
+++ ServantActivatorC.h	2004-11-17 09:22:12.654073600 +0100
@@ -74,6 +74,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_root/root_ch.cpp:63
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace TAO
 {
   class Collocation_Proxy_Broker;
@@ -244,6 +246,8 @@
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ServantActivator_ptr *); // non-copying
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ServantActivator_ptr &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:958
 
--- orig/ServantActivatorC.cpp	2004-11-17 09:22:03.941545600 +0100
+++ ServantActivatorC.cpp	2004-11-17 09:22:12.724174400 +0100
@@ -46,6 +46,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:64
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -239,3 +241,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantActivatorA.cpp	2004-11-17 09:22:03.961574400 +0100
+++ ServantActivatorA.cpp	2004-11-17 09:22:12.764232000 +0100
@@ -32,6 +32,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:295
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ServantActivator[] =
 {
     TAO_ENCAP_BYTE_ORDER, // byte order
@@ -154,3 +156,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
