--- orig/MessagingS.h   Sun Apr 30 18:03:08 2000
+++ MessagingS.h        Sun Apr 30 18:06:01 2000
@@ -11,11 +11,15 @@
 #define _TAO_IDL_ORIG_MESSAGINGS_H_

 #include "ace/pre.h"
+
+#include "orbconf.h"
+
+#if (TAO_HAS_CORBA_MESSAGING == 1)
+
 #include "TimeBaseS.h"
-#include "IOPS.h"
-#include "PolicyS.h"
-#include "PollableS.h"
 #include "MessagingC.h"
+#include "PortableServerC.h"
+#include "Servant_Base.h"


 #if !defined (ACE_LACKS_PRAGMA_ONCE)
@@ -31,6 +35,9 @@

 TAO_NAMESPACE  POA_Messaging
 {
+
+#if (TAO_HAS_AMI_CALLBACK == 1)
+
   class ReplyHandler;
   typedef ReplyHandler *ReplyHandler_ptr;
   class TAO_Export ReplyHandler :  public virtual PortableServer::ServantBase
@@ -45,7 +52,7 @@

     virtual CORBA::Boolean _is_a (
         const char* logical_type_id,
-        CORBA::Environment &ACE_TRY_ENV =
+        CORBA::Environment &ACE_TRY_ENV =
           TAO_default_environment ()
       );

@@ -57,7 +64,7 @@
         CORBA::ServerRequest &req,
         void *obj,
         void *context,
-        CORBA::Environment &ACE_TRY_ENV =
+        CORBA::Environment &ACE_TRY_ENV =
           TAO_default_environment ()
       );

@@ -65,19 +72,19 @@
         CORBA::ServerRequest &req,
         void *obj,
         void *context,
-        CORBA::Environment &ACE_TRY_ENV =
+        CORBA::Environment &ACE_TRY_ENV =
           TAO_default_environment ()
       );

     virtual void _dispatch (
         CORBA::ServerRequest &_tao_req,
         void *_tao_context,
-        CORBA::Environment &ACE_TRY_ENV =
+        CORBA::Environment &ACE_TRY_ENV =
           TAO_default_environment ()
       );

     ::Messaging::ReplyHandler *_this (
-        CORBA::Environment &ACE_TRY_ENV =
+        CORBA::Environment &ACE_TRY_ENV =
           TAO_default_environment ()
       );

@@ -100,11 +107,11 @@
         const CORBA::Char *logical_type_id,
         CORBA_Environment &ACE_TRY_ENV = TAO_default_environment ()
       );
-
+
     virtual CORBA::Boolean _non_existent(
         CORBA_Environment &ACE_TRY_ENV = TAO_default_environment ()
       );
-
+
       };


@@ -125,14 +132,14 @@
         const CORBA::Char *logical_type_id,
         CORBA_Environment &ACE_TRY_ENV = TAO_default_environment ()
       );
-
+
     ReplyHandler_ptr _get_servant (void) const;
-
+
     virtual CORBA::Boolean _non_existent(
         CORBA_Environment &ACE_TRY_ENV = TAO_default_environment ()
       );
-
-
+
+
   private:
     ReplyHandler_ptr servant_;
   };
@@ -140,7 +147,7 @@

 #endif /* end #if !defined */

-
+
 #if defined (ACE_HAS_USING_KEYWORD)
   // TIE class: Refer to CORBA v2.2, Section 20.34.4
   template <class T>
@@ -157,7 +164,7 @@
     // ctor with T*, ownership flag and a POA
     ~ReplyHandler_tie (void);
     // dtor
-
+
     // TIE specific functions
     T *_tied_object (void);
     // return the underlying object
@@ -169,17 +176,17 @@
     // do we own it
     void _is_owner (CORBA::Boolean b);
     // set the ownership
-
+
     // overridden ServantBase operations
     PortableServer::POA_ptr _default_POA (
-        CORBA::Environment &ACE_TRY_ENV =
+        CORBA::Environment &ACE_TRY_ENV =
           TAO_default_environment ()
       );
 private:
     T *ptr_;
     PortableServer::POA_var poa_;
     CORBA::Boolean rel_;
-
+
     // copy and assignment are not allowed
     ReplyHandler_tie (const ReplyHandler_tie &);
     void operator= (const ReplyHandler_tie &);
@@ -187,6 +194,8 @@

 #endif /* ACE_HAS_USING_KEYWORD */

+#endif /* TAO_HAS_AMI_CALLBACK == 1 */
+
 }
 TAO_NAMESPACE_CLOSE

@@ -199,6 +208,8 @@
 #if defined(_MSC_VER) && (_MSC_VER >= 1200)
 #pragma warning(pop)
 #endif /* _MSC_VER */
+
+#endif /* TAO_HAS_CORBA_MESSAGING == 1 */

 #include "ace/post.h"
 #endif /* ifndef */
