--- orig/RTSchedulerC.h	2005-03-29 14:15:37.951680000 +0200
+++ RTSchedulerC.h	2005-03-29 14:15:38.292169600 +0200
@@ -54,6 +54,10 @@
 #include "tao/PortableServer/PortableServer_includeC.h"
 #include "tao/PortableInterceptorC.h"
 
+#ifndef TAO_RTSCHEDULER_SAFE_INCLUDE
+#error "You should not include RTSchedulerC.h directly, use RTScheduler.h"
+#endif /* !TAO_RTSCHEDULER_SAFE_INCLUDE */
+
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
@@ -95,6 +99,9 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+  // Hand crafted
+  typedef void* VoidData;
+
 } // module CORBA
 
 // TAO_IDL - Generated from
--- orig/RTSchedulerC.cpp	2005-03-29 14:15:37.951680000 +0200
+++ RTSchedulerC.cpp	2005-03-29 14:15:38.302184000 +0200
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "RTSchedulerC.h"
+#include "RTScheduler.h"
 #include "tao/CDR.h"
 #include "tao/ORB_Core.h"
 #include "tao/Null_RefCount_Policy.h"
--- orig/RTSchedulerA.cpp	2005-03-29 14:15:37.951680000 +0200
+++ RTSchedulerA.cpp	2005-03-29 14:15:38.312198400 +0200
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "RTSchedulerC.h"
+#include "RTScheduler.h"
 #include "tao/Null_RefCount_Policy.h"
 #include "tao/TypeCode_Constants.h"
 #include "tao/Alias_TypeCode.h"
