--- lex.yy.cpp.orig	Fri Jan 23 11:21:41 1998
+++ lex.yy.cpp	Fri Jan 23 11:23:06 1998
@@ -217,12 +217,14 @@
 void yy_delete_buffer YY_PROTO(( YY_BUFFER_STATE b ));
 void yy_init_buffer YY_PROTO(( YY_BUFFER_STATE b, FILE *file ));
 
+#if 0
 static int yy_start_stack_ptr = 0;
 static int yy_start_stack_depth = 0;
 static int *yy_start_stack = 0;
 static void yy_push_state YY_PROTO(( int new_state ));
 static void yy_pop_state YY_PROTO(( void ));
 static int yy_top_state YY_PROTO(( void ));
+#endif /* 0 */
 
 static void *yy_flex_alloc YY_PROTO(( unsigned int ));
 static void *yy_flex_realloc YY_PROTO(( void *, unsigned int ));
@@ -1780,6 +1782,7 @@
 	}
 
 
+#if 0
 #ifdef YY_USE_PROTOS
 static void yy_push_state( int new_state )
 #else
@@ -1825,7 +1828,7 @@
 	{
 	return yy_start_stack[yy_start_stack_ptr - 1];
 	}
-
+#endif /* 0 */
 
 #ifdef YY_USE_PROTOS
 static void yy_fatal_error( const char msg[] )
