--- lex.yy.cpp.orig	2009-06-23 13:18:22.000000000 -0500
+++ lex.yy.cpp	2009-06-23 13:49:49.000000000 -0500
@@ -27,7 +27,7 @@
 
 /* C99 systems have <inttypes.h>. Non-C99 systems may or may not. */
 
-#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L || defined (__HP_aCC)
 
 /* C99 says to define __STDC_LIMIT_MACROS before including stdint.h,
  * if you want the limit (max/min) macros for int types.
@@ -35,6 +35,7 @@
 #ifndef __STDC_LIMIT_MACROS
 #define __STDC_LIMIT_MACROS 1
 #endif
+#include <inttypes.h>
 
 typedef int8_t flex_int8_t;
 typedef uint8_t flex_uint8_t;
@@ -1773,7 +1773,7 @@ TAO_YY_RULE_SETUP
                   idl_global->set_lineno(idl_global->lineno() + 1);
                   idl_store_pragma(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 93:
 /* rule 93 can match eol */
 case 94:
@@ -1782,7 +1782,7 @@ TAO_YY_RULE_SETUP
 {/* ignore file */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+         break;
 case 95:
 /* rule 95 can match eol */
 case 96:
@@ -1791,7 +1791,7 @@ TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 97:
 /* rule 97 can match eol */
 case 98:
@@ -1800,7 +1800,7 @@ TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 99:
 /* rule 99 can match eol */
 case 100:
@@ -1809,7 +1809,7 @@ TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 101:
 /* rule 101 can match eol */
 case 102:
@@ -1818,7 +1818,7 @@ TAO_YY_RULE_SETUP
 {
                   idl_parse_line_and_file(ace_tao_yytext);
                 }
-        TAO_YY_BREAK
+         break;
 case 103:
 /* rule 103 can match eol */
 case 104:
@@ -1828,7 +1828,7 @@ TAO_YY_RULE_SETUP
                   /* ignore cpp ident */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+         break;
 case 105:
 /* rule 105 can match eol */
 TAO_YY_RULE_SETUP
@@ -1836,7 +1836,7 @@ TAO_YY_RULE_SETUP
                   /* ignore comments */
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+         break;
 case 106:
 TAO_YY_RULE_SETUP
 {
@@ -1856,18 +1856,18 @@ TAO_YY_RULE_SETUP
                     }
                   }
                 }
-        TAO_YY_BREAK
+        break;
 case 107:
 TAO_YY_RULE_SETUP
 ;
-        TAO_YY_BREAK
+         break;
 case 108:
 /* rule 108 can match eol */
 TAO_YY_RULE_SETUP
 {
                   idl_global->set_lineno(idl_global->lineno() + 1);
                 }
-        TAO_YY_BREAK
+         break;
 case 109:
 TAO_YY_RULE_SETUP
 return ace_tao_yytext [0];
@@ -1875,7 +1875,7 @@ return ace_tao_yytext [0];
 case 110:
 TAO_YY_RULE_SETUP
 TAO_YY_ECHO;
-        TAO_YY_BREAK
+         break;
 case TAO_YY_STATE_EOF(INITIAL):
         tao_yyterminate();
 
