--- PortableServerC.cpp	Thu May  3 11:29:20 2001
+++ PortableServerC.cpp.mod	Wed May  2 19:11:25 2001
@@ -20,13 +20,8 @@
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
 #include "PortableServerC.h"
-
-#include "tao/Stub.h"
-#include "tao/Invocation.h"
-#include "tao/ClientRequestInfo.h"
-#if TAO_HAS_INTERCEPTORS == 1
-#include "tao/RequestInfo_Util.h"
-#endif  /* TAO_HAS_INTERCEPTORS == 1 */
+#include "POA.h"
+#include "tao/ORB.h"
 
 #if defined (__BORLANDC__)
 #pragma option -w-rvl -w-rch -w-ccc -w-aus
@@ -263,6 +258,8 @@
 TAO_NAMESPACE_DEFINE (CORBA::TypeCode_ptr, _tc_ObjectId, &_tc_TAO_tc_PortableServer_ObjectId)
 TAO_NAMESPACE_END
 
+#if (TAO_HAS_MINIMUM_CORBA == 0)
+
 // Default constructor.
 PortableServer::ForwardRequest::ForwardRequest (void)
   : CORBA_UserException ("IDL:PortableServer/ForwardRequest:1.0")
@@ -421,6 +418,8 @@
   return ::PortableServer::_tc_ForwardRequest;
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 TAO_NAMESPACE_TYPE (const CORBA::ULong)
 TAO_NAMESPACE_BEGIN (PortableServer)
 TAO_NAMESPACE_DEFINE (const CORBA::ULong, THREAD_POLICY_ID, 16U)
@@ -449,6 +448,9 @@
 TAO_NAMESPACE_BEGIN (PortableServer)
 TAO_NAMESPACE_DEFINE (const CORBA::ULong, REQUEST_PROCESSING_POLICY_ID, 22U)
 TAO_NAMESPACE_END
+
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ThreadPolicyValue[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -497,49 +499,6 @@
 TAO_NAMESPACE_DEFINE (CORBA::TypeCode_ptr, _tc_ThreadPolicyValue, &_tc_TAO_tc_PortableServer_ThreadPolicyValue)
 TAO_NAMESPACE_END
 
-PortableServer::ThreadPolicy_ptr
-tao_PortableServer_ThreadPolicy_duplicate (
-    PortableServer::ThreadPolicy_ptr p
-  )
-{
-  return PortableServer::ThreadPolicy::_duplicate (p);
-}
-
-void
-tao_PortableServer_ThreadPolicy_release (
-    PortableServer::ThreadPolicy_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::ThreadPolicy_ptr
-tao_PortableServer_ThreadPolicy_nil (
-    void
-  )
-{
-  return PortableServer::ThreadPolicy::_nil ();
-}
-
-PortableServer::ThreadPolicy_ptr
-tao_PortableServer_ThreadPolicy_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::ThreadPolicy::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_ThreadPolicy_upcast (
-    void *src
-  )
-{
-  PortableServer::ThreadPolicy **tmp =
-    ACE_static_cast (PortableServer::ThreadPolicy **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::ThreadPolicy_var
 // *************************************************************
@@ -803,6 +762,8 @@
   return "IDL:PortableServer/ThreadPolicy:1.0";
 }
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 static const CORBA::Long _oc_PortableServer_LifespanPolicyValue[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -848,49 +809,6 @@
 TAO_NAMESPACE_DEFINE (CORBA::TypeCode_ptr, _tc_LifespanPolicyValue, &_tc_TAO_tc_PortableServer_LifespanPolicyValue)
 TAO_NAMESPACE_END
 
-PortableServer::LifespanPolicy_ptr
-tao_PortableServer_LifespanPolicy_duplicate (
-    PortableServer::LifespanPolicy_ptr p
-  )
-{
-  return PortableServer::LifespanPolicy::_duplicate (p);
-}
-
-void
-tao_PortableServer_LifespanPolicy_release (
-    PortableServer::LifespanPolicy_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::LifespanPolicy_ptr
-tao_PortableServer_LifespanPolicy_nil (
-    void
-  )
-{
-  return PortableServer::LifespanPolicy::_nil ();
-}
-
-PortableServer::LifespanPolicy_ptr
-tao_PortableServer_LifespanPolicy_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::LifespanPolicy::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_LifespanPolicy_upcast (
-    void *src
-  )
-{
-  PortableServer::LifespanPolicy **tmp =
-    ACE_static_cast (PortableServer::LifespanPolicy **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::LifespanPolicy_var
 // *************************************************************
@@ -1201,49 +1119,6 @@
 TAO_NAMESPACE_DEFINE (CORBA::TypeCode_ptr, _tc_IdUniquenessPolicyValue, &_tc_TAO_tc_PortableServer_IdUniquenessPolicyValue)
 TAO_NAMESPACE_END
 
-PortableServer::IdUniquenessPolicy_ptr
-tao_PortableServer_IdUniquenessPolicy_duplicate (
-    PortableServer::IdUniquenessPolicy_ptr p
-  )
-{
-  return PortableServer::IdUniquenessPolicy::_duplicate (p);
-}
-
-void
-tao_PortableServer_IdUniquenessPolicy_release (
-    PortableServer::IdUniquenessPolicy_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::IdUniquenessPolicy_ptr
-tao_PortableServer_IdUniquenessPolicy_nil (
-    void
-  )
-{
-  return PortableServer::IdUniquenessPolicy::_nil ();
-}
-
-PortableServer::IdUniquenessPolicy_ptr
-tao_PortableServer_IdUniquenessPolicy_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::IdUniquenessPolicy::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_IdUniquenessPolicy_upcast (
-    void *src
-  )
-{
-  PortableServer::IdUniquenessPolicy **tmp =
-    ACE_static_cast (PortableServer::IdUniquenessPolicy **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::IdUniquenessPolicy_var
 // *************************************************************
@@ -1553,49 +1428,6 @@
 TAO_NAMESPACE_DEFINE (CORBA::TypeCode_ptr, _tc_IdAssignmentPolicyValue, &_tc_TAO_tc_PortableServer_IdAssignmentPolicyValue)
 TAO_NAMESPACE_END
 
-PortableServer::IdAssignmentPolicy_ptr
-tao_PortableServer_IdAssignmentPolicy_duplicate (
-    PortableServer::IdAssignmentPolicy_ptr p
-  )
-{
-  return PortableServer::IdAssignmentPolicy::_duplicate (p);
-}
-
-void
-tao_PortableServer_IdAssignmentPolicy_release (
-    PortableServer::IdAssignmentPolicy_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::IdAssignmentPolicy_ptr
-tao_PortableServer_IdAssignmentPolicy_nil (
-    void
-  )
-{
-  return PortableServer::IdAssignmentPolicy::_nil ();
-}
-
-PortableServer::IdAssignmentPolicy_ptr
-tao_PortableServer_IdAssignmentPolicy_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::IdAssignmentPolicy::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_IdAssignmentPolicy_upcast (
-    void *src
-  )
-{
-  PortableServer::IdAssignmentPolicy **tmp =
-    ACE_static_cast (PortableServer::IdAssignmentPolicy **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::IdAssignmentPolicy_var
 // *************************************************************
@@ -1859,6 +1691,8 @@
   return "IDL:PortableServer/IdAssignmentPolicy:1.0";
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ImplicitActivationPolicyValue[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -1915,49 +1749,6 @@
 TAO_NAMESPACE_DEFINE (CORBA::TypeCode_ptr, _tc_ImplicitActivationPolicyValue, &_tc_TAO_tc_PortableServer_ImplicitActivationPolicyValue)
 TAO_NAMESPACE_END
 
-PortableServer::ImplicitActivationPolicy_ptr
-tao_PortableServer_ImplicitActivationPolicy_duplicate (
-    PortableServer::ImplicitActivationPolicy_ptr p
-  )
-{
-  return PortableServer::ImplicitActivationPolicy::_duplicate (p);
-}
-
-void
-tao_PortableServer_ImplicitActivationPolicy_release (
-    PortableServer::ImplicitActivationPolicy_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::ImplicitActivationPolicy_ptr
-tao_PortableServer_ImplicitActivationPolicy_nil (
-    void
-  )
-{
-  return PortableServer::ImplicitActivationPolicy::_nil ();
-}
-
-PortableServer::ImplicitActivationPolicy_ptr
-tao_PortableServer_ImplicitActivationPolicy_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::ImplicitActivationPolicy::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_ImplicitActivationPolicy_upcast (
-    void *src
-  )
-{
-  PortableServer::ImplicitActivationPolicy **tmp =
-    ACE_static_cast (PortableServer::ImplicitActivationPolicy **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::ImplicitActivationPolicy_var
 // *************************************************************
@@ -2269,49 +2060,6 @@
 TAO_NAMESPACE_DEFINE (CORBA::TypeCode_ptr, _tc_ServantRetentionPolicyValue, &_tc_TAO_tc_PortableServer_ServantRetentionPolicyValue)
 TAO_NAMESPACE_END
 
-PortableServer::ServantRetentionPolicy_ptr
-tao_PortableServer_ServantRetentionPolicy_duplicate (
-    PortableServer::ServantRetentionPolicy_ptr p
-  )
-{
-  return PortableServer::ServantRetentionPolicy::_duplicate (p);
-}
-
-void
-tao_PortableServer_ServantRetentionPolicy_release (
-    PortableServer::ServantRetentionPolicy_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::ServantRetentionPolicy_ptr
-tao_PortableServer_ServantRetentionPolicy_nil (
-    void
-  )
-{
-  return PortableServer::ServantRetentionPolicy::_nil ();
-}
-
-PortableServer::ServantRetentionPolicy_ptr
-tao_PortableServer_ServantRetentionPolicy_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::ServantRetentionPolicy::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_ServantRetentionPolicy_upcast (
-    void *src
-  )
-{
-  PortableServer::ServantRetentionPolicy **tmp =
-    ACE_static_cast (PortableServer::ServantRetentionPolicy **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::ServantRetentionPolicy_var
 // *************************************************************
@@ -2637,49 +2385,6 @@
 TAO_NAMESPACE_DEFINE (CORBA::TypeCode_ptr, _tc_RequestProcessingPolicyValue, &_tc_TAO_tc_PortableServer_RequestProcessingPolicyValue)
 TAO_NAMESPACE_END
 
-PortableServer::RequestProcessingPolicy_ptr
-tao_PortableServer_RequestProcessingPolicy_duplicate (
-    PortableServer::RequestProcessingPolicy_ptr p
-  )
-{
-  return PortableServer::RequestProcessingPolicy::_duplicate (p);
-}
-
-void
-tao_PortableServer_RequestProcessingPolicy_release (
-    PortableServer::RequestProcessingPolicy_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::RequestProcessingPolicy_ptr
-tao_PortableServer_RequestProcessingPolicy_nil (
-    void
-  )
-{
-  return PortableServer::RequestProcessingPolicy::_nil ();
-}
-
-PortableServer::RequestProcessingPolicy_ptr
-tao_PortableServer_RequestProcessingPolicy_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::RequestProcessingPolicy::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_RequestProcessingPolicy_upcast (
-    void *src
-  )
-{
-  PortableServer::RequestProcessingPolicy **tmp =
-    ACE_static_cast (PortableServer::RequestProcessingPolicy **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::RequestProcessingPolicy_var
 // *************************************************************
@@ -2943,48 +2648,7 @@
   return "IDL:PortableServer/RequestProcessingPolicy:1.0";
 }
 
-PortableServer::POAManager_ptr
-tao_PortableServer_POAManager_duplicate (
-    PortableServer::POAManager_ptr p
-  )
-{
-  return PortableServer::POAManager::_duplicate (p);
-}
-
-void
-tao_PortableServer_POAManager_release (
-    PortableServer::POAManager_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::POAManager_ptr
-tao_PortableServer_POAManager_nil (
-    void
-  )
-{
-  return PortableServer::POAManager::_nil ();
-}
-
-PortableServer::POAManager_ptr
-tao_PortableServer_POAManager_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::POAManager::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_POAManager_upcast (
-    void *src
-  )
-{
-  PortableServer::POAManager **tmp =
-    ACE_static_cast (PortableServer::POAManager **, src);
-  return *tmp;
-}
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
 
 // *************************************************************
 // Operations for class PortableServer::POAManager_var
@@ -3305,48 +2969,7 @@
   return retval;
 }
 
-PortableServer::AdapterActivator_ptr
-tao_PortableServer_AdapterActivator_duplicate (
-    PortableServer::AdapterActivator_ptr p
-  )
-{
-  return PortableServer::AdapterActivator::_duplicate (p);
-}
-
-void
-tao_PortableServer_AdapterActivator_release (
-    PortableServer::AdapterActivator_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::AdapterActivator_ptr
-tao_PortableServer_AdapterActivator_nil (
-    void
-  )
-{
-  return PortableServer::AdapterActivator::_nil ();
-}
-
-PortableServer::AdapterActivator_ptr
-tao_PortableServer_AdapterActivator_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::AdapterActivator::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_AdapterActivator_upcast (
-    void *src
-  )
-{
-  PortableServer::AdapterActivator **tmp =
-    ACE_static_cast (PortableServer::AdapterActivator **, src);
-  return *tmp;
-}
+#if (TAO_HAS_MINIMUM_POA == 0)
 
 // *************************************************************
 // Operations for class PortableServer::AdapterActivator_var
@@ -3599,49 +3222,6 @@
   return "IDL:PortableServer/AdapterActivator:1.0";
 }
 
-PortableServer::ServantManager_ptr
-tao_PortableServer_ServantManager_duplicate (
-    PortableServer::ServantManager_ptr p
-  )
-{
-  return PortableServer::ServantManager::_duplicate (p);
-}
-
-void
-tao_PortableServer_ServantManager_release (
-    PortableServer::ServantManager_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::ServantManager_ptr
-tao_PortableServer_ServantManager_nil (
-    void
-  )
-{
-  return PortableServer::ServantManager::_nil ();
-}
-
-PortableServer::ServantManager_ptr
-tao_PortableServer_ServantManager_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::ServantManager::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_ServantManager_upcast (
-    void *src
-  )
-{
-  PortableServer::ServantManager **tmp =
-    ACE_static_cast (PortableServer::ServantManager **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::ServantManager_var
 // *************************************************************
@@ -3893,49 +3473,6 @@
   return "IDL:PortableServer/ServantManager:1.0";
 }
 
-PortableServer::ServantActivator_ptr
-tao_PortableServer_ServantActivator_duplicate (
-    PortableServer::ServantActivator_ptr p
-  )
-{
-  return PortableServer::ServantActivator::_duplicate (p);
-}
-
-void
-tao_PortableServer_ServantActivator_release (
-    PortableServer::ServantActivator_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::ServantActivator_ptr
-tao_PortableServer_ServantActivator_nil (
-    void
-  )
-{
-  return PortableServer::ServantActivator::_nil ();
-}
-
-PortableServer::ServantActivator_ptr
-tao_PortableServer_ServantActivator_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::ServantActivator::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_ServantActivator_upcast (
-    void *src
-  )
-{
-  PortableServer::ServantActivator **tmp =
-    ACE_static_cast (PortableServer::ServantActivator **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::ServantActivator_var
 // *************************************************************
@@ -4199,49 +3736,6 @@
   return "IDL:PortableServer/ServantActivator:1.0";
 }
 
-PortableServer::ServantLocator_ptr
-tao_PortableServer_ServantLocator_duplicate (
-    PortableServer::ServantLocator_ptr p
-  )
-{
-  return PortableServer::ServantLocator::_duplicate (p);
-}
-
-void
-tao_PortableServer_ServantLocator_release (
-    PortableServer::ServantLocator_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::ServantLocator_ptr
-tao_PortableServer_ServantLocator_nil (
-    void
-  )
-{
-  return PortableServer::ServantLocator::_nil ();
-}
-
-PortableServer::ServantLocator_ptr
-tao_PortableServer_ServantLocator_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::ServantLocator::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_ServantLocator_upcast (
-    void *src
-  )
-{
-  PortableServer::ServantLocator **tmp =
-    ACE_static_cast (PortableServer::ServantLocator **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::ServantLocator_var
 // *************************************************************
@@ -4505,48 +3999,7 @@
   return "IDL:PortableServer/ServantLocator:1.0";
 }
 
-PortableServer::POA_ptr
-tao_PortableServer_POA_duplicate (
-    PortableServer::POA_ptr p
-  )
-{
-  return PortableServer::POA::_duplicate (p);
-}
-
-void
-tao_PortableServer_POA_release (
-    PortableServer::POA_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::POA_ptr
-tao_PortableServer_POA_nil (
-    void
-  )
-{
-  return PortableServer::POA::_nil ();
-}
-
-PortableServer::POA_ptr
-tao_PortableServer_POA_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::POA::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_POA_upcast (
-    void *src
-  )
-{
-  PortableServer::POA **tmp =
-    ACE_static_cast (PortableServer::POA **, src);
-  return *tmp;
-}
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
 
 // *************************************************************
 // Operations for class PortableServer::POA_var
@@ -5013,6 +4466,8 @@
   this->index = _tao_index;
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Default constructor.
 PortableServer::POA::NoServant::NoServant (void)
   : CORBA_UserException ("IDL:PortableServer/POA/NoServant:1.0")
@@ -5081,6 +4536,8 @@
   return retval;
 }
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 // Default constructor.
 PortableServer::POA::ObjectAlreadyActive::ObjectAlreadyActive (void)
   : CORBA_UserException ("IDL:PortableServer/POA/ObjectAlreadyActive:1.0")
@@ -5489,49 +4946,6 @@
   return retval;
 }
 
-PortableServer::Current_ptr
-tao_PortableServer_Current_duplicate (
-    PortableServer::Current_ptr p
-  )
-{
-  return PortableServer::Current::_duplicate (p);
-}
-
-void
-tao_PortableServer_Current_release (
-    PortableServer::Current_ptr p
-  )
-{
-  CORBA::release (p);
-}
-
-PortableServer::Current_ptr
-tao_PortableServer_Current_nil (
-    void
-  )
-{
-  return PortableServer::Current::_nil ();
-}
-
-PortableServer::Current_ptr
-tao_PortableServer_Current_narrow (
-    CORBA::Object *p,
-    CORBA::Environment &ACE_TRY_ENV
-  )
-{
-  return PortableServer::Current::_narrow (p, ACE_TRY_ENV);
-}
-
-CORBA::Object *
-tao_PortableServer_Current_upcast (
-    void *src
-  )
-{
-  PortableServer::Current **tmp =
-    ACE_static_cast (PortableServer::Current **, src);
-  return *tmp;
-}
-
 // *************************************************************
 // Operations for class PortableServer::Current_var
 // *************************************************************
@@ -5863,6 +5277,30 @@
   return retval;
 }
 
+char *
+PortableServer::ObjectId_to_string (const PortableServer::ObjectId &id)
+{
+  return TAO_POA::ObjectId_to_string (id);
+}
+
+CORBA::WChar *
+PortableServer::ObjectId_to_wstring (const PortableServer::ObjectId &id)
+{
+  return TAO_POA::ObjectId_to_wstring (id);
+}
+
+PortableServer::ObjectId *
+PortableServer::string_to_ObjectId (const char *id)
+{
+  return TAO_POA::string_to_ObjectId (id);
+}
+
+PortableServer::ObjectId *
+PortableServer::wstring_to_ObjectId (const CORBA::WChar *id)
+{
+  return TAO_POA::wstring_to_ObjectId (id);
+}
+
 void operator<<= (
     CORBA::Any &_tao_any,
     const PortableServer::ObjectId &_tao_elem
@@ -5954,6 +5392,8 @@
   return 0;
 }
 
+#if (TAO_HAS_MINIMUM_CORBA == 0)
+
 void operator<<= (CORBA::Any &_tao_any, const PortableServer::ForwardRequest &_tao_elem) // copying
 {
   TAO_OutputCDR stream;
@@ -6044,6 +5484,10 @@
   return 0;
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 void operator<<= (CORBA::Any &_tao_any, PortableServer::ThreadPolicyValue _tao_elem)
 {
   TAO_OutputCDR stream;
@@ -6091,6 +5535,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::ThreadPolicy,PortableServer::ThreadPolicy_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 void operator<<= (CORBA::Any &_tao_any, PortableServer::LifespanPolicyValue _tao_elem)
 {
   TAO_OutputCDR stream;
@@ -6232,6 +5678,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::IdAssignmentPolicy,PortableServer::IdAssignmentPolicy_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 void operator<<= (CORBA::Any &_tao_any, PortableServer::ImplicitActivationPolicyValue _tao_elem)
 {
   TAO_OutputCDR stream;
@@ -6373,6 +5821,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::RequestProcessingPolicy,PortableServer::RequestProcessingPolicy_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 #if defined (ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION)  || \
     defined (ACE_HAS_GNU_REPO)
     template class TAO_Object_Manager<PortableServer::POAManager,PortableServer::POAManager_var>;
@@ -6380,6 +5830,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::POAManager,PortableServer::POAManager_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 #if defined (ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION)  || \
     defined (ACE_HAS_GNU_REPO)
     template class TAO_Object_Manager<PortableServer::AdapterActivator,PortableServer::AdapterActivator_var>;
@@ -6408,6 +5860,8 @@
   #  pragma instantiate TAO_Object_Manager<PortableServer::ServantLocator,PortableServer::ServantLocator_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 #if defined (ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION)  || \
     defined (ACE_HAS_GNU_REPO)
     template class TAO_Object_Manager<PortableServer::POA,PortableServer::POA_var>;
@@ -6462,11 +5916,6 @@
     // If length is 0 we return true.
     if (0 >= _tao_seq_len) 
       return 1;
-    // Add a check to the length of the sequence
-    // to make sure it does not exceed the length
-    // of the stream. (See bug 58.)
-    if (_tao_seq_len > strm.length())
-      return 0;
     // retrieve all the elements
     
 #if (TAO_NO_COPY_OCTET_SEQUENCES == 1)
