--- Svc_Conf_y.cpp.orig	Thu Oct 30 21:08:18 1997
+++ Svc_Conf_y.cpp	Thu Oct 30 21:09:35 1997
@@ -1,3 +1,5 @@
+// $Id$
+
 #ifndef lint
 char ace_yysccsid[] = "@(#)yaccpar	1.4 (Berkeley) 02/25/90 \n\
  Modified 5/2/90 by J. Roskind to support graphic debugging modes";
@@ -12,9 +14,9 @@
 static ACE_Module_Type *get_module (ACE_Static_Node *str_rec, ACE_Static_Node *svc_type);
 static ACE_Module_Type *get_module (ACE_Static_Node *str_rec, const char *svc_name);
 
-#define YYDEBUG_LEXER_TEXT (ace_yytext[ace_yyleng] = '\0', ace_yytext) 
+#define YYDEBUG_LEXER_TEXT (ace_yytext[ace_yyleng] = '\0', ace_yytext)
 /* Force the pretty debugging code to compile.*/
-#define YYDEBUG 1 
+#define YYDEBUG 1
 
 /* Efficient memory allocation technique.*/
 ACE_Obstack *ace_obstack;
@@ -250,7 +252,7 @@
 void
 ace_yyerror (char *s)
 {
-  ACE_ERROR ((LM_ERROR, "[error %d] on line %d: %s\n", 
+  ACE_ERROR ((LM_ERROR, "[error %d] on line %d: %s\n",
 	      ++ace_yyerrno, ace_yylineno, s));
 }
 
@@ -309,9 +311,9 @@
 }
 
 ACE_Service_Type_Impl *
-ace_create_service_type (const char *name, 
-			 int type, 
-			 const void *symbol, 
+ace_create_service_type (const char *name,
+			 int type,
+			 const void *symbol,
 			 u_int flags)
 {
   ACE_Service_Type_Impl *stp = 0;
@@ -324,17 +326,17 @@
     {
     case ACE_SVC_OBJ_T:
       ACE_NEW_RETURN (stp,
-		      ACE_Service_Object_Type ((ACE_Service_Object *) symbol, name, flags), 
+		      ACE_Service_Object_Type ((ACE_Service_Object *) symbol, name, flags),
 		      0);
       break;
     case ACE_MODULE_T:
       ACE_NEW_RETURN (stp,
-		      ACE_Module_Type (symbol, name, flags), 
+		      ACE_Module_Type (symbol, name, flags),
 		      0);
       break;
     case ACE_STREAM_T:
       ACE_NEW_RETURN (stp,
-		      ACE_Stream_Type (symbol, name, flags), 
+		      ACE_Stream_Type (symbol, name, flags),
 		      0);
       break;
     default:
@@ -352,11 +354,11 @@
 // Name given on the command-line to envoke the program.
 char *program_name;
 
-// Main driver program. 
+// Main driver program.
 
-int 
+int
 main (int argc, char *argv[])
-{ 
+{
   ace_yyin = stdin;
   ace_obstack = new ACE_Obstack;
 
@@ -395,7 +397,7 @@
 #endif /* YYDEBUG_INDENT */
 #ifndef YYDEBUG_REDUCE
 #ifdef __cplusplus
-void YYDEBUG_REDUCE(int ace_yynew_state, int ace_yyrule_num, char *ace_yyrule_string, int ace_yynew_indent, int ace_yyrhs_count)
+void YYDEBUG_REDUCE(int /* ace_yynew_state */, int /* ace_yyrule_num */, char *ace_yyrule_string, int ace_yynew_indent, int ace_yyrhs_count)
 #else
 YYDEBUG_REDUCE(ace_yynew_state, ace_yyrule_num, ace_yyrule_string, ace_yynew_indent, ace_yyrhs_count)
 int ace_yynew_state;
@@ -425,7 +427,7 @@
 #endif /* YYDEBUG_REDUCE */
 #ifndef YYDEBUG_SHIFT_LEXEME
 #ifdef __cplusplus
-void YYDEBUG_SHIFT_LEXEME(int ace_yyold_state, int ace_yynew_state, char *ace_yytoken_string, int ace_yynew_indent)
+void YYDEBUG_SHIFT_LEXEME(int /* ace_yyold_state */, int /* ace_yynew_state */, char *ace_yytoken_string, int ace_yynew_indent)
 #else
 YYDEBUG_SHIFT_LEXEME(ace_yyold_state, ace_yynew_state, ace_yytoken_string, ace_yynew_indent)
 int ace_yyold_state;
@@ -440,7 +442,7 @@
 #endif /*  YYDEBUG_SHIFT_LEXEME */
 #ifndef YYDEBUG_LOOK_AHEAD
 #ifdef __cplusplus
-void YYDEBUG_LOOK_AHEAD(int ace_yynew_state, int ace_yytoken_num, char *ace_yytoken_string, int ace_yyindent)
+void YYDEBUG_LOOK_AHEAD(int /* ace_yynew_state */, int ace_yytoken_num, char *ace_yytoken_string, int ace_yyindent)
 #else
 YYDEBUG_LOOK_AHEAD(ace_yynew_state, ace_yytoken_num, ace_yytoken_string, ace_yyindent)
 int ace_yynew_state;
@@ -457,7 +459,7 @@
 #endif /* YYDEBUG_LOOK_AHEAD */
 #ifndef YYDEBUG_DISCARD_STATE
 #ifdef __cplusplus
-void YYDEBUG_DISCARD_STATE(int ace_yynew_state, int ace_yyindent)
+void YYDEBUG_DISCARD_STATE(int /* ace_yynew_state */, int ace_yyindent)
 #else
 YYDEBUG_DISCARD_STATE(ace_yynew_state, ace_yyindent)
 int ace_yynew_state;
@@ -485,7 +487,7 @@
 #endif /* YYDEBUG_DISCARD_STATE */
 #ifndef YYDEBUG_DISCARD_TOKEN
 #ifdef __cplusplus
-void YYDEBUG_DISCARD_TOKEN(int ace_yynew_state, int ace_yytoken_num, char *ace_yytoken_string, int ace_yyindent)
+void YYDEBUG_DISCARD_TOKEN(int /* ace_yynew_state */, int /* ace_yytoken_num */, char *ace_yytoken_string, int ace_yyindent)
 #else
 YYDEBUG_DISCARD_TOKEN(ace_yynew_state, ace_yytoken_num, ace_yytoken_string, ace_yyindent)
 int ace_yynew_state;
@@ -500,7 +502,7 @@
 #endif /* YYDEBUG_DISCARD_TOKEN */
 #ifndef YYDEBUG_SHIFT_ERROR_LEXEME
 #ifdef __cplusplus
-void YYDEBUG_SHIFT_ERROR_LEXEME(int ace_yyold_state, int ace_yynew_state, int ace_yyindent)
+void YYDEBUG_SHIFT_ERROR_LEXEME(int /* ace_yyold_state */, int /* ace_yynew_state */, int ace_yyindent)
 #else
 YYDEBUG_SHIFT_ERROR_LEXEME(ace_yyold_state, ace_yynew_state, ace_yyindent)
 int ace_yyold_state;
@@ -526,7 +528,7 @@
     extern char *ace_foo();
 #endif
 
-    if (ace_yys = ACE_OS::getenv("YYDEBUG"))
+    if ((ace_yys = ACE_OS::getenv("YYDEBUG")))
     {
         ace_yyn = *ace_yys;
         if (ace_yyn >= '0' && ace_yyn <= '9')
@@ -543,7 +545,7 @@
     *ace_yyssp = ace_yystate = 0;
 
 ace_yyloop:
-    if (ace_yyn = ace_yydefred[ace_yystate]) goto ace_yyreduce;
+    if ((ace_yyn = ace_yydefred[ace_yystate])) goto ace_yyreduce;
     if (ace_yychar < 0)
     {
         if ((ace_yychar = ace_yylex()) < 0) ace_yychar = 0;
@@ -653,7 +655,7 @@
             if (5 < ace_yydebug)
                 printf("ace_yydebug: state %d, error recovery discards token %d (%s)\n",
                     ace_yystate, ace_yychar, ace_yys);
-            else 
+            else
                 YYDEBUG_DISCARD_TOKEN(ace_yystate, ace_yychar, ace_yys, ace_yyssp-ace_yyss);
         }
 #endif
@@ -680,18 +682,18 @@
       {
 	ace_yyvsp[0].parse_node_->apply (); delete ace_yyvsp[0].parse_node_;
       }
-      ace_obstack->release (); 
+      ace_obstack->release ();
     }
 break;
 case 2:
 #line 46 "Svc_Conf.y"
-{ 
-      ace_obstack->release (); 
+{
+      ace_obstack->release ();
     }
 break;
 case 10:
 #line 63 "Svc_Conf.y"
-{ 
+{
       if (ace_yyvsp[-1].svc_record_ != 0)
 	ace_yyval.parse_node_ = new ACE_Dynamic_Node (ace_yyvsp[-1].svc_record_, ace_yyvsp[0].ident_);
       else
@@ -700,25 +702,25 @@
 break;
 case 11:
 #line 73 "Svc_Conf.y"
-{ 
+{
       ace_yyval.parse_node_ = new ACE_Static_Node (ace_yyvsp[-1].ident_, ace_yyvsp[0].ident_);
     }
 break;
 case 12:
 #line 80 "Svc_Conf.y"
-{ 
+{
       ace_yyval.parse_node_ = new ACE_Suspend_Node (ace_yyvsp[0].ident_);
     }
 break;
 case 13:
 #line 87 "Svc_Conf.y"
-{ 
+{
       ace_yyval.parse_node_ = new ACE_Resume_Node (ace_yyvsp[0].ident_);
     }
 break;
 case 14:
 #line 94 "Svc_Conf.y"
-{ 
+{
       ace_yyval.parse_node_ = new ACE_Remove_Node (ace_yyvsp[0].ident_);
     }
 break;
@@ -750,9 +752,9 @@
 break;
 case 20:
 #line 121 "Svc_Conf.y"
-{ 
+{
       /* Initialize left context...*/
-      ace_yyval.static_node_ = ace_yyvsp[-1].static_node_; 
+      ace_yyval.static_node_ = ace_yyvsp[-1].static_node_;
     }
 break;
 case 21:
@@ -790,7 +792,7 @@
 break;
 case 26:
 #line 152 "Svc_Conf.y"
-{ 
+{
       ACE_Module_Type *mt = get_module (ace_yyvsp[-2].static_node_, ace_yyvsp[0].static_node_->name ());
 
       if (((ACE_Stream_Type *) (ace_yyvsp[-2].static_node_)->record ()->type ())->push (mt) == -1)
@@ -799,7 +801,7 @@
 break;
 case 27:
 #line 159 "Svc_Conf.y"
-{ 
+{
       ACE_Module_Type *mt = get_module (ace_yyvsp[-2].static_node_, ace_yyvsp[0].static_node_->name ());
       if (mt != 0)
 	mt->suspend ();
@@ -815,9 +817,9 @@
 break;
 case 29:
 #line 171 "Svc_Conf.y"
-{ 
+{
       ACE_Module_Type *mt = get_module (ace_yyvsp[-2].static_node_, ace_yyvsp[0].static_node_->name ());
-      if (mt != 0 
+      if (mt != 0
 	  && ((ACE_Stream_Type *) (ace_yyvsp[-2].static_node_)->record ()->type ())->remove (mt) == -1)
 	{
 	  ACE_ERROR ((LM_ERROR, "cannot remove Module_Type %s from STREAM_Type %s\n",
@@ -829,8 +831,8 @@
 case 30:
 #line 185 "Svc_Conf.y"
 {
-      u_int flags 
-	= ACE_Service_Type::DELETE_THIS 
+      u_int flags
+	= ACE_Service_Type::DELETE_THIS
 	| (ace_yyvsp[-1].location_node_->dispose () == 0 ? 0 : ACE_Service_Type::DELETE_OBJ);
       const void *sym = ace_yyvsp[-1].location_node_->symbol ();
 
