--- InterceptorC.cpp.prediff	Wed May  3 13:02:33 2000
+++ InterceptorC.cpp	Wed May  3 13:03:00 2000
@@ -9,6 +9,8 @@
 
 #include "InterceptorC.h"
 
+#if (TAO_HAS_INTERCEPTORS == 1)
+
 #if !defined (__ACE_INLINE__)
 #include "InterceptorC.i"
 #endif /* !defined INLINE */
@@ -431,3 +433,4 @@
   #  pragma instantiate TAO_Object_Manager<PortableInterceptor::ClientRequestInterceptor,PortableInterceptor::ClientRequestInterceptor_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_INTERCEPTORS == 1 */
