--- orig/OctetSeqC.h	2003-04-25 22:07:24.000000000 -0500
+++ OctetSeqC.h	2003-04-25 21:48:43.000000000 -0500
@@ -27,20 +27,17 @@
 
 // TAO_IDL - Generated from
 // be/be_codegen.cpp:151
-
 #ifndef _TAO_IDL_ORIG_OCTETSEQC_H_
 #define _TAO_IDL_ORIG_OCTETSEQC_H_
-
 #include "ace/pre.h"
-
-#include "tao/corba.h"
+#include "tao/Sequence.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
-
-#include "tao/corbafwd.h"
+#include "tao/Seq_Var_T.h"
+#include "tao/Seq_Out_T.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
@@ -70,22 +67,22 @@
 
 TAO_NAMESPACE  CORBA
 {
-  
+
   // TAO_IDL - Generated from
   // be/be_visitor_sequence/sequence_ch.cpp:89
 
 #if !defined (_CORBA_OCTETSEQ_CH_)
 #define _CORBA_OCTETSEQ_CH_
-  
+
   class OctetSeq;
-  
+
   typedef
     TAO_FixedSeq_Var_T<
         OctetSeq,
         CORBA::Octet
       >
     OctetSeq_var;
-  
+
   typedef
     TAO_Seq_Out_T<
         OctetSeq,
@@ -93,7 +90,7 @@
         CORBA::Octet
       >
     OctetSeq_out;
-  
+
   class TAO_Export OctetSeq
     : public
         TAO_Unbounded_Sequence<
@@ -106,14 +103,14 @@
     OctetSeq (
         CORBA::ULong max,
         CORBA::ULong length,
-        CORBA::Octet* buffer, 
+        CORBA::Octet* buffer,
         CORBA::Boolean release = 0
       );
     OctetSeq (const OctetSeq &);
     ~OctetSeq (void);
-    
+
     static void _tao_any_destructor (void *);
-    
+
     typedef OctetSeq_var _var_type;
 
 #if (TAO_NO_COPY_OCTET_SEQUENCES == 1)
@@ -126,10 +123,10 @@
   };
 
 #endif /* end #if !defined */
-  
+
   // TAO_IDL - Generated from
   // be/be_visitor_typecode/typecode_decl.cpp:44
-  
+
   TAO_NAMESPACE_STORAGE_CLASS ::CORBA::TypeCode_ptr _tc_OctetSeq;
 
 // TAO_IDL - Generated from
@@ -190,4 +187,3 @@
 
 #include "ace/post.h"
 #endif /* ifndef */
-
