--- orig/DomainC.h	Mon Feb  4 17:27:57 2002
+++ DomainC.h	Fri Feb  8 10:50:20 2002
@@ -31,7 +31,7 @@
 
 #include "TAO_Export.h"
 #include "PolicyC.h"
-#include "InterfaceDefC.h"
+#include "Sequence.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
--- orig/DomainC.cpp	Mon Feb  4 17:27:57 2002
+++ DomainC.cpp	Fri Feb  8 10:50:20 2002
@@ -31,6 +31,9 @@
 #include "tao/ClientInterceptorAdapter.h"
 #endif  /* TAO_HAS_INTERCEPTORS == 1 */
 
+#include "tao/IFR_Client_Adapter.h"
+#include "ace/Dynamic_Service.h"
+
 #if defined (__BORLANDC__)
 #pragma option -w-rvl -w-rch -w-ccc -w-aus -w-sig
 #endif /* __BORLANDC__ */
@@ -1099,7 +1102,15 @@
   parameter_list->length (2);
   CORBA::ULong len = 0;
   
-    (*parameter_list)[len].argument <<=  this->object_type_;
+  TAO_IFR_Client_Adapter *adapter =
+    ACE_Dynamic_Service<TAO_IFR_Client_Adapter>::instance (
+        TAO_ORB_Core::ifr_client_adapter_name ()
+      );
+
+  adapter->interfacedef_any_insert (
+      (*parameter_list)[len].argument,
+      this->object_type_
+    );
   (*parameter_list)[len].mode = CORBA::PARAM_IN;
   len++;
     (*parameter_list)[len].argument <<= CORBA::Any::from_boolean (this->constr_policy_);
@@ -1186,6 +1197,11 @@
   
 #endif  /* TAO_HAS_INTERCEPTORS */
   
+  TAO_IFR_Client_Adapter *adapter =
+    ACE_Dynamic_Service<TAO_IFR_Client_Adapter>::instance (
+        TAO_ORB_Core::ifr_client_adapter_name ()
+      );
+
   for (;;)
     {
       _invoke_status = TAO_INVOKE_EXCEPTION;
@@ -1234,7 +1250,7 @@
 
           TAO_OutputCDR &_tao_out = _tao_call.out_stream ();
           if (!(
-                            (_tao_out << object_type) &&
+              (adapter->interfacedef_cdr_insert (_tao_out, object_type)) &&
               (_tao_out << CORBA::Any::from_boolean (constr_policy))
             ))
             TAO_INTERCEPTOR_THROW (
--- orig/DomainS.h	Mon Feb  4 17:27:57 2002
+++ Domain/DomainS.h	Fri Feb  8 10:53:14 2002
@@ -23,9 +23,10 @@
 #define _TAO_IDL_ORIG_DOMAINS_H_
 
 #include "ace/pre.h"
-#include "PolicyS.h"
-#include "InterfaceDefS.h"
-#include "DomainC.h"
+
+#include "domain_export.h"
+#include "tao/PortableServer/PolicyS.h"
+#include "tao/DomainC.h"
 
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
@@ -58,7 +59,7 @@
   class _TAO_DomainManager_Direct_Proxy_Impl;
   class _TAO_DomainManager_Strategized_Proxy_Broker;
   
-  class TAO_Export DomainManager
+  class TAO_Domain_Export DomainManager
     : public virtual PortableServer::ServantBase
   {
   protected:
@@ -134,7 +135,7 @@
   //               Strategized Proxy Broker Declaration 
   //
   
-  class TAO_Export _TAO_DomainManager_Strategized_Proxy_Broker : public virtual ::CORBA::_TAO_DomainManager_Proxy_Broker
+  class TAO_Domain_Export _TAO_DomainManager_Strategized_Proxy_Broker : public virtual ::CORBA::_TAO_DomainManager_Proxy_Broker
   {
   public: 
     _TAO_DomainManager_Strategized_Proxy_Broker (void);
@@ -179,7 +180,7 @@
   //                    ThruPOA  Impl. Declaration
   //
   
-  class TAO_Export _TAO_DomainManager_ThruPOA_Proxy_Impl : 
+  class TAO_Domain_Export _TAO_DomainManager_ThruPOA_Proxy_Impl :
     public virtual ::CORBA::_TAO_DomainManager_Proxy_Impl,
     public virtual TAO_ThruPOA_Object_Proxy_Impl
   {
@@ -209,7 +210,7 @@
 //                    Direct  Impl. Declaration
 //
 
-class TAO_Export _TAO_DomainManager_Direct_Proxy_Impl : 
+class TAO_Domain_Export _TAO_DomainManager_Direct_Proxy_Impl :
   public virtual CORBA::_TAO_DomainManager_Proxy_Impl,
   public virtual TAO_Direct_Object_Proxy_Impl
 {
@@ -242,7 +243,7 @@
 class _TAO_ConstructionPolicy_Direct_Proxy_Impl;
 class _TAO_ConstructionPolicy_Strategized_Proxy_Broker;
 
-class TAO_Export ConstructionPolicy
+class TAO_Domain_Export ConstructionPolicy
   : public virtual POA_CORBA::Policy
 {
 protected:
@@ -340,7 +341,7 @@
 //               Strategized Proxy Broker Declaration 
 //
 
-class TAO_Export _TAO_ConstructionPolicy_Strategized_Proxy_Broker : public virtual ::CORBA::_TAO_ConstructionPolicy_Proxy_Broker
+class TAO_Domain_Export _TAO_ConstructionPolicy_Strategized_Proxy_Broker : public virtual ::CORBA::_TAO_ConstructionPolicy_Proxy_Broker
 {
 public: 
   _TAO_ConstructionPolicy_Strategized_Proxy_Broker (void);
@@ -385,7 +386,7 @@
 //                    ThruPOA  Impl. Declaration
 //
 
-class TAO_Export _TAO_ConstructionPolicy_ThruPOA_Proxy_Impl : 
+class TAO_Domain_Export _TAO_ConstructionPolicy_ThruPOA_Proxy_Impl :
   public virtual ::CORBA::_TAO_ConstructionPolicy_Proxy_Impl,
   public virtual TAO_ThruPOA_Object_Proxy_Impl,
   public virtual ::POA_CORBA::_TAO_Policy_ThruPOA_Proxy_Impl
@@ -418,7 +419,7 @@
 //                    Direct  Impl. Declaration
 //
 
-class TAO_Export _TAO_ConstructionPolicy_Direct_Proxy_Impl : 
+class TAO_Domain_Export _TAO_ConstructionPolicy_Direct_Proxy_Impl :
   public virtual CORBA::_TAO_ConstructionPolicy_Proxy_Impl,
   public virtual TAO_Direct_Object_Proxy_Impl,
   public virtual POA_CORBA::_TAO_Policy_Direct_Proxy_Impl
--- orig/DomainS.cpp	Mon Feb  4 17:27:57 2002
+++ Domain/DomainS.cpp	Fri Feb  8 11:17:22 2002
@@ -39,7 +39,9 @@
 #include "tao/PortableServer/ServerInterceptorAdapter.h"
 #endif  /* TAO_HAS_INTERCEPTORS == 1 */
 
+#include "tao/IFR_Client/IFR_ExtendedC.h"
 #include "ace/Dynamic_Service.h"
+
 #if defined (__BORLANDC__)
 #pragma option -w-rvl -w-rch -w-ccc -w-aus
 #endif /* __BORLANDC__ */
