--- orig/MessagingS_T.i	Sat Mar 31 14:51:18 2001
+++ PortableServer/MessagingS_T.i	Sat Mar 31 15:31:16 2001
@@ -20,6 +20,7 @@
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
 
+#if (TAO_HAS_AMI_CALLBACK == 1)
 #if defined (ACE_HAS_USING_KEYWORD)
 template <class T> ACE_INLINE
 POA_Messaging::ReplyHandler_tie<T>::ReplyHandler_tie (T &t)
@@ -99,3 +100,4 @@
 }
 
 #endif /* ACE_HAS_USING_KEYWORD */
+#endif /* TAO_HAS_AMI_CALLBACK == 1 */
