--- orig/ObjectReferenceTemplateC.h	Thu Feb  7 15:04:56 2002
+++ ObjectReferenceTemplateC.h	Fri Feb  8 11:23:31 2002
@@ -23,7 +23,7 @@
 #define _TAO_IDL_ORIG_OBJECTREFERENCETEMPLATEC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+#include "tao/corbafwd.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
@@ -32,6 +32,8 @@
 #include "TAO_Export.h"
 #include "OctetSeqC.h"
 #include "StringSeqC.h"
+#include "ValueBase.h"
+#include "Exception.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
--- orig/ObjectReferenceTemplateC.cpp	Thu Feb  7 15:04:56 2002
+++ ObjectReferenceTemplateC.cpp	Fri Feb  8 11:23:01 2002
@@ -24,6 +24,7 @@
 #include "tao/Stub.h"
 #include "tao/Invocation.h"
 #include "tao/PortableInterceptor.h"
+#include "tao/ValueFactory.h"
 
 #if TAO_HAS_INTERCEPTORS == 1
 #include "tao/RequestInfo_Util.h"
