--- orig/Policy_ForwardC.h	Wed Feb  6 19:23:32 2002
+++ Policy_ForwardC.h	Wed Feb  6 19:23:58 2002
@@ -23,13 +23,14 @@
 #define _TAO_IDL_ORIG_POLICY_FORWARDC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+#include "tao/corbafwd.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 #include "TAO_Export.h"
+#include "Sequence.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
