--- orig/PollableC.i	Mon Apr 16 22:13:53 2001
+++ PollableC.i	Mon Apr 16 22:31:02 2001
@@ -1,4 +1,4 @@
-// -*- C++ -*-
+// -*- C++ -*- $Id$ */
 //
 // $Id$
 
@@ -126,7 +126,6 @@
 // *************************************************************
 // Inline operations for exception CORBA_PollableSet::UnknownPollable
 // *************************************************************
-
 
 #if !defined (_CORBA_POLLABLESET___CI_)
 #define _CORBA_POLLABLESET___CI_
