--- orig/PollableC.cpp	Mon Apr 16 22:13:53 2001
+++ PollableC.cpp	Mon Apr 16 22:41:17 2001
@@ -1,4 +1,4 @@
-// -*- C++ -*-
+// -*- C++ -*- $Id$ */
 //
 // $Id$
 
@@ -21,21 +21,18 @@
 
 #include "PollableC.h"
 
+#if (TAO_HAS_AMI_POLLER == 1)
+
 #include "tao/Stub.h"
 #include "tao/Invocation.h"
-#include "tao/ClientRequestInfo.h"
-#if TAO_HAS_INTERCEPTORS == 1
-#include "tao/RequestInfo_Util.h"
-#endif  /* TAO_HAS_INTERCEPTORS == 1 */
-
-#if defined (__BORLANDC__)
-#pragma option -w-rvl -w-rch -w-ccc -w-aus
-#endif /* __BORLANDC__ */
+#include "tao/Any.h"
 
 #if !defined (__ACE_INLINE__)
 #include "PollableC.i"
 #endif /* !defined INLINE */
 
+ACE_RCSID(tao, PollableC, "$Id$")
+
 // *************************************************************
 // Operations for class CORBA_Pollable_var
 // *************************************************************
@@ -283,7 +280,7 @@
 
 const char* CORBA_Pollable::_interface_repository_id (void) const
 {
-  return "IDL:omg.org/CORBA_Pollable:1.0";
+  return "IDL:omg.org/CORBA/Pollable:1.0";
 }
 
 // *************************************************************
@@ -545,7 +542,7 @@
 
 const char* CORBA_DIIPollable::_interface_repository_id (void) const
 {
-  return "IDL:omg.org/CORBA_DIIPollable:1.0";
+  return "IDL:omg.org/CORBA/DIIPollable:1.0";
 }
 
 // *************************************************************
@@ -837,11 +834,6 @@
     }
 }
 
-void CORBA_PollableSet::NoPossiblePollable::_raise ()
-{
-  TAO_RAISE (*this);
-}
-
 void CORBA_PollableSet::NoPossiblePollable::_tao_encode (
     TAO_OutputCDR &,
     CORBA::Environment &ACE_TRY_ENV
@@ -858,7 +850,7 @@
   ACE_THROW (CORBA::MARSHAL ());
 }
 
-// TAO extension - the _alloc method.
+// TAO extension - the _alloc method
 CORBA::Exception *CORBA_PollableSet::NoPossiblePollable::_alloc (void)
 {
   CORBA::Exception *retval = 0;
@@ -912,12 +904,12 @@
 
 void CORBA_PollableSet::UnknownPollable::_tao_encode (
     TAO_OutputCDR &,
-    CORBA::Environment &ACE_TRY_ENV
-  ) const
+    CORBA::Environment &ACE_TRY_ENV) const
 {
   ACE_THROW (CORBA::MARSHAL ());
 }
 
+
 void CORBA_PollableSet::UnknownPollable::_tao_decode (
     TAO_InputCDR &,
     CORBA::Environment &ACE_TRY_ENV
@@ -955,3 +947,4 @@
   #  pragma instantiate TAO_Object_Manager<CORBA_PollableSet,CORBA_PollableSet_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_AMI_POLLER == 1 */
