--- PortableInterceptorC.h.old	Mon Oct 23 01:32:44 2000
+++ PortableInterceptorC.h	Mon Oct 23 01:34:21 2000
@@ -23,13 +23,12 @@
 #define _TAO_IDL_PORTABLEINTERCEPTORC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+#include "tao/corbafwd.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
-#include "tao/corbafwd.h"
 #include "DynamicC.h"
 #include "MessagingC.h"
 
@@ -152,8 +151,7 @@
       }
 
     virtual char * name (
-                CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        TAO_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -792,8 +790,7 @@
 
     virtual IOP::TaggedComponent * get_effective_component (
         IOP::ComponentId id,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -801,8 +798,7 @@
 
     virtual CORBA::Policy_ptr get_request_policy (
         CORBA::PolicyType type,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -811,8 +807,7 @@
     virtual void add_request_service_context (
         const IOP::ServiceContext & service_context,
         CORBA::Boolean replace,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -931,32 +926,28 @@
       }
 
     virtual CORBA::Any * sending_exception (
-                CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
 
     virtual CORBA::OctetSeq * object_id (
-                CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
 
     virtual CORBA::OctetSeq * adapter_id (
-                CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
 
     virtual char * target_most_derived_interface (
-                CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -964,8 +955,7 @@
 
     virtual CORBA::Policy_ptr get_server_policy (
         CORBA::PolicyType type,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -974,9 +964,8 @@
     virtual void set_slot (
         PortableInterceptor::SlotId id,
         const CORBA::Any & data,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
-      )
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
+        )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
         PortableInterceptor::InvalidSlot
@@ -984,8 +973,7 @@
 
     virtual CORBA::Boolean target_is_a (
         const char * id,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -994,9 +982,8 @@
     virtual void add_reply_service_context (
         const IOP::ServiceContext & service_context,
         CORBA::Boolean replace,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
-      )
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
+        )
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
@@ -1114,47 +1101,42 @@
       }
 
     virtual void send_request (
-        PortableInterceptor::ClientRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
-      )
+        PortableInterceptor::ClientRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
+        )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
         PortableInterceptor::ForwardRequest
       )) = 0;
 
     virtual void send_poll (
-        PortableInterceptor::ClientRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        PortableInterceptor::ClientRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS        
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
 
     virtual void receive_reply (
-        PortableInterceptor::ClientRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
-      )
+        PortableInterceptor::ClientRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
+        )
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
 
     virtual void receive_exception (
-        PortableInterceptor::ClientRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
-      )
+        PortableInterceptor::ClientRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
+        )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
         PortableInterceptor::ForwardRequest
       )) = 0;
 
     virtual void receive_other (
-        PortableInterceptor::ClientRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        PortableInterceptor::ClientRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
@@ -1274,9 +1256,8 @@
       }
 
     virtual void receive_request_service_contexts (
-        PortableInterceptor::ServerRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        PortableInterceptor::ServerRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
@@ -1284,38 +1265,34 @@
       )) = 0;
 
     virtual void receive_request (
-        PortableInterceptor::ServerRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
-      )
+        PortableInterceptor::ServerRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
+        )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
         PortableInterceptor::ForwardRequest
       )) = 0;
 
     virtual void send_reply (
-        PortableInterceptor::ServerRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
-      )
+        PortableInterceptor::ServerRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
+        )
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
 
     virtual void send_exception (
-        PortableInterceptor::ServerRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
-      )
+        PortableInterceptor::ServerRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
+        )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
         PortableInterceptor::ForwardRequest
       )) = 0;
 
     virtual void send_other (
-        PortableInterceptor::ServerRequestInfo_ptr ri,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        PortableInterceptor::ServerRequestInfo_ptr ri
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS        
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
@@ -1436,8 +1413,7 @@
 
     virtual CORBA::Policy_ptr get_effective_policy (
         CORBA::PolicyType type,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -1445,8 +1421,7 @@
 
     virtual void add_ior_component (
         const IOP::TaggedComponent & component,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -1455,8 +1430,7 @@
     virtual void add_ior_component_to_profile (
         const IOP::TaggedComponent & component,
         IOP::ProfileId profile_id,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        CORBA::Environment &ACE_TRY_ENV = TAO_default_environment ()
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -1575,9 +1549,8 @@
       }
 
     virtual void establish_components (
-        PortableInterceptor::IORInfo_ptr info,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        PortableInterceptor::IORInfo_ptr info
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -1697,9 +1670,8 @@
 
     virtual CORBA::Policy_ptr create_policy (
         CORBA::PolicyType type,
-        const CORBA::Any & value,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        const CORBA::Any & value
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
@@ -2105,18 +2077,16 @@
       }
 
     virtual void pre_init (
-        PortableInterceptor::ORBInitInfo_ptr info,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        PortableInterceptor::ORBInitInfo_ptr info
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
 
     virtual void post_init (
-        PortableInterceptor::ORBInitInfo_ptr info,
-        CORBA::Environment &ACE_TRY_ENV = 
-          TAO_default_environment ()
+        PortableInterceptor::ORBInitInfo_ptr info
+        TAO_ENV_ARG_DECL_WITH_DEFAULTS
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException
@@ -2137,6 +2107,11 @@
 
 #endif /* end #if !defined */
 
+ TAO_NAMESPACE_STORAGE_CLASS void register_orb_initializer (
+   ORBInitializer_ptr init,
+   CORBA::Environment & = TAO_default_environment ());
+ ///< Register an ORBInitializer with the global ORBInitializer
+ ///< table.
 
 }
 TAO_NAMESPACE_CLOSE // module PortableInterceptor
--- PortableInterceptorC.cpp.old	Mon Oct 23 01:32:44 2000
+++ PortableInterceptorC.cpp	Mon Oct 23 01:34:33 2000
@@ -25,6 +25,7 @@
 #include "PortableInterceptorC.i"
 #endif /* !defined INLINE */
 
+#include "tao/CORBA_String.h"
 
 // default constructor
 PortableInterceptor::Interceptor::Interceptor (void)
