--- orig/RTCORBAS.cpp	Wed Mar 28 20:40:59 2001
+++ RTCORBAS.cpp	Wed Mar 28 20:42:23 2001
@@ -22,6 +22,10 @@
 #ifndef _TAO_IDL_ORIG_RTCORBAS_CPP_
 #define _TAO_IDL_ORIG_RTCORBAS_CPP_
 
+#include "tao/orbconf.h"
+
+#if (TAO_HAS_RT_CORBA == 1)
+
 #include "RTCORBAS.h"
 #include "tao/PortableServer/Object_Adapter.h"
 #include "tao/PortableServer/Operation_Table.h"
@@ -43,3 +47,5 @@
 
 
 #endif /* ifndef */
+
+#endif /* (TAO_HAS_RT_CORBA == 1) */
