--- orig/Typecode_typesC.h	2005-04-08 10:17:37.123060800 +0200
+++ Typecode_typesC.h	2005-04-08 10:17:41.739699200 +0200
@@ -68,26 +68,6 @@
   // TAO_IDL - Generated from
   // be\be_interface.cpp:598
 
-#if !defined (_CORBA_TYPECODE__VAR_OUT_CH_)
-#define _CORBA_TYPECODE__VAR_OUT_CH_
-  
-  class TypeCode;
-  typedef TypeCode *TypeCode_ptr;
-  
-  typedef
-    TAO_Objref_Var_T<
-        TypeCode
-      >
-    TypeCode_var;
-  
-  typedef
-    TAO_Objref_Out_T<
-        TypeCode
-      >
-    TypeCode_out;
-
-#endif /* end #if !defined */
-  
   // TAO_IDL - Generated from
   // be\be_visitor_enum/enum_ch.cpp:57
   
@@ -130,6 +110,11 @@
     tk_component,
     tk_home,
     tk_event
+
+    // This symbol is not defined by CORBA 3.0.  It's used to speed up
+    // dispatch based on TCKind values, and lets many important ones
+    // just be table lookups.  It must always be the last enum value!!
+    , TAO_TC_KIND_COUNT
   };
   
   typedef TCKind &TCKind_out;
@@ -148,13 +133,6 @@
 }
 
 // TAO_IDL - Generated from
-// be\be_visitor_interface_fwd/cdr_op_ch.cpp:66
-
-TAO_Export CORBA::Boolean operator<< (TAO_OutputCDR &, const CORBA::TypeCode_ptr );
-TAO_Export CORBA::Boolean operator>> (TAO_InputCDR &, CORBA::TypeCode_ptr &);
-
-
-// TAO_IDL - Generated from
 // be\be_visitor_enum/cdr_op_ch.cpp:50
 
 TAO_Export CORBA::Boolean operator<< (TAO_OutputCDR &, CORBA::TCKind);
