--- IFR_Extended.pidl.old	Fri Jun  1 23:02:10 2001
+++ IFR_Extended.pidl	Fri Jun  1 22:58:48 2001
@@ -71,123 +71,115 @@
 
 module CORBA
 {
-  typedef short ValueModifier; // PIDL
-  const ValueModifier VM_NONE = 0;
-  const ValueModifier VM_CUSTOM = 1;
-  const ValueModifier VM_ABSTRACT = 2;
-  const ValueModifier VM_TRUNCATABLE = 3;
-
   typedef short Visibility;
-  const Visibility PRIVATE_MEMBER = 0;
-  const Visibility PUBLIC_MEMBER = 1;
+};
 
-  interface FixedDef : IDLType 
+  interface CORBA_FixedDef : CORBA_IDLType 
   {
     attribute unsigned short digits;
     attribute short scale;
   };
 
-  struct ValueMember 
+  struct CORBA_ValueMember 
   {
-    Identifier name;
-    RepositoryId id;
-    RepositoryId defined_in;
-    VersionSpec version;
-    TypeCode type;
-    IDLType type_def;
-    Visibility access;
+    string name;
+    string id;
+    string defined_in;
+    string version;
+    CORBA::TypeCode type;
+    CORBA_IDLType type_def;
+    CORBA::Visibility access;
   };
 
-  typedef sequence <ValueMember> ValueMemberSeq;
+  typedef sequence <CORBA_ValueMember> CORBA_ValueMemberSeq;
 
-  interface ValueMemberDef : Contained 
+  interface CORBA_ValueMemberDef : CORBA_Contained 
   {
-    readonly attribute TypeCode type;
-    attribute IDLType type_def;
-    attribute Visibility access;
+    readonly attribute CORBA::TypeCode type;
+    attribute CORBA_IDLType type_def;
+    attribute CORBA::Visibility access;
   };
 
-  interface ValueDef : Container, Contained, IDLType 
+  interface CORBA_ValueDef : CORBA_Container, CORBA_Contained, CORBA_IDLType 
   {
     // read/write interface
-    attribute InterfaceDefSeq supported_interfaces;
-    attribute InitializerSeq initializers;
-    attribute ValueDef base_value;
-    attribute ValueDefSeq abstract_base_values;
+    attribute CORBA_InterfaceDefSeq supported_interfaces;
+    attribute CORBA_InitializerSeq initializers;
+    attribute CORBA_ValueDef base_value;
+    attribute CORBA_ValueDefSeq abstract_base_values;
     // read interface
     attribute boolean is_abstract;
     attribute boolean is_custom;
     attribute boolean is_truncatable;
     boolean is_a (
-        in RepositoryId id
+        in string id
       );
 
     struct FullValueDescription 
     {
-      Identifier name;
-      RepositoryId id;
+      string name;
+      string id;
       boolean is_abstract;
       boolean is_custom;
-      RepositoryId defined_in;
-      VersionSpec version;
-      OpDescriptionSeq operations;
-      AttrDescriptionSeq attributes;
-      ValueMemberSeq members;
-      InitializerSeq initializers;
-      RepositoryIdSeq supported_interfaces;
-      RepositoryIdSeq abstract_base_values;
+      string defined_in;
+      string version;
+      CORBA_OpDescriptionSeq operations;
+      CORBA_AttrDescriptionSeq attributes;
+      CORBA_ValueMemberSeq members;
+      CORBA_InitializerSeq initializers;
+      CORBA_RepositoryIdSeq supported_interfaces;
+      CORBA_RepositoryIdSeq abstract_base_values;
       boolean is_truncatable;
-      RepositoryId base_value;
-      TypeCode type;
+      string base_value;
+      CORBA::TypeCode type;
     };
 
     FullValueDescription describe_value ();
       
-    ValueMemberDef create_value_member  (
-        in RepositoryId id,
-        in Identifier name,
-        in VersionSpec version,
-        in IDLType type,
-        in Visibility access
+    CORBA_ValueMemberDef create_value_member  (
+        in string id,
+        in string name,
+        in string version,
+        in CORBA_IDLType type,
+        in CORBA::Visibility access
       );
-    AttributeDef create_attribute (
-        in RepositoryId id,
-        in Identifier name,
-        in VersionSpec version,
-        in IDLType type,
-        in AttributeMode mode
+    CORBA_AttributeDef create_attribute (
+        in string id,
+        in string name,
+        in string version,
+        in CORBA_IDLType type,
+        in CORBA::AttributeMode mode
       );
-    OperationDef create_operation (
-        in RepositoryId id,
-        in Identifier name,
-        in VersionSpec version,
-        in IDLType result,
-        in OperationMode mode,
-        in ParDescriptionSeq params,
-        in ExceptionDefSeq exceptions,
-        in ContextIdSeq contexts
+    CORBA_OperationDef create_operation (
+        in string id,
+        in string name,
+        in string version,
+        in CORBA_IDLType result,
+        in CORBA::OperationMode mode,
+        in CORBA_ParDescriptionSeq params,
+        in CORBA_ExceptionDefSeq exceptions,
+        in CORBA_ContextIdSeq contexts
       );
   };
 
-  struct ValueDescription 
+  struct CORBA_ValueDescription 
   {
-    Identifier name;
-    RepositoryId id;
+    string name;
+    string id;
     boolean is_abstract;
     boolean is_custom;
-    RepositoryId defined_in;
-    VersionSpec version;
-    RepositoryIdSeq supported_interfaces;
-    RepositoryIdSeq abstract_base_values;
+    string defined_in;
+    string version;
+    CORBA_RepositoryIdSeq supported_interfaces;
+    CORBA_RepositoryIdSeq abstract_base_values;
     boolean is_truncatable;
-    RepositoryId base_value;
+    string base_value;
   };
 
-  interface ValueBoxDef : TypedefDef 
+  interface CORBA_ValueBoxDef : CORBA_TypedefDef 
   {
-    attribute IDLType original_type_def;
+    attribute CORBA_IDLType original_type_def;
   };
-};
 
 #pragma prefix ""
 
