--- orig/MessagingC.h	2005-02-14 10:51:00.842680000 +0100
+++ MessagingC.h	2005-02-14 10:51:01.093040000 +0100
@@ -36,6 +36,10 @@
 
 #include "ace/config-all.h"
 
+#ifndef TAO_MESSAGING_SAFE_INCLUDE
+#error "You should not include MessagingC.h directly, use Messaging.h"
+#endif /* !TAO_MESSAGING_SAFE_INCLUDE */
+
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
--- orig/MessagingC.cpp	2005-02-14 10:51:00.842680000 +0100
+++ MessagingC.cpp	2005-02-14 10:51:01.103054400 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "MessagingC.h"
+#include "Messaging.h"
 #include "tao/CDR.h"
 #include "tao/Object_T.h"
 #include "tao/ORB_Core.h"
--- orig/MessagingA.cpp	2005-02-14 10:51:00.842680000 +0100
+++ MessagingA.cpp	2005-02-14 10:51:01.123083200 +0100
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "MessagingC.h"
+#include "Messaging.h"
 #include "tao/Typecode.h"
 #include "tao/CDR.h"
 #include "tao/Any.h"
--- orig/MessagingS.h	2005-02-14 10:51:00.842680000 +0100
+++ MessagingS.h	2005-02-14 10:52:20.537275200 +0100
@@ -34,9 +34,7 @@
 #include /**/ "ace/pre.h"
 
 #include "tao/PollableS.h"
-#include "tao/Messaging/Messaging_SyncScope_PolicyS.h"
-#include "tao/Messaging/Messaging_RT_PolicyS.h"
-#include "tao/Messaging/Messaging_No_ImplS.h"
+#include "tao/Messaging/Messaging.h"
 #include "tao/OctetSeqS.h"
 #include "MessagingC.h"
 
