--- orig/MessagingC.h	Tue Sep  3 16:30:43 2002
+++ MessagingC.h	Tue Sep  3 16:15:16 2002
@@ -26,8 +26,12 @@
 #ifndef _TAO_IDL_ORIG_MESSAGINGC_H_
 #define _TAO_IDL_ORIG_MESSAGINGC_H_
 
+#ifndef TAO_MESSAGING_SAFE_INCLUDE
+#error "You should not include MessagingC.h directly, use Messaging.h"
+#endif /* !TAO_MESSAGING_SAFE_INCLUDE */
+
 #include "ace/pre.h"
-#include "tao/corba.h"
+#include "tao/corbafwd.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
@@ -37,6 +41,8 @@
 #include "Messaging_SyncScope_PolicyC.h"
 #include "Messaging_RT_PolicyC.h"
 #include "Messaging_No_ImplC.h"
+#include "tao/ValueBase.h"
+#include "tao/ValueFactory.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
--- orig/MessagingC.cpp	Tue Sep  3 16:30:43 2002
+++ MessagingC.cpp	Tue Sep  3 16:19:50 2002
@@ -23,7 +23,7 @@
 // TAO_IDL - Generated from
 // be/be_codegen.cpp:315
 
-#include "MessagingC.h"
+#include "Messaging.h"
 
 #include "tao/Stub.h"
 #include "tao/Invocation.h"
--- orig/MessagingS.h	Tue Sep  3 16:30:43 2002
+++ MessagingS.h	Tue Sep  3 16:21:38 2002
@@ -27,10 +27,7 @@
 #define _TAO_IDL_ORIG_MESSAGINGS_H_
 
 #include "ace/pre.h"
-#include "Messaging_SyncScope_PolicyS.h"
-#include "Messaging_RT_PolicyS.h"
-#include "Messaging_No_ImplS.h"
-#include "MessagingC.h"
+#include "Messaging.h"
 
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
