--- orig/MessagingC.h	2005-03-29 14:10:15.848518400 +0200
+++ MessagingC.h	2005-03-29 14:10:19.884321600 +0200
@@ -40,6 +40,10 @@
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
+#ifndef TAO_MESSAGING_SAFE_INCLUDE
+#error "You should not include MessagingC.h directly, use Messaging.h"
+#endif /* !TAO_MESSAGING_SAFE_INCLUDE */
+
 #include "messaging_export.h"
 #include "tao/Valuetype/ValueBase.h"
 #include "tao/Valuetype/Valuetype_Adapter_Impl.h"
--- orig/MessagingC.cpp	2005-03-29 14:10:15.848518400 +0200
+++ MessagingC.cpp	2005-03-29 14:10:19.884321600 +0200
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "MessagingC.h"
+#include "Messaging.h"
 #include "tao/CDR.h"
 #include "tao/Object_T.h"
 #include "tao/ORB_Core.h"
--- orig/MessagingA.cpp	2005-03-29 14:10:15.848518400 +0200
+++ MessagingA.cpp	2005-03-29 14:11:22.284048000 +0200
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "MessagingC.h"
+#include "Messaging.h"
 #include "tao/Null_RefCount_Policy.h"
 #include "tao/TypeCode_Constants.h"
 #include "tao/Alias_TypeCode.h"
--- orig/MessagingS.h	2005-03-29 14:10:15.848518400 +0200
+++ MessagingS.h	2005-03-29 14:10:19.954422400 +0200
@@ -33,11 +33,8 @@
 
 #include /**/ "ace/pre.h"
 
-#include "MessagingC.h"
+#include "Messaging.h"
 #include "tao/PollableS.h"
-#include "tao/Messaging/Messaging_SyncScope_PolicyS.h"
-#include "tao/Messaging/Messaging_RT_PolicyS.h"
-#include "tao/Messaging/Messaging_No_ImplS.h"
 #include "tao/OctetSeqS.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
