--- orig/DomainS.h	2005-02-22 13:41:14.148656000 +0100
+++ DomainS.h	2005-02-22 13:42:55.244024000 +0100
@@ -33,17 +33,17 @@
 
 #include /**/ "ace/pre.h"
 
-#include "DomainC.h"
-#include "tao/PolicyS.h"
-#include "tao/InterfaceDefS.h"
+#include "tao/DomainC.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
+#include "tao/PortableServer/PolicyS.h"
 #include "tao/Collocation_Proxy_Broker.h"
 #include "tao/PortableServer/PortableServer.h"
 #include "tao/PortableServer/Servant_Base.h"
+#include "tao/IFR_Client/IFR_ExtendedC.h"
 
 #if defined(_MSC_VER)
 #pragma warning(push)
