--- orig/RTCORBAC.h	2005-02-18 12:15:48.103436800 +0100
+++ RTCORBAC.h	2005-02-18 12:15:48.443926400 +0100
@@ -36,12 +36,18 @@
 
 #include "ace/config-all.h"
 
+#ifndef TAO_RTCORBA_SAFE_INCLUDE
+#error "You should not include RTCORBAC.h directly, use RTCORBA.h"
+#endif /* !TAO_RTCORBA_SAFE_INCLUDE */
+
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 #include "rtcorba_export.h"
+#include "rtcorba_typedefs.h"
 #include "tao/ORB.h"
+#include "tao/TimeBaseC.h"
 #include "tao/SystemException.h"
 #include "tao/Environment.h"
 #include "tao/Object.h"
@@ -50,7 +56,10 @@
 #include "tao/Seq_Var_T.h"
 #include "tao/Seq_Out_T.h"
 #include "tao/VarOut_T.h"
+#include "tao/Encodable.h"
+#include "tao/PolicyC.h"
 
+#include "tao/IOPC.h"
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
@@ -530,6 +539,7 @@
   
   class TAO_RTCORBA_Export ProtocolProperties
     : public virtual CORBA::Object
+    , public virtual TAO_Encodable
   {
   public:
     typedef ProtocolProperties_ptr _ptr_type;
@@ -566,6 +576,9 @@
     virtual const char* _interface_repository_id (void) const;
     virtual CORBA::Boolean marshal (TAO_OutputCDR &cdr);
   
+    // CDR Encoder/Decoder methods. (handcrafted)
+    virtual CORBA::Boolean _tao_encode (TAO_OutputCDR &out_cdr);
+    virtual CORBA::Boolean _tao_decode (TAO_InputCDR &in_cdr);
   protected:
     // Abstract or local interface only.
     ProtocolProperties (void);
--- orig/RTCORBAC.cpp	2005-02-18 12:15:48.103436800 +0100
+++ RTCORBAC.cpp	2005-02-18 12:15:48.453940800 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "RTCORBAC.h"
+#include "RTCORBA.h"
 #include "tao/CDR.h"
 #include "tao/Typecode.h"
 #include "tao/Basic_Arguments.h"
@@ -53,6 +53,22 @@
 {
 }
 
+//////////////////////////////////////////////////////
+// CDR Encoding Support hand crafted
+
+CORBA::Boolean
+RTCORBA::ProtocolProperties::_tao_encode (TAO_OutputCDR &)
+{
+  return 1;
+}
+// TAO_IDL - Generated from
+// be/be_visitor_arg_traits.cpp:60
+
+CORBA::Boolean
+RTCORBA::ProtocolProperties::_tao_decode (TAO_InputCDR &)
+{
+  return 1;
+}
 
 // TAO_IDL - Generated from 
 // be\be_visitor_structure/structure_cs.cpp:66
--- orig/RTCORBAA.cpp	2005-02-18 12:15:48.103436800 +0100
+++ RTCORBAA.cpp	2005-02-18 12:15:48.473969600 +0100
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "RTCORBAC.h"
+#include "RTCORBA.h"
 #include "tao/Typecode.h"
 #include "tao/CDR.h"
 #include "tao/Any.h"
