--- y.tab.cpp	Tue Dec 19 09:31:07 2000
+++ y.tab.cpp.mod	Tue Dec 19 09:28:09 2000
@@ -960,7 +960,7 @@
 #ifdef TAO_YYMAXDEPTH
 #define TAO_YYSTACKSIZE TAO_YYMAXDEPTH
 #else
-#define TAO_YYSTACKSIZE 300
+#define TAO_YYSTACKSIZE 10000
 #endif
 #endif
 int tao_yydebug;
@@ -1134,7 +1134,7 @@
     extern char *ace_foo();
 #endif
 
-    if (tao_yys = ACE_OS::getenv("TAO_YYDEBUG"))
+    if ((tao_yys = ACE_OS::getenv("TAO_YYDEBUG")))
     {
         tao_yyn = *tao_yys;
         if (tao_yyn >= '0' && tao_yyn <= '9')
@@ -1151,7 +1151,7 @@
     *tao_yyssp = tao_yystate = 0;
 
 tao_yyloop:
-    if (tao_yyn = tao_yydefred[tao_yystate]) goto tao_yyreduce;
+    if ((tao_yyn = tao_yydefred[tao_yystate])) goto tao_yyreduce;
     if (tao_yychar < 0)
     {
         if ((tao_yychar = tao_yylex()) < 0) tao_yychar = 0;
@@ -1220,11 +1220,10 @@
     goto tao_yynewerror;
 #endif
 
-    tao_yyerror("syntax error");
+//    tao_yyerror("syntax error");
 #ifdef lint
     goto tao_yyerrlab;
 #endif
-tao_yyerrlab:
     ++tao_yynerrs;
 tao_yyinrecovery:
     if (tao_yyerrflag < 3)
