--- orig/ServantLocatorC.h	2005-04-11 16:44:02.053817600 +0200
+++ ServantLocatorC.h	2005-04-11 16:44:04.727662400 +0200
@@ -69,6 +69,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:48
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace PortableServer
 {
   
@@ -105,6 +107,8 @@
     : public virtual ::PortableServer::ServantManager
   {
   public:
+    typedef void *Cookie;
+
     typedef ServantLocator_ptr _ptr_type;
     typedef ServantLocator_var _var_type;
     
@@ -232,6 +236,8 @@
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ServantLocator_ptr *); // non-copying
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ServantLocator_ptr &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:955
 
--- orig/ServantLocatorC.cpp	2005-04-11 16:44:02.053817600 +0200
+++ ServantLocatorC.cpp	2005-04-11 16:50:01.380504000 +0200
@@ -41,6 +41,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -191,3 +193,5 @@
 {
   return false;
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantLocatorA.cpp	2005-04-11 16:44:02.053817600 +0200
+++ ServantLocatorA.cpp	2005-04-11 16:49:56.843980800 +0200
@@ -35,6 +35,8 @@
 #include "tao/Any.h"
 #include "tao/Any_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/objref_typecode.cpp:76
 
@@ -121,3 +123,5 @@
         _tao_elem
       );
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
