--- orig/ServantActivatorC.h	2005-04-11 16:44:01.853529600 +0200
+++ ServantActivatorC.h	2005-04-11 16:44:04.597475200 +0200
@@ -68,6 +68,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:48
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace PortableServer
 {
   
@@ -229,6 +231,8 @@
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ServantActivator_ptr *); // non-copying
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ServantActivator_ptr &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:955
 
--- orig/ServantActivatorC.cpp	2005-04-11 16:44:01.853529600 +0200
+++ ServantActivatorC.cpp	2005-04-11 16:49:49.753785600 +0200
@@ -41,6 +41,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -191,3 +193,5 @@
 {
   return false;
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantActivatorA.cpp	2005-04-11 16:44:01.853529600 +0200
+++ ServantActivatorA.cpp	2005-04-11 16:49:46.378932800 +0200
@@ -34,6 +34,8 @@
 #include "tao/Any.h"
 #include "tao/Any_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/objref_typecode.cpp:76
 
@@ -120,3 +122,5 @@
         _tao_elem
       );
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
