--- orig/DynamicAnyC.h	2004-11-24 08:54:01.902620800 +0100
+++ DynamicAnyC.h	2004-11-24 08:57:51.632956800 +0100
@@ -34,6 +34,10 @@
 #include /**/ "ace/pre.h"
 
 
+#ifndef TAO_DYNAMICANY_SAFE_INCLUDE
+#error "You should not include DynamicAnyC.h directly, use DynamicAny.h"
+#endif /* !TAO_DYNAMICANY_SAFE_INCLUDE */
+
 #include "ace/config-all.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
@@ -41,7 +45,6 @@
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 #include "dynamicany_export.h"
-#include "tao/Valuetype/ValueBase.h"
 #include "tao/ORB.h"
 #include "tao/SystemException.h"
 #include "tao/Environment.h"
--- orig/DynamicAnyC.cpp	2004-11-24 08:54:01.902620800 +0100
+++ DynamicAnyC.cpp	2004-11-24 08:58:10.009380800 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:301
 
 
-#include "DynamicAnyC.h"
+#include "DynamicAny.h"
 #include "tao/CDR.h"
 #include "ace/OS_NS_string.h"
 
