--- orig/ForwardRequestC.h	2005-04-11 16:44:02.454393600 +0200
+++ ForwardRequestC.h	2005-04-11 16:44:04.937964800 +0200
@@ -59,6 +59,8 @@
 #pragma option push -w-rvl -w-rch -w-ccc -w-inl
 #endif /* __BORLANDC__ */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:48
 
@@ -153,6 +155,8 @@
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:955
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 #if defined(_MSC_VER)
 #pragma warning(pop)
 #endif /* _MSC_VER */
--- orig/ForwardRequestC.cpp	2005-04-11 16:44:02.454393600 +0200
+++ ForwardRequestC.cpp	2005-04-11 16:44:51.324665600 +0200
@@ -38,6 +38,8 @@
 #pragma option -w-rvl -w-rch -w-ccc -w-aus -w-sig
 #endif /* __BORLANDC__ */
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
@@ -202,3 +204,5 @@
     (strm >> _tao_aggregate.forward_reference.out ())
   );
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ForwardRequestA.cpp	2005-04-11 16:44:02.454393600 +0200
+++ ForwardRequestA.cpp	2005-04-11 16:44:44.885406400 +0200
@@ -62,6 +62,8 @@
 
 
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/any_op_cs.cpp:50
 
@@ -146,3 +148,5 @@
         _tao_elem
       );
 }
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
