--- orig/IOPC.h	Tue Feb  5 14:56:23 2002
+++ IOPC.h	Tue Feb  5 14:57:05 2002
@@ -23,13 +23,15 @@
 #define _TAO_IDL_ORIG_IOPC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+#include "tao/corbafwd.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 #include "TAO_Export.h"
+#include "Exception.h"
+#include "Object.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
