*** Component_BaseC.cpp	Sun Jun 30 16:16:30 2002
--- Component_BaseC.cpp.mod	Sun Jun 30 16:15:32 2002
***************
*** 50826,50849 ****
  
  // accessor to set the member
  void
! OBV_Components::ConfigValue::value (CORBA::Any val) // set
  {
!   ACE_NEW (
!       this->_pd_value,
!       CORBA::Any (val)
!     );
  }
  // retrieve the member
! const CORBA::Any
  OBV_Components::ConfigValue::value (void) const
  {
!   return *this->_pd_value;
  }
  // retrieve the member
! CORBA::Any
  OBV_Components::ConfigValue::value (void)
  {
!   return *this->_pd_value;
  }
  
  CORBA::Boolean 
--- 50826,50846 ----
  
  // accessor to set the member
  void
! OBV_Components::ConfigValue::value (CORBA::Any &val) // set
  {
!   this->_pd_value = val;
  }
  // retrieve the member
! const CORBA::Any &
  OBV_Components::ConfigValue::value (void) const
  {
!   return this->_pd_value;
  }
  // retrieve the member
! CORBA::Any &
  OBV_Components::ConfigValue::value (void)
  {
!   return this->_pd_value;
  }
  
  CORBA::Boolean
***************
*** 65403,65406 ****
    ACE_ENDTRY;
    return 0;
  }
- 
--- 65400,65402 ----
