*** Component_BaseC.cpp	Sat Jun 29 22:56:16 2002
--- Component_BaseC.cpp.mod	Sat Jun 29 22:47:10 2002
***************
*** 50826,50849 ****
  
  // accessor to set the member
  void
! OBV_Components::ConfigValue::value (CORBA::Any val) // set
  {
!   ACE_NEW (
!       this->_pd_value,
!       CORBA::Any (val)
!     );
  }
  // retrieve the member
! const CORBA::Any
  OBV_Components::ConfigValue::value (void) const
  {
!   return *this->_pd_value;
  }
  // retrieve the member
! CORBA::Any
  OBV_Components::ConfigValue::value (void)
  {
!   return *this->_pd_value;
  }
  
  CORBA::Boolean 
--- 50826,50848 ----
  
  // accessor to set the member
  void
! OBV_Components::ConfigValue::value (CORBA::Any &val) // set
  {
!   this->_pd_value = val;
  }
  // retrieve the member
! const CORBA::Any *
  OBV_Components::ConfigValue::value (void) const
  {
!   // @@ Shouldn't we return a duplicate here?
!   return &this->_pd_value;
  }
  // retrieve the member
! CORBA::Any *
  OBV_Components::ConfigValue::value (void)
  {
!   // @@ Shouldn't we return a duplicate here?
!   return &this->_pd_value;
  }
  
  CORBA::Boolean
***************
*** 65403,65406 ****
    ACE_ENDTRY;
    return 0;
  }
- 
--- 65402,65404 ----
