--- CSD_FrameworkA.h	2005-10-03 16:07:54.484375000 -0700
+++ orig/CSD_FrameworkA.h	2005-10-03 13:11:38.312500000 -0700
@@ -34,8 +34,8 @@
 #include /**/ "ace/pre.h"
 
 #include "CSD_FW_Export.h"
-#include "CSD_FrameworkC.h"
-//#include "tao/PortableServer/PortableServerA.h"
+#include "tao/orig/CSD_FrameworkC.h"
+#include "tao/PortableServer/PortableServerA.h"
 
 
 // TAO_IDL - Generated from
--- CSD_FrameworkC.h	2005-10-03 16:08:02.500000000 -0700
+++ orig/CSD_FrameworkC.h	2005-10-03 13:11:38.312500000 -0700
@@ -47,7 +47,7 @@
 #include "tao/Object.h"
 #include "tao/Objref_VarOut_T.h"
 
-#include "tao/PortableServer/PortableServer.h"
+#include "tao/PortableServer/PortableServerC.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
@@ -305,6 +305,13 @@
 #endif /* end #if !defined */
 }
 
+// TAO_IDL - Generated from
+// .\be\be_codegen.cpp:1040
+
+#if defined (__ACE_INLINE__)
+#include "CSD_FrameworkC.inl"
+#endif /* defined INLINE */
+
 #include /**/ "ace/post.h"
 
 #endif /* ifndef */
