--- orig/RTCORBAC.cpp	Tue Jan 30 19:10:17 2001
+++ RTCORBAC.cpp	Tue Jan 30 19:07:30 2001
@@ -21,6 +21,11 @@
 
 #include "RTCORBAC.h"
 
+#if (TAO_HAS_RT_CORBA == 1)
+
+#include "tao/TAOS.h"
+#include "tao/Typecode.h"
+
 #if TAO_HAS_INTERCEPTORS == 1
 #include "tao/RequestInfo_Util.h"
 #endif  /* TAO_HAS_INTERCEPTORS == 1 */
@@ -843,6 +848,20 @@
   return "IDL:omg.org/RTCORBA/ProtocolProperties:1.0";
 }
 
+//////////////////////////////////////////////////////
+// CDR Encoding Support
+
+CORBA::Boolean
+RTCORBA::ProtocolProperties::_tao_encode (TAO_OutputCDR &)
+{
+  return 1;
+}
+
+CORBA::Boolean
+RTCORBA::ProtocolProperties::_tao_decode (TAO_InputCDR &)
+{
+  return 1;
+}
 
 // default constructor
 RTCORBA::TCPProtocolProperties::TCPProtocolProperties ()
@@ -1747,7 +1766,7 @@
 
 // Narrow.
 RTCORBA::RTORB::InvalidThreadpool *
-RTCORBA::RTORB::InvalidThreadpool::_downcast (CORBA::Exception *exc)
+RTCORBA::RTORB::InvalidThreadpool::_narrow (CORBA::Exception *exc)
 {
   if (!ACE_OS::strcmp ("IDL:omg.org/RTCORBA/RTORB/InvalidThreadpool:1.0", exc->_id ()))
     {
@@ -2347,3 +2366,5 @@
   }
   return 0; // error
 }
+
+#endif /* TAO_HAS_RT_CORBA == 1 */
