--- orig/PortableServerC.h	2005-04-11 16:44:01.152521600 +0200
+++ PortableServerC.h	2005-04-11 16:44:04.287028800 +0200
@@ -36,6 +36,10 @@
 
 #include "ace/config-all.h"
 
+#ifndef TAO_PORTABLESERVER_SAFE_INCLUDE
+#error "You should not include PortableServerC.h directly, use PortableServer.h"
+#endif /* !TAO_PORTABLESERVER_SAFE_INCLUDE */
+
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
@@ -465,6 +469,8 @@
     // TAO_IDL - Generated from
     // be\be_visitor_exception/exception_ch.cpp:51
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 #if !defined (_PORTABLESERVER_POA_NOSERVANT_CH_)
 #define _PORTABLESERVER_POA_NOSERVANT_CH_
     
@@ -512,6 +518,8 @@
 
 #endif /* end #if !defined */
     
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
     // TAO_IDL - Generated from
     // be\be_visitor_exception/exception_ch.cpp:51
 
@@ -855,6 +863,7 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ThreadPolicy_ptr create_thread_policy (
         ::PortableServer::ThreadPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -862,6 +871,7 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
@@ -899,6 +909,7 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ImplicitActivationPolicy_ptr create_implicit_activation_policy (
         ::PortableServer::ImplicitActivationPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -906,10 +917,12 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ServantRetentionPolicy_ptr create_servant_retention_policy (
         ::PortableServer::ServantRetentionPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -917,10 +930,12 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::RequestProcessingPolicy_ptr create_request_processing_policy (
         ::PortableServer::RequestProcessingPolicyValue value
         ACE_ENV_ARG_DECL_WITH_DEFAULTS
@@ -928,6 +943,7 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
@@ -972,6 +988,8 @@
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
+
     virtual ::PortableServer::AdapterActivator_ptr the_activator (
         ACE_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
@@ -989,10 +1007,12 @@
       ACE_THROW_SPEC ((
         CORBA::SystemException
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
     virtual ::PortableServer::ServantManager_ptr get_servant_manager (
         ACE_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
@@ -1012,10 +1032,13 @@
         CORBA::SystemException,
         ::PortableServer::POA::WrongPolicy
       )) = 0;
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
+#if (TAO_HAS_MINIMUM_POA == 0)
+
     virtual ::PortableServer::Servant get_servant (
         ACE_ENV_SINGLE_ARG_DECL_WITH_DEFAULTS
       )
@@ -1037,6 +1060,8 @@
         ::PortableServer::POA::WrongPolicy
       )) = 0;
     
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
@@ -1298,11 +1323,15 @@
 // TAO_IDL - Generated from
 // be\be_visitor_exception/any_op_ch.cpp:52
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, const PortableServer::POA::NoServant &); // copying version
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::POA::NoServant*); // noncopying version
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::POA::NoServant *&); // deprecated
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, const PortableServer::POA::NoServant *&);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_visitor_exception/any_op_ch.cpp:52
 
@@ -1366,4 +1395,3 @@
 
 #endif /* ifndef */
 
-
--- orig/PortableServerC.cpp	2005-04-11 16:44:01.152521600 +0200
+++ PortableServerC.cpp	2005-04-11 16:44:04.317072000 +0200
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:291
 
 
-#include "PortableServerC.h"
+#include "PortableServer.h"
 #include "tao/CDR.h"
 #include "tao/ORB_Core.h"
 #include "tao/Object_Argument_T.h"
@@ -45,7 +45,6 @@
 // Arg traits specializations.
 namespace TAO
 {
-
 #if !defined (_PORTABLESERVER_ADAPTERACTIVATOR__ARG_TRAITS_CS_)
 #define _PORTABLESERVER_ADAPTERACTIVATOR__ARG_TRAITS_CS_
   
@@ -64,7 +63,6 @@
 #endif /* end #if !defined */
 }
 
-
 // TAO_IDL - Generated from 
 // be\be_visitor_sequence/sequence_cs.cpp:65
 
@@ -463,6 +461,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/exception_cs.cpp:63
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 PortableServer::POA::NoServant::NoServant (void)
   : CORBA::UserException (
         "IDL:omg.org/PortableServer/POA/NoServant:2.3",
@@ -555,6 +555,8 @@
   return ::PortableServer::POA::_tc_NoServant;
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/exception_cs.cpp:63
 
--- orig/PortableServerA.cpp	2005-04-11 16:44:01.152521600 +0200
+++ PortableServerA.cpp	2005-04-11 16:44:04.337100800 +0200
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "PortableServerC.h"
+#include "PortableServer.h"
 #include "tao/Null_RefCount_Policy.h"
 #include "tao/TypeCode_Constants.h"
 #include "tao/Alias_TypeCode.h"
@@ -154,6 +154,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/struct_typecode.cpp:70
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static TAO::TypeCode::Struct_Field<char const *, CORBA::TypeCode_ptr const *> const * const _tao_fields_PortableServer_POA_NoServant = 0;
 static TAO::TypeCode::Struct<char const *,
                              CORBA::TypeCode_ptr const *,
@@ -170,6 +172,8 @@
 ::CORBA::TypeCode_ptr const PortableServer::POA::_tc_NoServant =
   &_tao_tc_PortableServer_POA_NoServant;
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/struct_typecode.cpp:70
 
@@ -662,6 +666,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/any_op_cs.cpp:50
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 template<>
 CORBA::Boolean
 TAO::Any_Dual_Impl_T<PortableServer::POA::NoServant>::marshal_value (TAO_OutputCDR &)
@@ -731,6 +737,8 @@
       );
 }
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from 
 // be\be_visitor_exception/any_op_cs.cpp:50
 
