--- orig/TAOC.cpp	Mon May  1 09:12:20 2000
+++ TAOC.cpp	Mon May  1 09:11:02 2000
@@ -9,10 +9,15 @@
 
 #include "TAOC.h"
 
+#include "tao/TAOS.h"
+#include "tao/Typecode.h"
+
 #if !defined (__ACE_INLINE__)
 #include "TAOC.i"
 #endif /* !defined INLINE */
 
+#if (TAO_HAS_CLIENT_PRIORITY_POLICY == 1)
+
 static const CORBA::Long _oc_TAO_PrioritySelectionMode[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -118,11 +123,11 @@
   void *retv = 0;
   if (type == ACE_reinterpret_cast
     (ptr_arith_t,
-      &ACE_NESTED_CLASS (TAO, ClientPriorityPolicy)::_narrow))
+      &ClientPriorityPolicy::_narrow))
     retv = ACE_reinterpret_cast (void*, this);
   else if (type == ACE_reinterpret_cast
     (ptr_arith_t,
-      &ACE_NESTED_CLASS (CORBA, Policy)::_narrow))
+      &CORBA::Policy::_narrow))
     retv = ACE_reinterpret_cast
       (
         void *,
@@ -135,7 +140,7 @@
   else if (type == ACE_reinterpret_cast (ptr_arith_t, &CORBA::Object::_narrow))
     retv = ACE_reinterpret_cast (void *,
       ACE_static_cast (CORBA::Object_ptr, this));
-    
+
   if (retv)
     this->_add_ref ();
   return retv;
@@ -146,6 +151,10 @@
   return "IDL:TAO/ClientPriorityPolicy:1.0";
 }
 
+#endif /* TAO_HAS_CLIENT_PRIORITY_POLICY == 1 */
+
+#if (TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1)
+
 static const CORBA::Long _oc_TAO_BufferingConstraintMode[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -264,11 +273,11 @@
   void *retv = 0;
   if (type == ACE_reinterpret_cast
     (ptr_arith_t,
-      &ACE_NESTED_CLASS (TAO, BufferingConstraintPolicy)::_narrow))
+      &BufferingConstraintPolicy::_narrow))
     retv = ACE_reinterpret_cast (void*, this);
   else if (type == ACE_reinterpret_cast
     (ptr_arith_t,
-      &ACE_NESTED_CLASS (CORBA, Policy)::_narrow))
+      &CORBA::Policy::_narrow))
     retv = ACE_reinterpret_cast
       (
         void *,
@@ -281,7 +290,7 @@
   else if (type == ACE_reinterpret_cast (ptr_arith_t, &CORBA::Object::_narrow))
     retv = ACE_reinterpret_cast (void *,
       ACE_static_cast (CORBA::Object_ptr, this));
-    
+
   if (retv)
     this->_add_ref ();
   return retv;
@@ -292,6 +301,8 @@
   return "IDL:TAO/BufferingConstraintPolicy:1.0";
 }
 
+#endif /* TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1 */
+
 static const CORBA::Long _oc_TAO_SyncScope[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -329,7 +340,10 @@
 TAO_NAMESPACE_BEGIN (TAO)
 TAO_NAMESPACE_DEFINE (const CORBA::Short, SYNC_DELAYED_BUFFERING, -1)
 TAO_NAMESPACE_END
-void operator<<= (CORBA::Any &_tao_any, const TAO::PrioritySpecification &_tao_elem) // copying
+
+#if (TAO_HAS_CLIENT_PRIORITY_POLICY == 1)
+
+ void operator<<= (CORBA::Any &_tao_any, const TAO::PrioritySpecification &_tao_elem) // copying
 {
   TAO_OutputCDR stream;
   stream << _tao_elem;
@@ -410,6 +424,10 @@
   return 0;
 }
 
+#endif /* TAO_HAS_CLIENT_PRIORITY_POLICY == 1 */
+
+#if (TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1)
+
 void operator<<= (CORBA::Any &_tao_any, const TAO::BufferingConstraint &_tao_elem) // copying
 {
   TAO_OutputCDR stream;
@@ -491,3 +509,4 @@
   return 0;
 }
 
+#endif /* TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1 */
