--- CurrentC.h	Sat Jun 24 02:48:51 2000
+++ CurrentC.h.mod	Sat Jun 24 02:47:49 2000
@@ -11,7 +11,7 @@
 #define _TAO_IDL_CURRENTC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+#include "tao/Object.h"
 
 #if defined (ACE_HAS_MINIMUM_IOSTREAMH_INCLUSION)
 #include "ace/streams.h"
@@ -162,7 +162,7 @@
 
 
 #if defined (__ACE_INLINE__)
-#include "CurrentC.i"
+#include "tao/CurrentC.i"
 #endif /* defined INLINE */
 
 #if defined(_MSC_VER) && (_MSC_VER >= 1200)
