--- orig/ObjectReferenceTemplateC.h	2005-03-25 19:44:50.584169600 +0100
+++ ObjectReferenceTemplateC.h	2005-03-25 19:44:51.054846400 +0100
@@ -33,6 +33,9 @@
 
 #include /**/ "ace/pre.h"
 
+#ifndef TAO_OBJREF_TEMPLATE_SAFE_INCLUDE
+#error "You should not include ObjectReferenceTemplateC.h directly, use ObjectReferenceTemplate.h"
+#endif /* !TAO_OBJREF_TEMPLATE_SAFE_INCLUDE */
 
 #include "ace/config-all.h"
 
--- orig/ObjectReferenceTemplateC.cpp	2005-03-25 19:44:50.584169600 +0100
+++ ObjectReferenceTemplateC.cpp	2005-03-25 19:44:51.114932800 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "ObjectReferenceTemplateC.h"
+#include "ObjectReferenceTemplate.h"
 #include "tao/CDR.h"
 #include "tao/Exception_Data.h"
 #include "tao/Invocation_Adapter.h"
--- orig/ObjectReferenceTemplateA.cpp	2005-03-25 19:44:50.584169600 +0100
+++ ObjectReferenceTemplateA.cpp	2005-03-25 19:45:21.758996800 +0100
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "ObjectReferenceTemplateC.h"
+#include "ObjectReferenceTemplate.h"
 #include "tao/Null_RefCount_Policy.h"
 #include "tao/TypeCode_Constants.h"
 #include "tao/Alias_TypeCode.h"
