--- IOPC.h.old	Fri Feb 23 15:57:27 2001
+++ IOPC.h	Fri Feb 23 18:22:02 2001
@@ -23,13 +23,18 @@
 #define _TAO_IDL_IOPC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+
+#include "TAO_Export.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
-#include "TAO_Export.h"
+#include "Any.h"
+#include "Object.h"
+#include "Exception.h"
+#include "CDR.h"
+#include "Sequence.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
--- IOPC.cpp.old	Fri Feb 23 15:57:27 2001
+++ IOPC.cpp	Fri Feb 23 18:22:02 2001
@@ -21,10 +21,6 @@
 
 #include "IOPC.h"
 
-#if TAO_HAS_INTERCEPTORS == 1
-#include "tao/RequestInfo_Util.h"
-#endif  /* TAO_HAS_INTERCEPTORS == 1 */
-
 #if defined (__BORLANDC__)
 #pragma option -w-rvl -w-rch -w-ccc -w-aus
 #endif /* __BORLANDC__ */
@@ -32,6 +28,8 @@
 #if !defined (__ACE_INLINE__)
 #include "IOPC.i"
 #endif /* !defined INLINE */
+
+#include "Typecode.h"
 
 static const CORBA::Long _oc_IOP_ProfileId[] =
 {
