diff -rupd ACE_wrappers.orig/bin/bootstrap ACE_wrappers/bin/bootstrap
--- ACE_wrappers.orig/bin/bootstrap	2009-12-12 17:01:21.844361244 +0100
+++ ACE_wrappers/bin/bootstrap	2009-12-12 17:02:12.824683481 +0100
@@ -73,9 +73,13 @@ if test -d m4; then
 #  echo "Creating a NEWS file"
 #  cp VERSION NEWS
 
-#    autoreconf -I m4 --install --force --symlink --verbose
-    autoreconf -I m4 --install --force 
+    cd $ACE_ROOT && autoupdate -f
+    cd $TAO_ROOT && autoupdate -f
 
+#    autoreconf -I m4 --install --force --symlink --verbose
+    cd $ACE_ROOT && autoreconf -I m4 --install --force 
+    cd $TAO_ROOT && autoreconf -I m4 --install --force
+    
 #   # Generate the man pages.
 #   # Only generate man pages if bootstrapping a release.
 #   if test $bootstrap_release = yes; then
