--- orig/PolicyFactoryC.h	2005-04-15 11:58:39.449555200 +0200
+++ PolicyFactoryC.h	2005-04-15 12:22:48.683451200 +0200
@@ -142,6 +142,16 @@
         ::CORBA::PolicyError
       )) = 0;
     
+    // Hand-crafted addition.
+    virtual ::CORBA::Policy_ptr _create_policy (
+        CORBA::PolicyType type
+        ACE_ENV_ARG_DECL_WITH_DEFAULTS
+      )
+      ACE_THROW_SPEC ((
+        CORBA::SystemException
+        , CORBA::PolicyError
+      ));
+
     // TAO_IDL - Generated from
     // c:\ace\latest\ace_wrappers\tao\tao_idl\be\be_visitor_interface/interface_ch.cpp:210
     
--- orig/PolicyFactoryC.cpp	2005-04-15 11:58:39.489612800 +0200
+++ PolicyFactoryC.cpp	2005-04-15 12:23:19.297472000 +0200
@@ -186,3 +186,15 @@
 {
   return false;
 }
+
+// Hand-crafted.
+::CORBA::Policy_ptr
+PortableInterceptor::PolicyFactory::_create_policy (CORBA::PolicyType
+                                                    ACE_ENV_ARG_DECL)
+  ACE_THROW_SPEC ((CORBA::SystemException,
+                   CORBA::PolicyError))
+{
+  ACE_THROW_RETURN (CORBA::PolicyError (CORBA::BAD_POLICY_TYPE),
+                    CORBA::Policy::_nil ());
+}
+
