--- EDF_SchedulingC.h.orig	2003-12-04 00:55:30.000000000 -0600
+++ EDF_SchedulingC.h	2003-12-04 00:55:53.000000000 -0600
@@ -44,8 +44,8 @@
 #include "tao/Objref_VarOut_T.h"
 #include "tao/VarOut_T.h"
 
-#include "tao/RTScheduling/RTSchedulerC.h"
-#include "tao/RTCORBA/RTCORBAC.h"
+#include "tao/RTScheduling/RTScheduler.h"
+#include "tao/RTCORBA/RTCORBA.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
