--- orig/PollableC.h	Tue Feb  5 15:23:35 2002
+++ PollableC.h	Tue Feb  5 15:26:57 2002
@@ -23,13 +23,16 @@
 #define _TAO_IDL_ORIG_POLLABLEC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+#include "tao/corbafwd.h"
 
+#if (TAO_HAS_AMI_POLLER == 1)
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 #include "TAO_Export.h"
+#include "Exception.h"
+#include "Object.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
@@ -676,6 +679,8 @@
 #if defined (__BORLANDC__)
 #pragma option pop
 #endif /* __BORLANDC__ */
+
+#endif /* TAO_HAS_AMI_POLLER == 1 */
 
 #include "ace/post.h"
 #endif /* ifndef */
--- orig/PollableC.cpp	Tue Feb  5 15:23:35 2002
+++ PollableC.cpp	Tue Feb  5 15:26:06 2002
@@ -21,6 +21,8 @@
 
 #include "PollableC.h"
 
+#if (TAO_HAS_AMI_POLLER == 1)
+
 #include "tao/Stub.h"
 #include "tao/Invocation.h"
 #include "tao/PortableInterceptor.h"
@@ -1131,3 +1133,4 @@
   #  pragma instantiate TAO_Object_Manager<CORBA::PollableSet,CORBA::PollableSet_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_AMI_POLLER == 1 */
