--- orig/PortableInterceptorC.h	2005-04-15 12:04:07.461212800 +0200
+++ PortableInterceptorC.h	2005-04-15 12:24:20.435384000 +0200
@@ -61,6 +61,12 @@
 #include "tao/PolicyFactoryC.h"
 #include "tao/ORBInitInfoC.h"
 
+#include "PI_ForwardC.h"
+#include "tao/Messaging_SyncScopeC.h"
+#include "tao/IOP_CodecC.h"
+#include "tao/PolicyC.h"
+#include "tao/CurrentC.h"
+
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
@@ -75,6 +81,16 @@
 #pragma option push -w-rvl -w-rch -w-ccc -w-inl
 #endif /* __BORLANDC__ */
 
+// Hand crafted, keeps DynamicC.h and Typecode.h out of header files.
+namespace Dynamic
+{
+  class ParameterList;
+  class ExceptionList;
+  typedef CORBA::StringSeq ContextList;
+  typedef CORBA::StringSeq RequestContext;
+}
+// TAO_IDL - Generated from
+
 // TAO_IDL - Generated from
 // c:\ace\latest\ace_wrappers\tao\tao_idl\be\be_visitor_module/module_ch.cpp:48
 
@@ -625,15 +641,19 @@
     // TAO_IDL - Generated from
     // c:\ace\latest\ace_wrappers\tao\tao_idl\be\be_visitor_operation/operation_ch.cpp:46
     
+#if TAO_HAS_EXTENDED_FT_INTERCEPTORS == 1
+
     virtual void tao_ft_interception_point (
         ::PortableInterceptor::ServerRequestInfo_ptr ri,
         ::CORBA::OctetSeq_out os
-        ACE_ENV_ARG_DECL_WITH_DEFAULTS
+        ACE_ENV_ARG_DECL_NOT_USED
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
         ::PortableInterceptor::ForwardRequest
-      )) = 0;
+      )) {}
+
+#endif /*TAO_HAS_EXTENDED_FT_INTERCEPTORS*/
     
     // TAO_IDL - Generated from
     // c:\ace\latest\ace_wrappers\tao\tao_idl\be\be_visitor_operation/operation_ch.cpp:46
