--- y.tab.cpp	2003-08-26 17:09:54.000000000 -0500
+++ y.tab.cpp.mine	2003-08-26 17:09:45.000000000 -0500
@@ -1752,11 +1752,6 @@
   else
     TAO_YYFPRINTF (tao_yyoutput, "nterm %s (", tao_yytname[tao_yytype]);
 
-  switch (tao_yytype)
-    {
-      default:
-        break;
-    }
   TAO_YYFPRINTF (tao_yyoutput, ")");
 }
 
@@ -1778,12 +1773,6 @@
   /* Pacify ``unused variable'' warnings.  */
   (void) tao_yyvaluep;
 
-  switch (tao_yytype)
-    {
-
-      default:
-        break;
-    }
 }
 
 
