--- orig/MessagingC.i	Mon May  1 21:10:26 2000
+++ MessagingC.i	Mon May  1 21:10:15 2000
@@ -7,6 +7,8 @@
 // Information about TAO is available at:
 //                 http://www.cs.wustl.edu/~schmidt/TAO.html
 
+#if (TAO_HAS_REBIND_POLICY == 1)
+
 ACE_INLINE
 Messaging::RebindPolicy::RebindPolicy (void) // default constructor
 {}
@@ -69,14 +71,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RebindPolicy_var::operator const ::Messaging::RebindPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RebindPolicy_var::operator ::Messaging::RebindPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RebindPolicy_var::operator ::Messaging::RebindPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -168,7 +170,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RebindPolicy_out::operator ::Messaging::RebindPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -189,6 +191,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REBIND_POLICY == 1 */
+
+#if (TAO_HAS_SYNC_SCOPE_POLICY == 1)
+
 ACE_INLINE
 Messaging::SyncScopePolicy::SyncScopePolicy (void) // default constructor
 {}
@@ -251,14 +257,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::SyncScopePolicy_var::operator const ::Messaging::SyncScopePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::SyncScopePolicy_var::operator ::Messaging::SyncScopePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::SyncScopePolicy_var::operator ::Messaging::SyncScopePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -350,7 +356,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::SyncScopePolicy_out::operator ::Messaging::SyncScopePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -371,6 +377,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_SYNC_SCOPE_POLICY == 1 */
+
+#if (TAO_HAS_PRIORITY_POLICIES == 1)
+
 // *************************************************************
 // Inline operations for class Messaging::PriorityRange_var
 // *************************************************************
@@ -457,13 +467,13 @@
 }
 
 ACE_INLINE
-Messaging::PriorityRange_var::operator ::Messaging::PriorityRange &() // cast 
+Messaging::PriorityRange_var::operator ::Messaging::PriorityRange &() // cast
 {
   return *this->ptr_;
 }
 
 ACE_INLINE
-Messaging::PriorityRange_var::operator ::Messaging::PriorityRange &() const // cast 
+Messaging::PriorityRange_var::operator ::Messaging::PriorityRange &() const // cast
 {
   return *this->ptr_;
 }
@@ -480,7 +490,7 @@
   return *this->ptr_;
 }
 
-// mapping for fixed size 
+// mapping for fixed size
 ACE_INLINE ::Messaging::PriorityRange &
 Messaging::PriorityRange_var::out (void)
 {
@@ -561,14 +571,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestPriorityPolicy_var::operator const ::Messaging::RequestPriorityPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RequestPriorityPolicy_var::operator ::Messaging::RequestPriorityPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RequestPriorityPolicy_var::operator ::Messaging::RequestPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -660,7 +670,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestPriorityPolicy_out::operator ::Messaging::RequestPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -743,14 +753,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyPriorityPolicy_var::operator const ::Messaging::ReplyPriorityPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ReplyPriorityPolicy_var::operator ::Messaging::ReplyPriorityPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::ReplyPriorityPolicy_var::operator ::Messaging::ReplyPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -842,7 +852,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyPriorityPolicy_out::operator ::Messaging::ReplyPriorityPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -863,6 +873,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_PRIORITY_POLICIES == 1 */
+
+#if (TAO_HAS_REQUEST_START_TIME_POLICY == 1)
+
 ACE_INLINE
 Messaging::RequestStartTimePolicy::RequestStartTimePolicy (void) // default constructor
 {}
@@ -925,14 +939,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestStartTimePolicy_var::operator const ::Messaging::RequestStartTimePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RequestStartTimePolicy_var::operator ::Messaging::RequestStartTimePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RequestStartTimePolicy_var::operator ::Messaging::RequestStartTimePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1024,7 +1038,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestStartTimePolicy_out::operator ::Messaging::RequestStartTimePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1045,6 +1059,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REQUEST_START_TIME_POLICY == 1 */
+
+#if (TAO_HAS_REQUEST_END_TIME_POLICY == 1)
+
 ACE_INLINE
 Messaging::RequestEndTimePolicy::RequestEndTimePolicy (void) // default constructor
 {}
@@ -1107,14 +1125,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestEndTimePolicy_var::operator const ::Messaging::RequestEndTimePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RequestEndTimePolicy_var::operator ::Messaging::RequestEndTimePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RequestEndTimePolicy_var::operator ::Messaging::RequestEndTimePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1206,7 +1224,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RequestEndTimePolicy_out::operator ::Messaging::RequestEndTimePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1227,6 +1245,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REQUEST_END_TIME_POLICY == 1 */
+
+#if (TAO_HAS_REPLY_START_TIME_POLICY == 1)
+
 ACE_INLINE
 Messaging::ReplyStartTimePolicy::ReplyStartTimePolicy (void) // default constructor
 {}
@@ -1289,14 +1311,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyStartTimePolicy_var::operator const ::Messaging::ReplyStartTimePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ReplyStartTimePolicy_var::operator ::Messaging::ReplyStartTimePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::ReplyStartTimePolicy_var::operator ::Messaging::ReplyStartTimePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1388,7 +1410,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyStartTimePolicy_out::operator ::Messaging::ReplyStartTimePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1409,6 +1431,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REPLY_START_TIME_POLICY == 1 */
+
+#if (TAO_HAS_REPLY_END_TIME_POLICY == 1)
+
 ACE_INLINE
 Messaging::ReplyEndTimePolicy::ReplyEndTimePolicy (void) // default constructor
 {}
@@ -1471,14 +1497,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyEndTimePolicy_var::operator const ::Messaging::ReplyEndTimePolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ReplyEndTimePolicy_var::operator ::Messaging::ReplyEndTimePolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::ReplyEndTimePolicy_var::operator ::Messaging::ReplyEndTimePolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1570,7 +1596,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyEndTimePolicy_out::operator ::Messaging::ReplyEndTimePolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1591,6 +1617,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_REPLY_END_TIME_POLICY == 1 */
+
+#if (TAO_HAS_RELATIVE_REQUEST_TIMEOUT_POLICY == 1)
+
 ACE_INLINE
 Messaging::RelativeRequestTimeoutPolicy::RelativeRequestTimeoutPolicy (void) // default constructor
 {}
@@ -1653,14 +1683,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RelativeRequestTimeoutPolicy_var::operator const ::Messaging::RelativeRequestTimeoutPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RelativeRequestTimeoutPolicy_var::operator ::Messaging::RelativeRequestTimeoutPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RelativeRequestTimeoutPolicy_var::operator ::Messaging::RelativeRequestTimeoutPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1752,7 +1782,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RelativeRequestTimeoutPolicy_out::operator ::Messaging::RelativeRequestTimeoutPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1773,6 +1803,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_RELATIVE_REQUEST_TIMEOUT_POLICY == 1 */
+
+#if (TAO_HAS_RELATIVE_ROUNDTRIP_TIMEOUT_POLICY == 1)
+
 ACE_INLINE
 Messaging::RelativeRoundtripTimeoutPolicy::RelativeRoundtripTimeoutPolicy (void) // default constructor
 {}
@@ -1835,14 +1869,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RelativeRoundtripTimeoutPolicy_var::operator const ::Messaging::RelativeRoundtripTimeoutPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RelativeRoundtripTimeoutPolicy_var::operator ::Messaging::RelativeRoundtripTimeoutPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RelativeRoundtripTimeoutPolicy_var::operator ::Messaging::RelativeRoundtripTimeoutPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -1934,7 +1968,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RelativeRoundtripTimeoutPolicy_out::operator ::Messaging::RelativeRoundtripTimeoutPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -1959,6 +1993,10 @@
 // Inline operations for class Messaging::RoutingTypeRange_var
 // *************************************************************
 
+#endif /* TAO_HAS_RELATIVE_ROUNDTRIP_TIMEOUT_POLICY == 1 */
+
+#if (TAO_HAS_ROUTING_POLICY == 1)
+
 ACE_INLINE
 Messaging::RoutingTypeRange_var::RoutingTypeRange_var (void) // default constructor
   : ptr_ (0)
@@ -2041,13 +2079,13 @@
 }
 
 ACE_INLINE
-Messaging::RoutingTypeRange_var::operator ::Messaging::RoutingTypeRange &() // cast 
+Messaging::RoutingTypeRange_var::operator ::Messaging::RoutingTypeRange &() // cast
 {
   return *this->ptr_;
 }
 
 ACE_INLINE
-Messaging::RoutingTypeRange_var::operator ::Messaging::RoutingTypeRange &() const // cast 
+Messaging::RoutingTypeRange_var::operator ::Messaging::RoutingTypeRange &() const // cast
 {
   return *this->ptr_;
 }
@@ -2064,7 +2102,7 @@
   return *this->ptr_;
 }
 
-// mapping for fixed size 
+// mapping for fixed size
 ACE_INLINE ::Messaging::RoutingTypeRange &
 Messaging::RoutingTypeRange_var::out (void)
 {
@@ -2145,14 +2183,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RoutingPolicy_var::operator const ::Messaging::RoutingPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::RoutingPolicy_var::operator ::Messaging::RoutingPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::RoutingPolicy_var::operator ::Messaging::RoutingPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -2244,7 +2282,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::RoutingPolicy_out::operator ::Messaging::RoutingPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -2265,6 +2303,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_ROUTING_POLICY == 1 */
+
+#if (TAO_HAS_MAX_HOPS_POLICY == 1)
+
 ACE_INLINE
 Messaging::MaxHopsPolicy::MaxHopsPolicy (void) // default constructor
 {}
@@ -2327,14 +2369,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::MaxHopsPolicy_var::operator const ::Messaging::MaxHopsPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::MaxHopsPolicy_var::operator ::Messaging::MaxHopsPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::MaxHopsPolicy_var::operator ::Messaging::MaxHopsPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -2426,7 +2468,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::MaxHopsPolicy_out::operator ::Messaging::MaxHopsPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -2447,6 +2489,10 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_MAX_HOPS_POLICY == 1 */
+
+#if (TAO_HAS_QUEUE_ORDER_POLICY == 1)
+
 ACE_INLINE
 Messaging::QueueOrderPolicy::QueueOrderPolicy (void) // default constructor
 {}
@@ -2509,14 +2555,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::QueueOrderPolicy_var::operator const ::Messaging::QueueOrderPolicy_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::QueueOrderPolicy_var::operator ::Messaging::QueueOrderPolicy_ptr &() // cast 
+ACE_INLINE
+Messaging::QueueOrderPolicy_var::operator ::Messaging::QueueOrderPolicy_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -2608,7 +2654,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::QueueOrderPolicy_out::operator ::Messaging::QueueOrderPolicy_ptr &() // cast
 {
   return this->ptr_;
@@ -2629,6 +2675,8 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_QUEUE_ORDER_POLICY == 1 */
+
 // *************************************************************
 // Inline operations for class Messaging::PolicyValue_var
 // *************************************************************
@@ -2696,20 +2744,20 @@
 }
 
 ACE_INLINE
-Messaging::PolicyValue_var::operator ::Messaging::PolicyValue &() // cast 
+Messaging::PolicyValue_var::operator ::Messaging::PolicyValue &() // cast
 {
   return *this->ptr_;
 }
 
 ACE_INLINE
-Messaging::PolicyValue_var::operator ::Messaging::PolicyValue &() const // cast 
+Messaging::PolicyValue_var::operator ::Messaging::PolicyValue &() const // cast
 {
   return *this->ptr_;
 }
 
 // variable-size types only
 ACE_INLINE
-Messaging::PolicyValue_var::operator ::Messaging::PolicyValue *&() // cast 
+Messaging::PolicyValue_var::operator ::Messaging::PolicyValue *&() // cast
 {
   return this->ptr_;
 }
@@ -2726,7 +2774,7 @@
   return *this->ptr_;
 }
 
-// mapping for variable size 
+// mapping for variable size
 ACE_INLINE ::Messaging::PolicyValue *&
 Messaging::PolicyValue_var::out (void)
 {
@@ -2787,7 +2835,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValue_out::operator ::Messaging::PolicyValue *&() // cast
 {
   return this->ptr_;
@@ -2888,20 +2936,20 @@
   return this->ptr_;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValue::_tao_seq_Octet_var::operator const ::Messaging::PolicyValue::_tao_seq_Octet &() const // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::PolicyValue::_tao_seq_Octet_var::operator ::Messaging::PolicyValue::_tao_seq_Octet &() // cast 
+ACE_INLINE
+Messaging::PolicyValue::_tao_seq_Octet_var::operator ::Messaging::PolicyValue::_tao_seq_Octet &() // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::PolicyValue::_tao_seq_Octet_var::operator ::Messaging::PolicyValue::_tao_seq_Octet &() const // cast 
+ACE_INLINE
+Messaging::PolicyValue::_tao_seq_Octet_var::operator ::Messaging::PolicyValue::_tao_seq_Octet &() const // cast
 {
   return *this->ptr_;
 }
@@ -2924,7 +2972,7 @@
   return *this->ptr_;
 }
 
-// mapping for variable size 
+// mapping for variable size
 ACE_INLINE ::Messaging::PolicyValue::_tao_seq_Octet *&
 Messaging::PolicyValue::_tao_seq_Octet_var::out (void)
 {
@@ -2985,7 +3033,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValue::_tao_seq_Octet_out::operator ::Messaging::PolicyValue::_tao_seq_Octet *&() // cast
 {
   return this->ptr_;
@@ -3014,7 +3062,7 @@
 
 
 #if !defined (TAO_USE_SEQUENCE_TEMPLATES)
-  
+
 #if !defined (__TAO_UNBOUNDED_SEQUENCE_MESSAGING_POLICYVALUESEQ_CI_)
 #define __TAO_UNBOUNDED_SEQUENCE_MESSAGING_POLICYVALUESEQ_CI_
 
@@ -3027,24 +3075,24 @@
     ACE_NEW_RETURN (retval, Messaging::PolicyValue[size], 0);
     return retval;
   }
-  
+
   ACE_INLINE void Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::freebuf (Messaging::PolicyValue *buffer)
   // Free the sequence.
   {
     delete [] buffer;
   }
-  
+
   ACE_INLINE
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq (void) // Default constructor.
   {
   }
-  
+
   ACE_INLINE
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq (CORBA::ULong maximum) // Constructor using a maximum length value.
     : TAO_Unbounded_Base_Sequence (maximum, _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::allocbuf (maximum))
   {
   }
-  
+
   ACE_INLINE
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq (CORBA::ULong maximum,
     CORBA::ULong length,
@@ -3053,7 +3101,7 @@
   : TAO_Unbounded_Base_Sequence (maximum, length, data, release)
   {
   }
-  
+
   ACE_INLINE
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq (const _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq &rhs)
   // Copy constructor.
@@ -3063,10 +3111,10 @@
     {
       Messaging::PolicyValue *tmp1 = _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::allocbuf (this->maximum_);
       Messaging::PolicyValue * const tmp2 = ACE_reinterpret_cast (Messaging::PolicyValue * ACE_CAST_CONST, rhs.buffer_);
-      
+
       for (CORBA::ULong i = 0; i < this->length_; ++i)
         tmp1[i] = tmp2[i];
-      
+
       this->buffer_ = tmp1;
     }
     else
@@ -3074,14 +3122,14 @@
       this->buffer_ = 0;
     }
   }
-  
+
   ACE_INLINE Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq &
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::operator= (const _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq &rhs)
   // Assignment operator.
   {
     if (this == &rhs)
       return *this;
-    
+
     if (this->release_)
     {
       if (this->maximum_ < rhs.maximum_)
@@ -3094,18 +3142,18 @@
     }
     else
       this->buffer_ = _TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::allocbuf (rhs.maximum_);
-    
+
     TAO_Unbounded_Base_Sequence::operator= (rhs);
-    
+
     Messaging::PolicyValue *tmp1 = ACE_reinterpret_cast (Messaging::PolicyValue *, this->buffer_);
     Messaging::PolicyValue * const tmp2 = ACE_reinterpret_cast (Messaging::PolicyValue * ACE_CAST_CONST, rhs.buffer_);
-    
+
     for (CORBA::ULong i = 0; i < this->length_; ++i)
       tmp1[i] = tmp2[i];
-    
+
     return *this;
   }
-  
+
   // = Accessors.
   ACE_INLINE Messaging::PolicyValue &
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::operator[] (CORBA::ULong i)
@@ -3115,7 +3163,7 @@
     Messaging::PolicyValue* tmp = ACE_reinterpret_cast(Messaging::PolicyValue*,this->buffer_);
     return tmp[i];
   }
-  
+
   ACE_INLINE const Messaging::PolicyValue &
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::operator[] (CORBA::ULong i) const
   // operator []
@@ -3124,9 +3172,9 @@
     Messaging::PolicyValue * const tmp = ACE_reinterpret_cast (Messaging::PolicyValue* ACE_CAST_CONST, this->buffer_);
     return tmp[i];
   }
-  
+
   // Implement the TAO_Base_Sequence methods (see Sequence.h)
-  
+
   ACE_INLINE Messaging::PolicyValue *
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::get_buffer (CORBA::Boolean orphan)
   {
@@ -3159,13 +3207,13 @@
     }
     return result;
   }
-  
+
   ACE_INLINE const Messaging::PolicyValue *
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::get_buffer (void) const
   {
     return ACE_reinterpret_cast(const Messaging::PolicyValue * ACE_CAST_CONST, this->buffer_);
   }
-  
+
   ACE_INLINE void
   Messaging::_TAO_Unbounded_Sequence_Messaging_PolicyValueSeq::replace (CORBA::ULong max,
   CORBA::ULong length,
@@ -3182,11 +3230,11 @@
     this->buffer_ = data;
     this->release_ = release;
   }
-  
+
 #endif /* end #if !defined */
 
 
-#endif /* !TAO_USE_SEQUENCE_TEMPLATES */ 
+#endif /* !TAO_USE_SEQUENCE_TEMPLATES */
 
 #if !defined (_MESSAGING_POLICYVALUESEQ_CI_)
 #define _MESSAGING_POLICYVALUESEQ_CI_
@@ -3251,27 +3299,27 @@
   return this->ptr_;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValueSeq_var::operator const ::Messaging::PolicyValueSeq &() const // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq &() // cast 
+ACE_INLINE
+Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq &() // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq &() const // cast 
+ACE_INLINE
+Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq &() const // cast
 {
   return *this->ptr_;
 }
 
 // variable-size types only
 ACE_INLINE
-Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq *&() // cast 
+Messaging::PolicyValueSeq_var::operator ::Messaging::PolicyValueSeq *&() // cast
 {
   return this->ptr_;
 }
@@ -3294,7 +3342,7 @@
   return *this->ptr_;
 }
 
-// mapping for variable size 
+// mapping for variable size
 ACE_INLINE ::Messaging::PolicyValueSeq *&
 Messaging::PolicyValueSeq_var::out (void)
 {
@@ -3355,7 +3403,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::PolicyValueSeq_out::operator ::Messaging::PolicyValueSeq *&() // cast
 {
   return this->ptr_;
@@ -3382,6 +3430,26 @@
 
 #endif /* end #if !defined */
 
+#if (TAO_HAS_AMI_CALLBACK == 1) || (TAO_HAS_AMI_POLLER == 1)
+
+ACE_INLINE
+Messaging::ExceptionHolder::ExceptionHolder () // default constructor
+{}
+
+ACE_INLINE
+Messaging::ExceptionHolder::~ExceptionHolder () // destructor
+{}
+
+ACE_INLINE const char*
+Messaging::ExceptionHolder::_tao_obv_static_repository_id ()
+{
+  return "IDL:omg.org/Messaging/ExceptionHolder:1.0";
+}
+
+
+#if !defined (_MESSAGING_EXCEPTIONHOLDER___VAR_CI_)
+#define _MESSAGING_EXCEPTIONHOLDER___VAR_CI_
+
 // *************************************************************
 // Inline operations for class Messaging::ExceptionHolder_var
 // *************************************************************
@@ -3392,173 +3460,174 @@
 {}
 
 ACE_INLINE
-Messaging::ExceptionHolder_var::ExceptionHolder_var (ExceptionHolder *p)
+Messaging::ExceptionHolder_var::ExceptionHolder_var (Messaging::ExceptionHolder* p)
   : ptr_ (p)
 {}
 
 ACE_INLINE
-Messaging::ExceptionHolder_var::ExceptionHolder_var (const ::Messaging::ExceptionHolder_var &p) // copy constructor
+Messaging::ExceptionHolder_var::ExceptionHolder_var (const Messaging::ExceptionHolder* p)
+  : ptr_ (ACE_const_cast(Messaging::ExceptionHolder*, p))
+{}
+
+ACE_INLINE Messaging::ExceptionHolder*
+Messaging::ExceptionHolder_var::ptr (void) const
 {
-  if (p.ptr_)
-    ACE_NEW (this->ptr_, ::Messaging::ExceptionHolder (*p.ptr_));
-  else
-    this->ptr_ = 0;
+  return this->ptr_;
+}
+
+ACE_INLINE
+Messaging::ExceptionHolder_var::ExceptionHolder_var (const Messaging::ExceptionHolder_var &p) // copy constructor
+{
+  CORBA::add_ref (p.ptr ());
+  this->ptr_ = p.ptr ();
 }
 
 ACE_INLINE
 Messaging::ExceptionHolder_var::~ExceptionHolder_var (void) // destructor
 {
-  delete this->ptr_;
+  CORBA::remove_ref (this->ptr_);
 }
 
 ACE_INLINE Messaging::ExceptionHolder_var &
-Messaging::ExceptionHolder_var::operator= (ExceptionHolder *p)
+Messaging::ExceptionHolder_var::operator= (Messaging::ExceptionHolder* p)
 {
-  delete this->ptr_;
+  CORBA::remove_ref (this->ptr_);
   this->ptr_ = p;
   return *this;
 }
 
-ACE_INLINE ::Messaging::ExceptionHolder_var &
-Messaging::ExceptionHolder_var::operator= (const ::Messaging::ExceptionHolder_var &p)
+ACE_INLINE Messaging::ExceptionHolder_var &
+Messaging::ExceptionHolder_var::operator= (const Messaging::ExceptionHolder_var &p)
 {
   if (this != &p)
   {
-    delete this->ptr_;
-    ACE_NEW_RETURN (this->ptr_, ::Messaging::ExceptionHolder (*p.ptr_), *this);
+    CORBA::remove_ref (this->ptr_);
+    ExceptionHolder* tmp = p.ptr ();
+    CORBA::add_ref (tmp);
+    this->ptr_ = tmp;
   }
   return *this;
 }
 
-ACE_INLINE const ::Messaging::ExceptionHolder *
-Messaging::ExceptionHolder_var::operator-> (void) const
-{
-  return this->ptr_;
-}
-
-ACE_INLINE ::Messaging::ExceptionHolder *
-Messaging::ExceptionHolder_var::operator-> (void)
-{
-  return this->ptr_;
-}
-
 ACE_INLINE
-Messaging::ExceptionHolder_var::operator const ::Messaging::ExceptionHolder &() const // cast
+Messaging::ExceptionHolder_var::operator const Messaging::ExceptionHolder* () const // cast
 {
-  return *this->ptr_;
-}
-
-ACE_INLINE
-Messaging::ExceptionHolder_var::operator ::Messaging::ExceptionHolder &() // cast 
-{
-  return *this->ptr_;
+  return this->ptr_;
 }
 
 ACE_INLINE
-Messaging::ExceptionHolder_var::operator ::Messaging::ExceptionHolder &() const // cast 
+Messaging::ExceptionHolder_var::operator Messaging::ExceptionHolder* () // cast
 {
-  return *this->ptr_;
+  return this->ptr_;
 }
 
-// variable-size types only
-ACE_INLINE
-Messaging::ExceptionHolder_var::operator ::Messaging::ExceptionHolder *&() // cast 
+ACE_INLINE Messaging::ExceptionHolder*
+Messaging::ExceptionHolder_var::operator-> (void) const
 {
   return this->ptr_;
 }
 
-ACE_INLINE const ::Messaging::ExceptionHolder &
+ACE_INLINE Messaging::ExceptionHolder*
 Messaging::ExceptionHolder_var::in (void) const
 {
-  return *this->ptr_;
+  return this->ptr_;
 }
 
-ACE_INLINE ::Messaging::ExceptionHolder &
+ACE_INLINE Messaging::ExceptionHolder* &
 Messaging::ExceptionHolder_var::inout (void)
 {
-  return *this->ptr_;
+  return this->ptr_;
 }
 
-// mapping for variable size 
-ACE_INLINE ::Messaging::ExceptionHolder *&
+ACE_INLINE Messaging::ExceptionHolder* &
 Messaging::ExceptionHolder_var::out (void)
 {
-  delete this->ptr_;
+  CORBA::remove_ref (this->ptr_);
   this->ptr_ = 0;
   return this->ptr_;
 }
 
-ACE_INLINE ::Messaging::ExceptionHolder *
+ACE_INLINE Messaging::ExceptionHolder*
 Messaging::ExceptionHolder_var::_retn (void)
 {
-  ::Messaging::ExceptionHolder *tmp = this->ptr_;
+  // yield ownership of managed obj reference
+  ExceptionHolder* tmp = this->ptr_;
   this->ptr_ = 0;
   return tmp;
 }
 
-ACE_INLINE ::Messaging::ExceptionHolder *
-Messaging::ExceptionHolder_var::ptr (void) const
-{
-  return this->ptr_;
-}
+
+#endif /* end #if !defined */
+
+
+#if !defined (_MESSAGING_EXCEPTIONHOLDER___OUT_CI_)
+#define _MESSAGING_EXCEPTIONHOLDER___OUT_CI_
 
 // *************************************************************
 // Inline operations for class Messaging::ExceptionHolder_out
 // *************************************************************
 
 ACE_INLINE
-Messaging::ExceptionHolder_out::ExceptionHolder_out (::Messaging::ExceptionHolder *&p)
+Messaging::ExceptionHolder_out::ExceptionHolder_out (Messaging::ExceptionHolder* &p)
   : ptr_ (p)
 {
   this->ptr_ = 0;
 }
 
 ACE_INLINE
-Messaging::ExceptionHolder_out::ExceptionHolder_out (ExceptionHolder_var &p) // constructor from _var
+Messaging::ExceptionHolder_out::ExceptionHolder_out (Messaging::ExceptionHolder_var &p) // constructor from _var
   : ptr_ (p.out ())
 {
-  delete this->ptr_;
+  CORBA::remove_ref (this->ptr_);
   this->ptr_ = 0;
 }
 
 ACE_INLINE
-Messaging::ExceptionHolder_out::ExceptionHolder_out (const ::Messaging::ExceptionHolder_out &p) // copy constructor
-  : ptr_ (ACE_const_cast (ExceptionHolder_out&, p).ptr_)
+Messaging::ExceptionHolder_out::ExceptionHolder_out (const Messaging::ExceptionHolder_out &p) // copy constructor
+  : ptr_ (ACE_const_cast (Messaging::ExceptionHolder_out&,p).ptr_)
 {}
 
 ACE_INLINE Messaging::ExceptionHolder_out &
-Messaging::ExceptionHolder_out::operator= (const ::Messaging::ExceptionHolder_out &p)
+Messaging::ExceptionHolder_out::operator= (const Messaging::ExceptionHolder_out &p)
 {
-  this->ptr_ = ACE_const_cast (ExceptionHolder_out&, p).ptr_;
+  this->ptr_ = ACE_const_cast (Messaging::ExceptionHolder_out&,p).ptr_;
   return *this;
 }
 
 ACE_INLINE Messaging::ExceptionHolder_out &
-Messaging::ExceptionHolder_out::operator= (ExceptionHolder *p)
+Messaging::ExceptionHolder_out::operator= (const Messaging::ExceptionHolder_var &p)
+{
+  ExceptionHolder* tmp = p.ptr ();
+  CORBA::add_ref (tmp);
+  this->ptr_ = tmp;
+  return *this;
+}
+
+ACE_INLINE Messaging::ExceptionHolder_out &
+Messaging::ExceptionHolder_out::operator= (Messaging::ExceptionHolder* p)
 {
   this->ptr_ = p;
   return *this;
 }
 
-ACE_INLINE 
-Messaging::ExceptionHolder_out::operator ::Messaging::ExceptionHolder *&() // cast
+ACE_INLINE
+Messaging::ExceptionHolder_out::operator Messaging::ExceptionHolder* &() // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE ::Messaging::ExceptionHolder *&
+ACE_INLINE Messaging::ExceptionHolder* &
 Messaging::ExceptionHolder_out::ptr (void) // ptr
 {
   return this->ptr_;
 }
 
-ACE_INLINE ::Messaging::ExceptionHolder *
+ACE_INLINE Messaging::ExceptionHolder*
 Messaging::ExceptionHolder_out::operator-> (void)
 {
   return this->ptr_;
 }
 
-
 #if !defined (_MESSAGING_EXCEPTIONHOLDER__TAO_SEQ_OCTET_CI_)
 #define _MESSAGING_EXCEPTIONHOLDER__TAO_SEQ_OCTET_CI_
 
@@ -3641,20 +3710,20 @@
   return this->ptr_;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ExceptionHolder::_tao_seq_Octet_var::operator const ::Messaging::ExceptionHolder::_tao_seq_Octet &() const // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ExceptionHolder::_tao_seq_Octet_var::operator ::Messaging::ExceptionHolder::_tao_seq_Octet &() // cast 
+ACE_INLINE
+Messaging::ExceptionHolder::_tao_seq_Octet_var::operator ::Messaging::ExceptionHolder::_tao_seq_Octet &() // cast
 {
   return *this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ExceptionHolder::_tao_seq_Octet_var::operator ::Messaging::ExceptionHolder::_tao_seq_Octet &() const // cast 
+ACE_INLINE
+Messaging::ExceptionHolder::_tao_seq_Octet_var::operator ::Messaging::ExceptionHolder::_tao_seq_Octet &() const // cast
 {
   return *this->ptr_;
 }
@@ -3677,7 +3746,7 @@
   return *this->ptr_;
 }
 
-// mapping for variable size 
+// mapping for variable size
 ACE_INLINE ::Messaging::ExceptionHolder::_tao_seq_Octet *&
 Messaging::ExceptionHolder::_tao_seq_Octet_var::out (void)
 {
@@ -3738,7 +3807,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ExceptionHolder::_tao_seq_Octet_out::operator ::Messaging::ExceptionHolder::_tao_seq_Octet *&() // cast
 {
   return this->ptr_;
@@ -3765,6 +3834,10 @@
 
 #endif /* end #if !defined */
 
+
+
+#endif /* end #if !defined */
+
 ACE_INLINE
 Messaging::ReplyHandler::ReplyHandler (void) // default constructor
 {}
@@ -3832,14 +3905,14 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyHandler_var::operator const ::Messaging::ReplyHandler_ptr &() const // cast
 {
   return this->ptr_;
 }
 
-ACE_INLINE 
-Messaging::ReplyHandler_var::operator ::Messaging::ReplyHandler_ptr &() // cast 
+ACE_INLINE
+Messaging::ReplyHandler_var::operator ::Messaging::ReplyHandler_ptr &() // cast
 {
   return this->ptr_;
 }
@@ -3931,7 +4004,7 @@
   return *this;
 }
 
-ACE_INLINE 
+ACE_INLINE
 Messaging::ReplyHandler_out::operator ::Messaging::ReplyHandler_ptr &() // cast
 {
   return this->ptr_;
@@ -3952,6 +4025,8 @@
 
 #endif /* end #if !defined */
 
+#endif /* TAO_HAS_AMI_CALLBACK == 1 || TAO_HAS_AMI_POLLER == 1 */
+
 ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const Messaging::PriorityRange &_tao_aggregate)
 {
   if (
@@ -3961,7 +4036,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &strm, Messaging::PriorityRange &_tao_aggregate)
@@ -3973,7 +4048,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const Messaging::RoutingTypeRange &_tao_aggregate)
@@ -3985,7 +4060,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &strm, Messaging::RoutingTypeRange &_tao_aggregate)
@@ -3997,7 +4072,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 
@@ -4024,7 +4099,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &strm, Messaging::PolicyValue &_tao_aggregate)
@@ -4036,7 +4111,7 @@
     return 1;
   else
     return 0;
-  
+
 }
 
 
@@ -4054,45 +4129,48 @@
 
 #endif /* _TAO_CDR_OP_Messaging_PolicyValueSeq_I_ */
 
+#if (TAO_HAS_AMI_CALLBACK == 1) || (TAO_HAS_AMI_POLLER == 1)
 
-#if !defined _TAO_CDR_OP_Messaging_ExceptionHolder__tao_seq_Octet_I_
-#define _TAO_CDR_OP_Messaging_ExceptionHolder__tao_seq_Octet_I_
-
-CORBA::Boolean TAO_Export operator<< (
-    TAO_OutputCDR &,
-    const Messaging::ExceptionHolder::_tao_seq_Octet &
-  );
-CORBA::Boolean TAO_Export operator>> (
-    TAO_InputCDR &,
-    Messaging::ExceptionHolder::_tao_seq_Octet &
-  );
+ACE_INLINE CORBA::Boolean
+operator<< (TAO_OutputCDR &strm, const Messaging::ExceptionHolder *_tao_valuetype)
+{
+  return CORBA_ValueBase::_tao_marshal (strm,
+    ACE_const_cast (Messaging::ExceptionHolder*, _tao_valuetype),
+    (ptr_arith_t) &Messaging::ExceptionHolder::_downcast);
+}
 
-#endif /* _TAO_CDR_OP_Messaging_ExceptionHolder__tao_seq_Octet_I_ */
+ACE_INLINE CORBA::Boolean
+operator>> (TAO_InputCDR &strm, Messaging::ExceptionHolder *&_tao_valuetype)
+{
+  return Messaging::ExceptionHolder::_tao_unmarshal (strm, _tao_valuetype);
+}
 
-ACE_INLINE CORBA::Boolean operator<< (TAO_OutputCDR &strm, const Messaging::ExceptionHolder &_tao_aggregate)
+ACE_INLINE CORBA::Boolean
+OBV_Messaging::ExceptionHolder::_tao_marshal_state (TAO_OutputCDR &strm)
 {
   if (
-    (strm << CORBA::Any::from_boolean (_tao_aggregate.is_system_exception)) &&
-    (strm << CORBA::Any::from_boolean (_tao_aggregate.byte_order)) &&
-    (strm << _tao_aggregate.marshaled_exception)
+    (strm << CORBA::Any::from_boolean (_pd_is_system_exception)) &&
+    (strm << CORBA::Any::from_boolean (_pd_byte_order)) &&
+    (strm << _pd_marshaled_exception)
   )
     return 1;
   else
     return 0;
-  
+
 }
 
-ACE_INLINE CORBA::Boolean operator>> (TAO_InputCDR &strm, Messaging::ExceptionHolder &_tao_aggregate)
+ACE_INLINE CORBA::Boolean
+OBV_Messaging::ExceptionHolder::_tao_unmarshal_state (TAO_InputCDR &strm)
 {
   if (
-    (strm >> CORBA::Any::to_boolean (_tao_aggregate.is_system_exception)) &&
-    (strm >> CORBA::Any::to_boolean (_tao_aggregate.byte_order)) &&
-    (strm >> _tao_aggregate.marshaled_exception)
+    (strm >> CORBA::Any::to_boolean (_pd_is_system_exception)) &&
+    (strm >> CORBA::Any::to_boolean (_pd_byte_order)) &&
+    (strm >> _pd_marshaled_exception)
   )
     return 1;
   else
     return 0;
-  
+
 }
 
 ACE_INLINE CORBA::Boolean
@@ -4144,3 +4222,4 @@
   return 0;
 }
 
+#endif /* TAO_HAS_AMI_CALLBACK == 1 || TAO_HAS_AMI_POLLER == 1 */
