--- orig/OctetSeqC.h	2005-04-08 10:17:34.108726400 +0200
+++ OctetSeqC.h	2005-04-08 10:17:41.829828800 +0200
@@ -141,6 +141,18 @@
 {
 }
 
+// Workaround for a Visual Studio .NET bug where this class is not
+// properly imported by an application if typedef'd or subclassed,
+// resulting in 'multiply defined' link errors. The export macro
+// here forces an explicit import by the application. Please see
+// http://support.microsoft.com/default.aspx?scid=kb;en-us;309801
+// The problem stems from use of the type below in PortableServer,
+// but we put the instantiation here because the application will
+// need to see it in *C.h to avoid the error.
+#if defined ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION_EXPORT
+  template class TAO_Export TAO::unbounded_value_sequence<CORBA::OctetSeq>;
+#endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION_EXPORT */
+
 // TAO_IDL - Generated from
 // be\be_visitor_sequence/any_op_ch.cpp:52
 
