--- RTPortableServerC.cpp	Fri Sep 29 01:51:22 2000
+++ ..\RTPortableServerC.cpp	Fri Sep 29 02:04:44 2000
@@ -21,6 +21,8 @@
 
 #include "RTPortableServerC.h"
 
+#if (TAO_HAS_RT_CORBA == 1)
+
 #if !defined (__ACE_INLINE__)
 #include "RTPortableServerC.i"
 #endif /* !defined INLINE */
@@ -112,3 +114,4 @@
   #  pragma instantiate TAO_Object_Manager<RTPortableServer::POA,RTPortableServer::POA_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
 
+#endif /* TAO_HAS_RT_CORBA == 1 */
