--- orig/MessagingS.cpp	Tue Apr  3 22:01:25 2001
+++ patched/MessagingS.cpp	Thu Apr  5 11:10:12 2001
@@ -22,6 +22,10 @@
 #ifndef _TAO_IDL_ORIG_MESSAGINGS_CPP_
 #define _TAO_IDL_ORIG_MESSAGINGS_CPP_
 
+#include "tao/orbconf.h"
+
+#if (TAO_HAS_CORBA_MESSAGING == 1)
+
 #include "MessagingS.h"
 
 #include "tao/PortableServer/Object_Adapter.h"
@@ -44,6 +48,8 @@
 #include "MessagingS.i"
 #endif /* !defined INLINE */
 
+#if (TAO_HAS_AMI_CALLBACK == 1) || (TAO_HAS_AMI_POLLER == 1)
+
 class TAO_Messaging_ReplyHandler_Perfect_Hash_OpTable : public TAO_Perfect_Hash_OpTable
 {
 private:
@@ -449,6 +455,8 @@
   return ::Messaging::ReplyHandler::_unchecked_narrow (obj.in ());
 }
 
+#endif /* TAO_HAS_AMI_CALLBACK == 1 || TAO_HAS_AMI_POLLER == 1 */
 
+#endif /* TAO_HAS_CORBA_MESSAGING == 1 */
 
 #endif /* ifndef */
