--- orig/ServantRetentionPolicyC.h	2004-11-17 09:22:05.163302400 +0100
+++ ServantRetentionPolicyC.h	2004-11-17 09:22:13.415168000 +0100
@@ -96,6 +96,7 @@
     NON_RETAIN
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef ServantRetentionPolicyValue &ServantRetentionPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -223,11 +224,13 @@
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
 } // module PortableServer
 
 // TAO_IDL - Generated from
 // be\be_visitor_traits.cpp:59
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 // Traits specializations.
 namespace TAO
 {
@@ -273,6 +276,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, const PortableServer::ServantRetentionPolicyValue &);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::ServantRetentionPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:958
 
--- orig/ServantRetentionPolicyC.cpp	2004-11-17 09:22:05.163302400 +0100
+++ ServantRetentionPolicyC.cpp	2004-11-17 09:22:13.485268800 +0100
@@ -45,6 +45,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:64
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -260,3 +262,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantRetentionPolicyA.cpp	2004-11-17 09:22:05.173316800 +0100
+++ ServantRetentionPolicyA.cpp	2004-11-17 09:22:13.545355200 +0100
@@ -33,6 +33,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:295
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 static const CORBA::Long _oc_PortableServer_ServantRetentionPolicyValue[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
@@ -250,3 +252,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
