--- orig/ServantManagerC.h	2004-11-17 09:22:04.832827200 +0100
+++ ServantManagerC.h	2004-11-17 09:22:13.154793600 +0100
@@ -70,6 +70,8 @@
 // TAO_IDL - Generated from 
 // be\be_visitor_root/root_ch.cpp:63
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 namespace TAO
 {
   class Collocation_Proxy_Broker;
@@ -282,6 +284,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, const PortableServer::ForwardRequest &);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::ForwardRequest &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:958
 
--- orig/ServantManagerC.cpp	2004-11-17 09:22:04.832827200 +0100
+++ ServantManagerC.cpp	2004-11-17 09:22:13.224894400 +0100
@@ -51,6 +51,7 @@
 {
 }
 
+#if (TAO_HAS_MINIMUM_POA == 0)
 
 // TAO_IDL - Generated from
 // be\be_visitor_interface/interface_cs.cpp:60
@@ -469,3 +470,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
--- orig/ServantManagerA.cpp	2004-11-17 09:22:04.852856000 +0100
+++ ServantManagerA.cpp	2004-11-17 09:22:13.284980800 +0100
@@ -30,6 +30,8 @@
 #include "tao/Any_Impl_T.h"
 #include "tao/Any_Dual_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/typecode_defn.cpp:295
 
@@ -249,3 +251,5 @@
       >
 
 #endif /* !ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */ 
+
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
