--- orig/IORTableC.h	2005-02-18 12:17:41.246128000 +0100
+++ IORTableC.h	2005-02-18 12:17:41.726819200 +0100
@@ -47,6 +47,10 @@
 #include "tao/Object.h"
 #include "tao/Objref_VarOut_T.h"
 
+ #ifndef TAO_IORTABLE_SAFE_INCLUDE
+ #error "You should not include IORTableC.h directly, use IORTable.h"
+ #endif /* !TAO_IORMANIP_SAFE_INCLUDE */
+
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
--- orig/IORTableC.cpp	2005-02-18 12:17:41.246128000 +0100
+++ IORTableC.cpp	2005-02-18 12:17:41.746848000 +0100
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:302
 
 
-#include "IORTableC.h"
+#include "IORTable.h"
 #include "tao/CDR.h"
 #include "ace/OS_NS_string.h"
 
