--- orig/GIOPS.cpp	2004-11-24 10:22:34.401619200 +0100
+++ GIOPS.cpp	2004-11-24 10:27:39.330084800 +0100
@@ -34,29 +34,6 @@
 
 
 #include "GIOPS.h"
-#include "tao/PortableServer/Servant_Upcall.h"
-#include "tao/PortableServer/Operation_Table.h"
-#include "tao/TAO_Server_Request.h"
-#include "tao/ORB_Core.h"
-#include "tao/Profile.h"
-#include "tao/Stub.h"
-#include "tao/IFR_Client_Adapter.h"
-#include "tao/Object_T.h"
-#include "tao/Typecode.h"
-#include "tao/DynamicC.h"
-#include "tao/CDR.h"
-#include "tao/PortableInterceptor.h"
-#if TAO_HAS_INTERCEPTORS == 1
-#include "tao/PortableServer/PICurrent_Guard.h"
-#include "tao/PortableServer/ServerRequestInfo.h"
-#include "tao/PortableServer/ServerInterceptorAdapter.h"
-#include "tao/RequestInfo_Util.h"
-#endif  /* TAO_HAS_INTERCEPTORS == 1 */
-
-#include "ace/Dynamic_Service.h"
-#include "ace/Malloc_Allocator.h"
-#include "ace/config-all.h"
-
 #if defined (__BORLANDC__)
 #pragma option -w-rvl -w-rch -w-ccc -w-aus
 #endif /* __BORLANDC__ */
--- orig/GIOPS.h	2004-11-24 10:22:34.401619200 +0100
+++ GIOPS.h	2004-11-24 10:27:53.119913600 +0100
@@ -33,17 +33,12 @@
 
 #include /**/ "ace/pre.h"
 
-#include "tao/IOP_IORS.h"
 #include "GIOPC.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
-#include "tao/Collocation_Proxy_Broker.h"
-#include "tao/PortableServer/PortableServer.h"
-#include "tao/PortableServer/Servant_Base.h"
-
 #if defined(_MSC_VER)
 #pragma warning(push)
 #pragma warning(disable:4250)
