--- orig/PortableInterceptorC.h	2004-11-12 09:20:16.175987200 +0100
+++ PortableInterceptorC.h	2004-11-12 09:23:25.498219200 +0100
@@ -47,6 +47,12 @@
 #include "tao/Any.h"
 #include "tao/Objref_VarOut_T.h"
 
+#include "PI_ForwardC.h"
+#include "tao/Messaging_SyncScopeC.h"
+#include "tao/IOP_CodecC.h"
+#include "tao/PolicyC.h"
+#include "tao/CurrentC.h"
+
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
@@ -68,6 +74,15 @@
 #pragma option push -w-rvl -w-rch -w-ccc -w-inl
 #endif /* __BORLANDC__ */
 
+// Hand crafted, keeps DynamicC.h and Typecode.h out of header files.
+namespace Dynamic
+{
+  class ParameterList;
+  class ExceptionList;
+  typedef CORBA::StringSeq ContextList;
+  typedef CORBA::StringSeq RequestContext;
+}
+
 // TAO_IDL - Generated from 
 // be\be_visitor_root/root_ch.cpp:63
 
@@ -1312,17 +1327,19 @@
     
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
+#if TAO_HAS_EXTENDED_FT_INTERCEPTORS == 1
     
     virtual void tao_ft_interception_point (
         ::PortableInterceptor::ServerRequestInfo_ptr ri,
-        ::CORBA::OctetSeq_out os
-        ACE_ENV_ARG_DECL_WITH_DEFAULTS
+        CORBA::OctetSeq_out
+        ACE_ENV_ARG_DECL_NOT_USED
       )
       ACE_THROW_SPEC ((
         CORBA::SystemException,
         ::PortableInterceptor::ForwardRequest
-      )) = 0;
+        )) {}
     
+#endif /*TAO_HAS_EXTENDED_FT_INTERCEPTORS*/
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
@@ -1469,6 +1486,16 @@
     
     static void _tao_any_destructor (void *);
     
+    // Hand-crafted addition.
+    virtual ::CORBA::Policy_ptr _create_policy (
+        CORBA::PolicyType type
+        ACE_ENV_ARG_DECL_WITH_DEFAULTS
+      )
+      ACE_THROW_SPEC ((
+        CORBA::SystemException
+        , CORBA::PolicyError
+      ));
+
     // TAO_IDL - Generated from
     // be\be_visitor_operation/operation_ch.cpp:46
     
--- orig/PortableInterceptorC.cpp	2004-11-12 09:20:16.175987200 +0100
+++ PortableInterceptorC.cpp	2004-11-12 09:20:16.726779200 +0100
@@ -1821,6 +1821,18 @@
   return 0;
 }
 
+// Hand-crafted.
+
+::CORBA::Policy_ptr
+PortableInterceptor::PolicyFactory::_create_policy (CORBA::PolicyType
+                                                    ACE_ENV_ARG_DECL)
+  ACE_THROW_SPEC ((CORBA::SystemException,
+                   CORBA::PolicyError))
+{
+  ACE_THROW_RETURN (CORBA::PolicyError (CORBA::BAD_POLICY_TYPE),
+                    CORBA::Policy::_nil ());
+}
+
 // TAO_IDL - Generated from
 // be\be_visitor_interface/interface_cs.cpp:60
 
