--- orig/DynamicAnyC.cpp	Thu Dec 07 17:45:21 2000
+++ DynamicAnyC.cpp	Thu Dec 07 17:53:48 2000
@@ -19,7 +19,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "DynamicAnyC.h"
+#include "DynamicAny.h"
 
 #if !defined (__ACE_INLINE__)
 #include "DynamicAnyC.i"
@@ -28,8 +28,7 @@
 
 // default constructor
 DynamicAny::DynAny::DynAny ()
-{
-  }
+{}
 
 // destructor
 DynamicAny::DynAny::~DynAny (void)
@@ -233,8 +232,7 @@
 
 // default constructor
 DynamicAny::DynFixed::DynFixed ()
-{
-  }
+{}
 
 // destructor
 DynamicAny::DynFixed::~DynFixed (void)
@@ -314,8 +312,7 @@
 
 // default constructor
 DynamicAny::DynEnum::DynEnum ()
-{
-  }
+{}
 
 // destructor
 DynamicAny::DynEnum::~DynEnum (void)
@@ -665,8 +662,7 @@
 
 // default constructor
 DynamicAny::DynStruct::DynStruct ()
-{
-  }
+{}
 
 // destructor
 DynamicAny::DynStruct::~DynStruct (void)
@@ -746,8 +742,7 @@
 
 // default constructor
 DynamicAny::DynUnion::DynUnion ()
-{
-  }
+{}
 
 // destructor
 DynamicAny::DynUnion::~DynUnion (void)
@@ -1071,8 +1066,7 @@
 
 // default constructor
 DynamicAny::DynSequence::DynSequence ()
-{
-  }
+{}
 
 // destructor
 DynamicAny::DynSequence::~DynSequence (void)
@@ -1152,8 +1146,7 @@
 
 // default constructor
 DynamicAny::DynArray::DynArray ()
-{
-  }
+{}
 
 // destructor
 DynamicAny::DynArray::~DynArray (void)
@@ -1233,8 +1226,7 @@
 
 // default constructor
 DynamicAny::DynValue::DynValue ()
-{
-  }
+{}
 
 // destructor
 DynamicAny::DynValue::~DynValue (void)
@@ -1314,8 +1306,7 @@
 
 // default constructor
 DynamicAny::DynAnyFactory::DynAnyFactory ()
-{
-  }
+{}
 
 // destructor
 DynamicAny::DynAnyFactory::~DynAnyFactory (void)
