--- orig/ThreadPolicyC.h	2005-04-11 16:44:02.895027200 +0200
+++ ThreadPolicyC.h	2005-04-11 16:51:37.889276800 +0200
@@ -78,6 +78,7 @@
     SINGLE_THREAD_MODEL
   };
   
+#if (TAO_HAS_MINIMUM_POA == 0)
   typedef ThreadPolicyValue &ThreadPolicyValue_out;
   
   // TAO_IDL - Generated from
@@ -204,6 +205,8 @@
   
   extern TAO_PortableServer_Export ::CORBA::TypeCode_ptr const _tc_ThreadPolicy;
 
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_visitor_module/module_ch.cpp:66
 
@@ -216,6 +219,8 @@
 namespace TAO
 {
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 #if !defined (_PORTABLESERVER_THREADPOLICY__TRAITS_CH_)
 #define _PORTABLESERVER_THREADPOLICY__TRAITS_CH_
   
@@ -236,11 +241,16 @@
   };
 
 #endif /* end #if !defined */
+
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
+
 }
 
 // TAO_IDL - Generated from
 // be\be_visitor_enum/any_op_ch.cpp:51
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 TAO_PortableServer_Export void operator<<= (CORBA::Any &, PortableServer::ThreadPolicyValue);
 TAO_PortableServer_Export CORBA::Boolean operator>>= (const CORBA::Any &, PortableServer::ThreadPolicyValue &);
 
@@ -257,6 +267,8 @@
 TAO_PortableServer_Export CORBA::Boolean operator<< (TAO_OutputCDR &, PortableServer::ThreadPolicyValue);
 TAO_PortableServer_Export CORBA::Boolean operator>> (TAO_InputCDR &, PortableServer::ThreadPolicyValue &);
 
+#endif /* TAO_HAS_MINIMUM_CORBA == 0 */
+
 // TAO_IDL - Generated from
 // be\be_codegen.cpp:955
 
--- orig/ThreadPolicyC.cpp	2005-04-11 16:44:02.895027200 +0200
+++ ThreadPolicyC.cpp	2005-04-11 16:51:11.561419200 +0200
@@ -40,6 +40,8 @@
 // TAO_IDL - Generated from
 // be\be_visitor_arg_traits.cpp:69
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // Arg traits specializations.
 namespace TAO
 {
@@ -211,3 +213,5 @@
   
   return _tao_success;
 }
+
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
--- orig/ThreadPolicyA.cpp	2005-04-11 16:44:02.895027200 +0200
+++ ThreadPolicyA.cpp	2005-04-11 16:51:08.827488000 +0200
@@ -36,6 +36,8 @@
 #include "tao/Any_Impl_T.h"
 #include "tao/Any_Basic_Impl_T.h"
 
+#if (TAO_HAS_MINIMUM_POA == 0)
+
 // TAO_IDL - Generated from
 // be\be_visitor_typecode/enum_typecode.cpp:34
 
@@ -177,3 +179,5 @@
         _tao_elem
       );
 }
+
+#endif /* TAO_HAS_MINIMUM_POA == 0 */
