--- orig/TAOC.h	Tue Apr 16 17:44:38 2002
+++ TAOC.h	Tue Apr 16 17:47:25 2002
@@ -23,16 +23,15 @@
 #define _TAO_IDL_ORIG_TAOC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+
+#include "tao/corbafwd.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
-#include "tao/corbafwd.h"
 #include "PolicyC.h"
 #include "TimeBaseC.h"
-#include "Messaging_SyncScopeC.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
@@ -59,6 +58,8 @@
 
 TAO_NAMESPACE  TAO
 {
+#if (TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1)
+
   typedef CORBA::UShort BufferingConstraintMode;
   typedef CORBA::UShort_out BufferingConstraintMode_out;
   TAO_NAMESPACE_STORAGE_CLASS ::CORBA::TypeCode_ptr _tc_BufferingConstraintMode;
@@ -275,14 +276,17 @@
   
   TAO_NAMESPACE_STORAGE_CLASS ::CORBA::TypeCode_ptr _tc_BufferingConstraintPolicy;
   
+#endif  /* TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1 */
+
   TAO_NAMESPACE_STORAGE_CLASS const CORBA::Short SYNC_EAGER_BUFFERING;
   
   TAO_NAMESPACE_STORAGE_CLASS const CORBA::Short SYNC_DELAYED_BUFFERING;
   
-  
 }
 TAO_NAMESPACE_CLOSE // module TAO
 
+#if (TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1)
+
 TAO_Export void operator<<= (CORBA::Any &, const TAO::BufferingConstraint &); // copying version
 TAO_Export void operator<<= (CORBA::Any &, TAO::BufferingConstraint*); // noncopying version
 TAO_Export CORBA::Boolean operator>>= (const CORBA::Any &, TAO::BufferingConstraint *&); // deprecated
@@ -303,6 +307,8 @@
 #if defined (__ACE_INLINE__)
 #include "TAOC.i"
 #endif /* defined INLINE */
+
+#endif  /* TAO_HAS_BUFFERING_CONSTRAINT_POLICY == 1 */
 
 #if defined(_MSC_VER) && (_MSC_VER >= 1200)
 #pragma warning(pop)
