--- orig/IORTableC.cpp	Sun Sep 10 16:20:41 2000
+++ IORTableC.cpp	Sun Sep 10 16:22:48 2000
@@ -63,23 +63,19 @@
 
 
 void IORTable::AlreadyBound::_tao_encode (
-    TAO_OutputCDR &cdr,
+    TAO_OutputCDR &,
     CORBA::Environment &ACE_TRY_ENV
   ) const
 {
-  if (cdr << *this)
-    return;
   ACE_THROW (CORBA::MARSHAL ());
 }
 
 
 void IORTable::AlreadyBound::_tao_decode (
-    TAO_InputCDR &cdr,
+    TAO_InputCDR &,
     CORBA::Environment &ACE_TRY_ENV
   )
 {
-  if (cdr >> *this)
-    return;
   ACE_THROW (CORBA::MARSHAL ());
 }
 
@@ -141,23 +137,19 @@
 
 
 void IORTable::NotFound::_tao_encode (
-    TAO_OutputCDR &cdr,
+    TAO_OutputCDR &,
     CORBA::Environment &ACE_TRY_ENV
   ) const
 {
-  if (cdr << *this)
-    return;
   ACE_THROW (CORBA::MARSHAL ());
 }
 
 
 void IORTable::NotFound::_tao_decode (
-    TAO_InputCDR &cdr,
+    TAO_InputCDR &,
     CORBA::Environment &ACE_TRY_ENV
   )
 {
-  if (cdr >> *this)
-    return;
   ACE_THROW (CORBA::MARSHAL ());
 }
 
