--- y.tab.cpp.orig	2005-01-28 10:00:07.000000000 -0600
+++ y.tab.cpp	2005-01-28 10:07:56.000000000 -0600
@@ -1759,11 +1759,6 @@
   else
     TAO_YYFPRINTF (tao_yyoutput, "nterm %s (", tao_yytname[tao_yytype]);
 
-  switch (tao_yytype)
-    {
-      default:
-        break;
-    }
   TAO_YYFPRINTF (tao_yyoutput, ")");
 }
 
@@ -1778,14 +1778,9 @@
 #endif
 {
   /* Pacify ``unused variable'' warnings.  */
+  (void) tao_yytype;
   (void) tao_yyvaluep;
 
-  switch (tao_yytype)
-    {
-
-      default:
-        break;
-    }
 }
 
@@ -8147,7 +8147,7 @@
 `---------------------------------------------------*/
 tao_yyerrorlab:
 
-#ifdef __GNUC__
+#if defined (__GNUC__) || defined (WIN32) || defined (__HP_aCC)
   /* Pacify GCC when the user code never invokes TAO_YYERROR and the label
      tao_yyerrorlab therefore never appears in user code.  */
   if (0)
