--- orig/IOP_COdecC.h	2005-06-07 13:09:58.255201600 +0200
+++ IOP_COdecC.h	2005-06-07 13:45:14.618382400 +0200
@@ -33,6 +33,9 @@
 
 #include /**/ "ace/pre.h"
 
+#ifndef TAO_CODECFACTORY_SAFE_INCLUDE
+#error "You should not include IOP_CodecC.h directly, use CodecFactory.h"
+#endif /* !TAO_CODECFACTORY_SAFE_INCLUDE */
 
 #include "ace/config-all.h"
 
--- orig/IOP_COdecC.cpp	2005-06-07 13:09:58.255201600 +0200
+++ IOP_COdecC.cpp	2005-06-07 13:43:36.437204800 +0200
@@ -29,7 +29,7 @@
 // be\be_codegen.cpp:291
 
 
-#include "IOP_CodecC.h"
+#include "CodecFactory.h"
 #include "tao/CDR.h"
 #include "tao/ORB_Core.h"
 #include "ace/OS_NS_string.h"
--- orig/IOP_COdecA.cpp	2005-06-07 13:09:58.255201600 +0200
+++ IOP_COdecA.cpp	2005-06-07 13:43:11.170873600 +0200
@@ -25,7 +25,7 @@
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "IOP_CodecC.h"
+#include "CodecFactory.h"
 #include "tao/Null_RefCount_Policy.h"
 #include "tao/TypeCode_Constants.h"
 #include "tao/Alias_TypeCode_Static.h"
