--- orig/IOP_IORC.h	2003-12-07 09:24:45.000000000 -0600
+++ IOP_IORC.h	2003-12-07 09:27:36.000000000 -0600
@@ -33,21 +33,15 @@
 
 #include /**/ "ace/pre.h"
 
-#include "tao/ORB.h"
+#include "OctetSeqC.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
-#include "TAO_Export.h"
-#include "tao/CDR.h"
-#include "tao/Environment.h"
-#include "tao/Sequence_T.h"
-#include "tao/Seq_Var_T.h"
-#include "tao/Seq_Out_T.h"
 #include "tao/VarOut_T.h"
+#include "tao/CDR.h"
 
-#include "OctetSeqC.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
