--- orig/RTSchedulerC.h	2003-12-29 10:11:07.000000000 -0600
+++ RTSchedulerC.h	2003-12-29 10:14:38.000000000 -0600
@@ -40,18 +40,15 @@
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 #include "rtscheduler_export.h"
-#include "tao/CDR.h"
-#include "tao/Environment.h"
-#include "tao/Object.h"
-#include "tao/Sequence_T.h"
-#include "tao/Objref_VarOut_T.h"
-#include "tao/Seq_Var_T.h"
-#include "tao/Seq_Out_T.h"
 
-#include "tao/RTCORBA/RTCORBAC.h"
+#include "tao/RTCORBA/RTCORBA.h"
 #include "tao/PortableServer/PortableServerC.h"
 #include "tao/PortableInterceptorC.h"
 
+#ifndef TAO_RTSCHEDULER_SAFE_INCLUDE
+#error "You should not include RTSchedulerC.h directly, use RTScheduler.h"
+#endif /* !TAO_RTSCHEDULER_SAFE_INCLUDE */
+
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
@@ -94,6 +91,7 @@
 
 // TAO_IDL - Generated from
 // be/be_visitor_module/module_ch.cpp:66
+  typedef void* VoidData;
 
 } // module CORBA
 
--- orig/RTSchedulerC.cpp	2003-12-29 10:11:07.000000000 -0600
+++ RTSchedulerC.cpp	2003-12-29 10:13:12.000000000 -0600
@@ -29,7 +29,7 @@
 // be/be_codegen.cpp:338
 
 
-#include "RTSchedulerC.h"
+#include "RTScheduler.h"
 #include "tao/ORB_Core.h"
 #include "tao/Typecode.h"
 #include "tao/Var_Size_Argument_T.h"
