// -*- C++ -*-
--- orig/RTCORBAC.h     2004-09-16 12:52:02.885009600 +0200
+++ RTCORBAC.h  2004-09-16 12:36:01.362408000 +0200
@@ -36,12 +36,18 @@
 
 #include "ace/config-all.h"
 
+#ifndef TAO_RTCORBA_SAFE_INCLUDE
+#error "You should not include RTCORBAC.h directly, use RTCORBA.h"
+#endif /* !TAO_RTCORBA_SAFE_INCLUDE */
+
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
 #include "rtcorba_export.h"
+#include "rtcorba_typedefs.h"
 #include "tao/ORB.h"
+#include "tao/TimeBaseC.h"
 #include "tao/SystemException.h"
 #include "tao/Environment.h"
 #include "tao/Object.h"
@@ -50,6 +56,8 @@
 #include "tao/Seq_Var_T.h"
 #include "tao/Seq_Out_T.h"
 #include "tao/VarOut_T.h"
+#include "tao/Encodable.h"
+#include "tao/PolicyC.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
@@ -532,6 +540,7 @@
   
   class TAO_RTCORBA_Export ProtocolProperties
     : public virtual CORBA::Object
+    , public virtual TAO_Encodable
   {
   public:
     typedef ProtocolProperties_ptr _ptr_type;
@@ -568,6 +577,9 @@
     virtual const char* _interface_repository_id (void) const;
     virtual CORBA::Boolean marshal (TAO_OutputCDR &cdr);
   
+    // CDR Encoder/Decoder methods. (handcrafted)
+    virtual CORBA::Boolean _tao_encode (TAO_OutputCDR &out_cdr);
+    virtual CORBA::Boolean _tao_decode (TAO_InputCDR &in_cdr);
   protected:
     // Abstract or local interface only.
     ProtocolProperties (void);
--- orig/RTCORBAC.cpp   2004-09-16 12:52:02.885009600 +0200
+++ RTCORBAC.cpp        2004-09-16 12:36:36.643139200 +0200
@@ -29,7 +29,7 @@
 // be/be_codegen.cpp:303
 
 
-#include "RTCORBAC.h"
+#include "RTCORBA.h"
 #include "tao/CDR.h"
 #include "tao/Typecode.h"
 #include "tao/Any_Impl_T.h"
@@ -56,6 +56,22 @@
 {
 }
 
+//////////////////////////////////////////////////////
+// CDR Encoding Support hand crafted
+
+CORBA::Boolean
+RTCORBA::ProtocolProperties::_tao_encode (TAO_OutputCDR &)
+{
+  return 1;
+}
+// TAO_IDL - Generated from
+// be/be_visitor_arg_traits.cpp:60
+
+CORBA::Boolean
+RTCORBA::ProtocolProperties::_tao_decode (TAO_InputCDR &)
+{
+  return 1;
+}
 
 // TAO_IDL - Generated from
 // be/be_visitor_typecode/typecode_defn.cpp:290
