--- CurrentC.cpp	Sat Jun 24 02:48:51 2000
+++ CurrentC.cpp.mod	Sat Jun 24 02:47:40 2000
@@ -7,10 +7,10 @@
 // Information about TAO is available at:
 //                 http://www.cs.wustl.edu/~schmidt/TAO.html
 
-#include "CurrentC.h"
+#include "tao/CurrentC.h"
 
 #if !defined (__ACE_INLINE__)
-#include "CurrentC.i"
+#include "tao/CurrentC.i"
 #endif /* !defined INLINE */
 
 
@@ -78,7 +78,7 @@
 
 const char* CORBA_Current::_interface_repository_id (void) const
 {
-  return "IDL:omg.org/CORBA_Current:1.0";
+  return "IDL:omg.org/CORBA/Current:1.0";
 }
 
 #if defined (ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION)  || \
@@ -87,4 +87,3 @@
   #elif defined (ACE_HAS_TEMPLATE_INSTANTIATION_PRAGMA)
   #  pragma instantiate TAO_Object_Manager<CORBA_Current,CORBA_Current_var>
 #endif /* ACE_HAS_EXPLICIT_TEMPLATE_INSTANTIATION */
-
