--- IOPC.h.old	Sun Sep 24 17:32:44 2000
+++ IOPC.h	Sun Sep 24 19:47:04 2000
@@ -21,17 +21,18 @@
 
 #ifndef _TAO_IDL_IOPC_H_
 #define _TAO_IDL_IOPC_H_
 
 #include "ace/pre.h"
-#include "tao/corba.h"
+#include "tao/TAO_Export.h"
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
-#include "tao/TAO_Export.h"
+#include "tao/CDR.h"
+#include "tao/Sequence.h"
 
 #if defined (TAO_EXPORT_MACRO)
 #undef TAO_EXPORT_MACRO
 #endif
 #define TAO_EXPORT_MACRO TAO_Export
--- IOPS.h.old	Sun Sep 24 17:32:44 2000
+++ IOPS.h	Sun Sep 24 19:47:08 2000
@@ -28,14 +28,10 @@
 
 #if !defined (ACE_LACKS_PRAGMA_ONCE)
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
-#include "tao/PortableServer/PortableServer.h"
-#include "tao/PortableServer/Servant_Base.h"
-#include "tao/PortableServer/Collocated_Object.h"
-
 #if defined(_MSC_VER)
 #if (_MSC_VER >= 1200)
 #pragma warning(push)
 #endif /* _MSC_VER >= 1200 */
 #pragma warning(disable:4250)
--- IOPC.cpp.old	Sun Sep 24 17:32:44 2000
+++ IOPC.cpp	Sun Sep 24 19:47:04 2000
@@ -23,10 +23,13 @@
 
 #if !defined (__ACE_INLINE__)
 #include "IOPC.i"
 #endif /* !defined INLINE */
 
+#include "tao/Any.h"
+#include "tao/Typecode.h"
+
 static const CORBA::Long _oc_IOP_ProfileId[] =
 {
   TAO_ENCAP_BYTE_ORDER, // byte order
   30, ACE_NTOHL (0x49444c3a), ACE_NTOHL (0x6f6d672e), ACE_NTOHL (0x6f72672f), ACE_NTOHL (0x494f502f), ACE_NTOHL (0x50726f66), ACE_NTOHL (0x696c6549), ACE_NTOHL (0x643a312e), ACE_NTOHL (0x30000000),  // repository ID = IDL:omg.org/IOP/ProfileId:1.0
   10, ACE_NTOHL (0x50726f66), ACE_NTOHL (0x696c6549), ACE_NTOHL (0x64000000),  // name = ProfileId
--- IOPS.cpp.old	Sun Sep 24 17:32:44 2000
+++ IOPS.cpp	Sun Sep 24 19:47:08 2000
@@ -18,12 +18,10 @@
 //
 // Information about TAO is available at:
 //     http://www.cs.wustl.edu/~schmidt/TAO.html
 
 #include "IOPS.h"
-#include "tao/PortableServer/Object_Adapter.h"
-#include "tao/PortableServer/Operation_Table.h"
 
 #if !defined (__ACE_INLINE__)
 #include "IOPS.i"
 #endif /* !defined INLINE */
 
